
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/V.%2016S_amplicon_analysis/05_01_Introduction/">
      
      
        <link rel="prev" href="../../IV.%20Phylogenetics/04_08_Tanglegram/">
      
      
        <link rel="next" href="../05_02_Crohns_disease/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.38">
    
    
      
        <title>16S analysis with DADA2 - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#16s-analysis-with-dada2" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              16S analysis with DADA2
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    NGS Analysis Handbook
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome to MkDocs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../00_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../01_Quality_Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../02_Variant_Calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../03_Whole_%28pan%29genome_analyses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_03_MSA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_04_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_05_Root_Date/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_07_Visualization_Pro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../IV.%20Phylogenetics/04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" checked>
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-downloading-the-necessary-packages-and-data" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Downloading the necessary packages and data
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-building-an-error-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Building an error model
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-dereplication-and-denoising" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Dereplication and "denoising"
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-merging-reads-summary-statistics" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Merging reads, summary statistics
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-removal-of-chimeras" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Removal of chimeras
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-taxonomy-identification" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Taxonomy identification
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-additional-translation-into-phyloseq-format" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7 (additional): Translation into Phyloseq format
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-downloading-the-necessary-packages-and-data" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Downloading the necessary packages and data
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-building-an-error-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Building an error model
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-dereplication-and-denoising" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Dereplication and "denoising"
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-merging-reads-summary-statistics" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Merging reads, summary statistics
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-removal-of-chimeras" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Removal of chimeras
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-taxonomy-identification" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Taxonomy identification
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-additional-translation-into-phyloseq-format" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7 (additional): Translation into Phyloseq format
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="16s-analysis-with-dada2"><strong>16S analysis with DADA2</strong></h1>
<blockquote>
<p>This folder contains a manual on 16S analysis with DADA2</p>
</blockquote>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p>You can run commands below in your <code>RStudio</code> in <code>R script</code>.<br>
Or if you want to write a beautiful &amp; convenient to read laboratory journal you can use <code>R Markdown</code>.</p>
<hr />
<h2 id="step-1-downloading-the-necessary-packages-and-data"><strong>Step 1: Downloading the necessary packages and data</strong></h2>
<p>First, let's download the necessary packages.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="nf">if </span><span class="p">(</span><span class="o">!</span><span class="nf">requireNamespace</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="w">    </span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;BiocManager&quot;</span><span class="p">)</span>
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a><span class="c1">#BiocManager::install(&quot;dada2&quot;)</span>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a><span class="nf">library</span><span class="p">(</span><span class="n">dada2</span><span class="p">)</span>
</span></code></pre></div>
<p>Then let's set the working directory.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="n">main_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dirname</span><span class="p">(</span><span class="n">rstudioapi</span><span class="o">::</span><span class="nf">getSourceEditorContext</span><span class="p">()</span><span class="o">$</span><span class="n">path</span><span class="p">)</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="nf">setwd</span><span class="p">(</span><span class="n">main_dir</span><span class="p">)</span>
</span></code></pre></div>
<p>The data we will be working with are the same as those used in the mothur MiSeq SOP. The data can be downloaded here: https://mothur.s3.us-east-2.amazonaws.com/wiki/miseqsopdata.zip.<br></p>
<p>Sometimes <code>R</code> does not like to download files. If it falls with error - download a zip archive manually and move it into working directory.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="n">url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;https://mothur.s3.us-east-2.amazonaws.com/wiki/miseqsopdata.zip&quot;</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a><span class="n">zipF</span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;miseqsopdata.zip&quot;</span>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a><span class="nf">download.file</span><span class="p">(</span><span class="n">url</span><span class="p">,</span><span class="w"> </span><span class="n">zipF</span><span class="p">)</span>
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="n">outDir</span><span class="o">&lt;-</span><span class="s">&quot;miseqsopdata&quot;</span>
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a><span class="nf">unzip</span><span class="p">(</span><span class="n">zipF</span><span class="p">,</span><span class="n">exdir</span><span class="o">=</span><span class="n">outDir</span><span class="p">)</span>
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a>
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a><span class="nf">if </span><span class="p">(</span><span class="nf">file.exists</span><span class="p">(</span><span class="n">zipF</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
</span><span id="__span-2-12"><a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a><span class="w">  </span><span class="nf">file.remove</span><span class="p">(</span><span class="n">zipF</span><span class="p">)</span>
</span><span id="__span-2-13"><a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a><span class="p">}</span>
</span></code></pre></div>
<p>These are the results of sequencing 2x250 Illumina Miseq amplicons of the V4 region of the 16S rRNA gene from gut samples collected from mice after weaning. For now, these are just paired reads for us.<br></p>
<p>Here we look at the contents of the folder and save the filenames of the forward and reverse reads as separate vectors (<code>forward.raw</code> and <code>reverse.raw</code>), distinguishing them by suffix. The last line creates a list of samples, removing all characters except the sample number.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="n">path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;miseqsopdata/MiSeq_SOP&#39;</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="p">)</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="n">fnFs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="o">=</span><span class="s">&quot;_R1_001.fastq&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="n">fnRs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="nf">list.files</span><span class="p">(</span><span class="n">path</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="o">=</span><span class="s">&quot;_R2_001.fastq&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">full.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="n">sample.names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="nf">basename</span><span class="p">(</span><span class="n">fnFs</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">`[`</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a> [1] &quot;F3D0_S188_L001_R1_001.fastq&quot;   &quot;F3D0_S188_L001_R2_001.fastq&quot;  
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a> [3] &quot;F3D1_S189_L001_R1_001.fastq&quot;   &quot;F3D1_S189_L001_R2_001.fastq&quot;  
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a> [5] &quot;F3D141_S207_L001_R1_001.fastq&quot; &quot;F3D141_S207_L001_R2_001.fastq&quot;
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a> [7] &quot;F3D142_S208_L001_R1_001.fastq&quot; &quot;F3D142_S208_L001_R2_001.fastq&quot;
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a> [9] &quot;F3D143_S209_L001_R1_001.fastq&quot; &quot;F3D143_S209_L001_R2_001.fastq&quot;
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>[11] &quot;F3D144_S210_L001_R1_001.fastq&quot; &quot;F3D144_S210_L001_R2_001.fastq&quot;
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a>[13] &quot;F3D145_S211_L001_R1_001.fastq&quot; &quot;F3D145_S211_L001_R2_001.fastq&quot;
</span><span id="__span-4-8"><a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a>[15] &quot;F3D146_S212_L001_R1_001.fastq&quot; &quot;F3D146_S212_L001_R2_001.fastq&quot;
</span><span id="__span-4-9"><a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a>[17] &quot;F3D147_S213_L001_R1_001.fastq&quot; &quot;F3D147_S213_L001_R2_001.fastq&quot;
</span><span id="__span-4-10"><a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a>[19] &quot;F3D148_S214_L001_R1_001.fastq&quot; &quot;F3D148_S214_L001_R2_001.fastq&quot;
</span><span id="__span-4-11"><a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a>[21] &quot;F3D149_S215_L001_R1_001.fastq&quot; &quot;F3D149_S215_L001_R2_001.fastq&quot;
</span><span id="__span-4-12"><a id="__codelineno-4-12" name="__codelineno-4-12" href="#__codelineno-4-12"></a>[23] &quot;F3D150_S216_L001_R1_001.fastq&quot; &quot;F3D150_S216_L001_R2_001.fastq&quot;
</span><span id="__span-4-13"><a id="__codelineno-4-13" name="__codelineno-4-13" href="#__codelineno-4-13"></a>[25] &quot;F3D2_S190_L001_R1_001.fastq&quot;   &quot;F3D2_S190_L001_R2_001.fastq&quot;  
</span><span id="__span-4-14"><a id="__codelineno-4-14" name="__codelineno-4-14" href="#__codelineno-4-14"></a>[27] &quot;F3D3_S191_L001_R1_001.fastq&quot;   &quot;F3D3_S191_L001_R2_001.fastq&quot;  
</span><span id="__span-4-15"><a id="__codelineno-4-15" name="__codelineno-4-15" href="#__codelineno-4-15"></a>[29] &quot;F3D5_S193_L001_R1_001.fastq&quot;   &quot;F3D5_S193_L001_R2_001.fastq&quot;  
</span><span id="__span-4-16"><a id="__codelineno-4-16" name="__codelineno-4-16" href="#__codelineno-4-16"></a>[31] &quot;F3D6_S194_L001_R1_001.fastq&quot;   &quot;F3D6_S194_L001_R2_001.fastq&quot;  
</span><span id="__span-4-17"><a id="__codelineno-4-17" name="__codelineno-4-17" href="#__codelineno-4-17"></a>[33] &quot;F3D7_S195_L001_R1_001.fastq&quot;   &quot;F3D7_S195_L001_R2_001.fastq&quot;  
</span><span id="__span-4-18"><a id="__codelineno-4-18" name="__codelineno-4-18" href="#__codelineno-4-18"></a>[35] &quot;F3D8_S196_L001_R1_001.fastq&quot;   &quot;F3D8_S196_L001_R2_001.fastq&quot;  
</span><span id="__span-4-19"><a id="__codelineno-4-19" name="__codelineno-4-19" href="#__codelineno-4-19"></a>[37] &quot;F3D9_S197_L001_R1_001.fastq&quot;   &quot;F3D9_S197_L001_R2_001.fastq&quot;  
</span><span id="__span-4-20"><a id="__codelineno-4-20" name="__codelineno-4-20" href="#__codelineno-4-20"></a>[39] &quot;filtered&quot;                      &quot;HMP_MOCK.v35.fasta&quot;           
</span><span id="__span-4-21"><a id="__codelineno-4-21" name="__codelineno-4-21" href="#__codelineno-4-21"></a>[41] &quot;Mock_S280_L001_R1_001.fastq&quot;   &quot;Mock_S280_L001_R2_001.fastq&quot;  
</span><span id="__span-4-22"><a id="__codelineno-4-22" name="__codelineno-4-22" href="#__codelineno-4-22"></a>[43] &quot;mouse.dpw.metadata&quot;            &quot;mouse.time.design&quot;            
</span><span id="__span-4-23"><a id="__codelineno-4-23" name="__codelineno-4-23" href="#__codelineno-4-23"></a>[45] &quot;stability.batch&quot;               &quot;stability.files&quot;              
</span></code></pre></div>
<p>Now let's check the quality of the data. This function creates graphs similar to <code>FastQC</code> or <code>multiQC</code> graphs.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="nf">plotQualityProfile</span><span class="p">(</span><span class="n">fnFs</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">])</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<p><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/qc_forward.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/qc_forward.png" width="100%"></a></p>
<p>We can do the same for reverse reads.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="nf">plotQualityProfile</span><span class="p">(</span><span class="n">fnRs</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">])</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<p><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/qc_reverse.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/qc_reverse.png" width="100%"></a></p>
<p>Now we can filter by quality and trim reads based on the resulting graphs:<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="n">filt_path</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">path</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;filtered&quot;</span><span class="p">)</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="n">filtFs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">filt_path</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">sample.names</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_F_filt.fastq.gz&quot;</span><span class="p">))</span>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a><span class="n">filtRs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">file.path</span><span class="p">(</span><span class="n">filt_path</span><span class="p">,</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">sample.names</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_R_filt.fastq.gz&quot;</span><span class="p">))</span>
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a><span class="n">out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filterAndTrim</span><span class="p">(</span><span class="n">fnFs</span><span class="p">,</span><span class="w"> </span><span class="n">filtFs</span><span class="p">,</span><span class="w"> </span><span class="n">fnRs</span><span class="p">,</span><span class="w"> </span><span class="n">filtRs</span><span class="p">,</span><span class="w"> </span><span class="n">truncLen</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">240</span><span class="p">,</span><span class="m">160</span><span class="p">),</span>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="w">              </span><span class="n">maxN</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">maxEE</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="n">truncQ</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">rm.phix</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="w">              </span><span class="n">compress</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1"># On Windows set multithread=FALSE</span>
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a><span class="n">out</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>                              reads.in reads.out
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>F3D0_S188_L001_R1_001.fastq       7793      7113
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>F3D1_S189_L001_R1_001.fastq       5869      5299
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>F3D141_S207_L001_R1_001.fastq     5958      5463
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>F3D142_S208_L001_R1_001.fastq     3183      2914
</span><span id="__span-8-6"><a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a>F3D143_S209_L001_R1_001.fastq     3178      2941
</span><span id="__span-8-7"><a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a>F3D144_S210_L001_R1_001.fastq     4827      4312
</span><span id="__span-8-8"><a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a>F3D145_S211_L001_R1_001.fastq     7377      6741
</span><span id="__span-8-9"><a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a>F3D146_S212_L001_R1_001.fastq     5021      4560
</span><span id="__span-8-10"><a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a>F3D147_S213_L001_R1_001.fastq    17070     15637
</span><span id="__span-8-11"><a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a>F3D148_S214_L001_R1_001.fastq    12405     11413
</span><span id="__span-8-12"><a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a>F3D149_S215_L001_R1_001.fastq    13083     12017
</span><span id="__span-8-13"><a id="__codelineno-8-13" name="__codelineno-8-13" href="#__codelineno-8-13"></a>F3D150_S216_L001_R1_001.fastq     5509      5032
</span><span id="__span-8-14"><a id="__codelineno-8-14" name="__codelineno-8-14" href="#__codelineno-8-14"></a>F3D2_S190_L001_R1_001.fastq      19620     18075
</span><span id="__span-8-15"><a id="__codelineno-8-15" name="__codelineno-8-15" href="#__codelineno-8-15"></a>F3D3_S191_L001_R1_001.fastq       6758      6250
</span><span id="__span-8-16"><a id="__codelineno-8-16" name="__codelineno-8-16" href="#__codelineno-8-16"></a>F3D5_S193_L001_R1_001.fastq       4448      4052
</span><span id="__span-8-17"><a id="__codelineno-8-17" name="__codelineno-8-17" href="#__codelineno-8-17"></a>F3D6_S194_L001_R1_001.fastq       7989      7369
</span><span id="__span-8-18"><a id="__codelineno-8-18" name="__codelineno-8-18" href="#__codelineno-8-18"></a>F3D7_S195_L001_R1_001.fastq       5129      4765
</span><span id="__span-8-19"><a id="__codelineno-8-19" name="__codelineno-8-19" href="#__codelineno-8-19"></a>F3D8_S196_L001_R1_001.fastq       5294      4871
</span><span id="__span-8-20"><a id="__codelineno-8-20" name="__codelineno-8-20" href="#__codelineno-8-20"></a>F3D9_S197_L001_R1_001.fastq       7070      6504
</span><span id="__span-8-21"><a id="__codelineno-8-21" name="__codelineno-8-21" href="#__codelineno-8-21"></a>Mock_S280_L001_R1_001.fastq       4779      4314
</span></code></pre></div>
<hr />
<h2 id="step-2-building-an-error-model"><strong>Step 2: Building an error model</strong></h2>
<p>Evaluating the error model for the <code>DADA2</code> algorithm using direct reads.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="n">errF</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">learnErrors</span><span class="p">(</span><span class="n">filtFs</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>33514080 total bases in 139642 reads from 20 samples will be used for learning the error rates.
</span></code></pre></div>
<p>Same model for reverse reads.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="n">errR</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">learnErrors</span><span class="p">(</span><span class="n">filtRs</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>22342720 total bases in 139642 reads from 20 samples will be used for learning the error rates.
</span></code></pre></div>
<p>Error graph for all possible base transitions.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="nf">plotErrors</span><span class="p">(</span><span class="n">errF</span><span class="p">,</span><span class="w"> </span><span class="n">nominalQ</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<p><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/error_graph.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/error_graph.png" width="100%"></a></p>
<hr />
<h2 id="step-3-dereplication-and-denoising"><strong>Step 3: Dereplication and "denoising"</strong></h2>
<p>To speed up the calculations, we can remove identical sequences, taking into account their number in the subsequent analysis.<br>
For the obtained objects, we keep the same sample names.<br></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="n">derepFs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">derepFastq</span><span class="p">(</span><span class="n">filtFs</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="n">derepRs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">derepFastq</span><span class="p">(</span><span class="n">filtRs</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-14-3"><a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="c1"># Name the derep-class objects by the sample names</span>
</span><span id="__span-14-4"><a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="nf">names</span><span class="p">(</span><span class="n">derepFs</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample.names</span>
</span><span id="__span-14-5"><a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a><span class="nf">names</span><span class="p">(</span><span class="n">derepRs</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample.names</span>
</span></code></pre></div>
<p>Now we run the main algorithm to determine ASV for forward and reverse reads.<br></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="n">dadaFs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dada</span><span class="p">(</span><span class="n">derepFs</span><span class="p">,</span><span class="w"> </span><span class="n">err</span><span class="o">=</span><span class="n">errF</span><span class="p">,</span><span class="w"> </span><span class="n">pool</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="n">dadaRs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dada</span><span class="p">(</span><span class="n">derepRs</span><span class="p">,</span><span class="w"> </span><span class="n">err</span><span class="o">=</span><span class="n">errR</span><span class="p">,</span><span class="w"> </span><span class="n">pool</span><span class="o">=</span><span class="bp">T</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>20 samples were pooled: 139642 reads in 23134 unique sequences.
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>20 samples were pooled: 139642 reads in 20631 unique sequences.
</span></code></pre></div>
<hr />
<h2 id="step-4-merging-reads-summary-statistics"><strong>Step 4: Merging reads, summary statistics</strong></h2>
<p>Now we can "glue" the forward and reverse reads for each pair into one sequence.<br></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="n">mergers</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">mergePairs</span><span class="p">(</span><span class="n">dadaFs</span><span class="p">,</span><span class="w"> </span><span class="n">derepFs</span><span class="p">,</span><span class="w"> </span><span class="n">dadaRs</span><span class="p">,</span><span class="w"> </span><span class="n">derepRs</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a>
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a><span class="n">seqtab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">makeSequenceTable</span><span class="p">(</span><span class="n">mergers</span><span class="p">)</span>
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="nf">dim</span><span class="p">(</span><span class="n">seqtab</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a>[1]  20 488
</span></code></pre></div>
<hr />
<h2 id="step-5-removal-of-chimeras"><strong>Step 5: Removal of chimeras</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="n">seqtab.nochim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">removeBimeraDenovo</span><span class="p">(</span><span class="n">seqtab</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s">&quot;consensus&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-19-2"><a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="nf">dim</span><span class="p">(</span><span class="n">seqtab.nochim</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a>Identified 204 bimeras out of 488 input sequences.
</span><span id="__span-20-2"><a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a>[1]  20 284
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="nf">dim</span><span class="p">(</span><span class="n">seqtab.nochim</span><span class="p">)</span>
</span><span id="__span-21-2"><a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="nf">sum</span><span class="p">(</span><span class="n">seqtab.nochim</span><span class="p">)</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">seqtab</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a>[1]  20 284
</span><span id="__span-22-2"><a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a>[1] 0.9180065
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-23-1"><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="n">getN</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="nf">getUniques</span><span class="p">(</span><span class="n">x</span><span class="p">))</span>
</span><span id="__span-23-2"><a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="n">track</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">cbind</span><span class="p">(</span><span class="n">out</span><span class="p">,</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="n">dadaFs</span><span class="p">,</span><span class="w"> </span><span class="n">getN</span><span class="p">),</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="n">dadaRs</span><span class="p">,</span><span class="w"> </span><span class="n">getN</span><span class="p">),</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="n">mergers</span><span class="p">,</span><span class="w"> </span><span class="n">getN</span><span class="p">),</span><span class="w"> </span><span class="nf">rowSums</span><span class="p">(</span><span class="n">seqtab.nochim</span><span class="p">))</span>
</span><span id="__span-23-3"><a id="__codelineno-23-3" name="__codelineno-23-3" href="#__codelineno-23-3"></a><span class="c1"># If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)</span>
</span><span id="__span-23-4"><a id="__codelineno-23-4" name="__codelineno-23-4" href="#__codelineno-23-4"></a><span class="nf">colnames</span><span class="p">(</span><span class="n">track</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;input&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;filtered&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;denoisedF&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;denoisedR&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;merged&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;nonchim&quot;</span><span class="p">)</span>
</span><span id="__span-23-5"><a id="__codelineno-23-5" name="__codelineno-23-5" href="#__codelineno-23-5"></a><span class="nf">rownames</span><span class="p">(</span><span class="n">track</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sample.names</span>
</span><span id="__span-23-6"><a id="__codelineno-23-6" name="__codelineno-23-6" href="#__codelineno-23-6"></a><span class="nf">head</span><span class="p">(</span><span class="n">track</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-24-1"><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a>       input filtered denoisedF denoisedR merged nonchim
</span><span id="__span-24-2"><a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a>F3D0    7793     7113      7077      7086   6758    6422
</span><span id="__span-24-3"><a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a>F3D1    5869     5299      5273      5283   5135    4992
</span><span id="__span-24-4"><a id="__codelineno-24-4" name="__codelineno-24-4" href="#__codelineno-24-4"></a>F3D141  5958     5463      5427      5444   5166    4727
</span><span id="__span-24-5"><a id="__codelineno-24-5" name="__codelineno-24-5" href="#__codelineno-24-5"></a>F3D142  3183     2914      2893      2904   2725    2432
</span><span id="__span-24-6"><a id="__codelineno-24-6" name="__codelineno-24-6" href="#__codelineno-24-6"></a>F3D143  3178     2941      2922      2928   2761    2462
</span><span id="__span-24-7"><a id="__codelineno-24-7" name="__codelineno-24-7" href="#__codelineno-24-7"></a>F3D144  4827     4312      4290      4301   4020    3515
</span></code></pre></div>
<hr />
<h2 id="step-6-taxonomy-identification"><strong>Step 6: Taxonomy identification</strong></h2>
<p>To determine the taxonomic position of the found sequences, we first need to download the classifier.<br>
https://zenodo.org/record/1172783/files/silva_nr_v132_train_set.fa.gz.<br></p>
<p>Sometimes <code>R</code> does not like to download files. If it falls with error - download a gz archive manually and move it into working directory.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-25-1"><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="n">url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;https://zenodo.org/record/1172783/files/silva_nr_v132_train_set.fa.gz&quot;</span>
</span><span id="__span-25-2"><a id="__codelineno-25-2" name="__codelineno-25-2" href="#__codelineno-25-2"></a>
</span><span id="__span-25-3"><a id="__codelineno-25-3" name="__codelineno-25-3" href="#__codelineno-25-3"></a><span class="n">classifier_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;silva_nr_v132_train_set.fa.gz&quot;</span>
</span><span id="__span-25-4"><a id="__codelineno-25-4" name="__codelineno-25-4" href="#__codelineno-25-4"></a>
</span><span id="__span-25-5"><a id="__codelineno-25-5" name="__codelineno-25-5" href="#__codelineno-25-5"></a><span class="nf">download.file</span><span class="p">(</span><span class="n">url</span><span class="p">,</span><span class="w"> </span><span class="n">classifier_1</span><span class="p">)</span>
</span></code></pre></div>
<p>A separate classifier will be required to identify species.<br>
https://zenodo.org/record/1172783/files/silva_species_assignment_v132.fa.gz?download=1.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-26-1"><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a><span class="n">url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;https://zenodo.org/record/1172783/files/silva_species_assignment_v132.fa.gz?download=1&quot;</span>
</span><span id="__span-26-2"><a id="__codelineno-26-2" name="__codelineno-26-2" href="#__codelineno-26-2"></a>
</span><span id="__span-26-3"><a id="__codelineno-26-3" name="__codelineno-26-3" href="#__codelineno-26-3"></a><span class="n">classifier_2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;silva_species_assignment_v132.fa.gz&quot;</span>
</span><span id="__span-26-4"><a id="__codelineno-26-4" name="__codelineno-26-4" href="#__codelineno-26-4"></a>
</span><span id="__span-26-5"><a id="__codelineno-26-5" name="__codelineno-26-5" href="#__codelineno-26-5"></a><span class="nf">download.file</span><span class="p">(</span><span class="n">url</span><span class="p">,</span><span class="w"> </span><span class="n">classifier_2</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-27-1"><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a><span class="n">taxa</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">assignTaxonomy</span><span class="p">(</span><span class="n">seqtab.nochim</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;silva_nr_v132_train_set.fa.gz&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">multithread</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span></code></pre></div>
<p>Adding Species Information.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-28-1"><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a><span class="n">taxa</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">addSpecies</span><span class="p">(</span><span class="n">taxa</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;silva_species_assignment_v132.fa.gz&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">verbose</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">allowMultiple</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-29-1"><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a>38 out of 284 were assigned to the species level.
</span><span id="__span-29-2"><a id="__codelineno-29-2" name="__codelineno-29-2" href="#__codelineno-29-2"></a>Of which 33 had genera consistent with the input table.
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-30-1"><a id="__codelineno-30-1" name="__codelineno-30-1" href="#__codelineno-30-1"></a><span class="n">taxa.print</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">taxa</span><span class="w"> </span><span class="c1"># Removing sequence rownames for display only</span>
</span><span id="__span-30-2"><a id="__codelineno-30-2" name="__codelineno-30-2" href="#__codelineno-30-2"></a><span class="nf">rownames</span><span class="p">(</span><span class="n">taxa.print</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NULL</span>
</span><span id="__span-30-3"><a id="__codelineno-30-3" name="__codelineno-30-3" href="#__codelineno-30-3"></a><span class="nf">head</span><span class="p">(</span><span class="n">taxa.print</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong>*</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-31-1"><a id="__codelineno-31-1" name="__codelineno-31-1" href="#__codelineno-31-1"></a>     Kingdom    Phylum          Class         Order           Family          
</span><span id="__span-31-2"><a id="__codelineno-31-2" name="__codelineno-31-2" href="#__codelineno-31-2"></a>[1,] &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot; &quot;Muribaculaceae&quot;
</span><span id="__span-31-3"><a id="__codelineno-31-3" name="__codelineno-31-3" href="#__codelineno-31-3"></a>[2,] &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot; &quot;Muribaculaceae&quot;
</span><span id="__span-31-4"><a id="__codelineno-31-4" name="__codelineno-31-4" href="#__codelineno-31-4"></a>[3,] &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot; &quot;Muribaculaceae&quot;
</span><span id="__span-31-5"><a id="__codelineno-31-5" name="__codelineno-31-5" href="#__codelineno-31-5"></a>[4,] &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot; &quot;Muribaculaceae&quot;
</span><span id="__span-31-6"><a id="__codelineno-31-6" name="__codelineno-31-6" href="#__codelineno-31-6"></a>[5,] &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot; &quot;Bacteroidaceae&quot;
</span><span id="__span-31-7"><a id="__codelineno-31-7" name="__codelineno-31-7" href="#__codelineno-31-7"></a>[6,] &quot;Bacteria&quot; &quot;Bacteroidetes&quot; &quot;Bacteroidia&quot; &quot;Bacteroidales&quot; &quot;Muribaculaceae&quot;
</span><span id="__span-31-8"><a id="__codelineno-31-8" name="__codelineno-31-8" href="#__codelineno-31-8"></a>     Genus         Species                  
</span><span id="__span-31-9"><a id="__codelineno-31-9" name="__codelineno-31-9" href="#__codelineno-31-9"></a>[1,] NA            NA                       
</span><span id="__span-31-10"><a id="__codelineno-31-10" name="__codelineno-31-10" href="#__codelineno-31-10"></a>[2,] NA            NA                       
</span><span id="__span-31-11"><a id="__codelineno-31-11" name="__codelineno-31-11" href="#__codelineno-31-11"></a>[3,] NA            NA                       
</span><span id="__span-31-12"><a id="__codelineno-31-12" name="__codelineno-31-12" href="#__codelineno-31-12"></a>[4,] NA            NA                       
</span><span id="__span-31-13"><a id="__codelineno-31-13" name="__codelineno-31-13" href="#__codelineno-31-13"></a>[5,] &quot;Bacteroides&quot; &quot;acidifaciens/caecimuris&quot;
</span><span id="__span-31-14"><a id="__codelineno-31-14" name="__codelineno-31-14" href="#__codelineno-31-14"></a>[6,] NA            NA                       
</span></code></pre></div>
<hr />
<h2 id="step-7-additional-translation-into-phyloseq-format"><strong>Step 7 (additional): Translation into <code>Phyloseq</code> format</strong></h2>
<p>First, download the necessary packages.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-32-1"><a id="__codelineno-32-1" name="__codelineno-32-1" href="#__codelineno-32-1"></a><span class="n">BiocManager</span><span class="o">::</span><span class="nf">install</span><span class="p">(</span><span class="s">&quot;phyloseq&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-33-1"><a id="__codelineno-33-1" name="__codelineno-33-1" href="#__codelineno-33-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">phyloseq</span><span class="p">);</span><span class="w"> </span><span class="nf">packageVersion</span><span class="p">(</span><span class="s">&quot;phyloseq&quot;</span><span class="p">)</span>
</span><span id="__span-33-2"><a id="__codelineno-33-2" name="__codelineno-33-2" href="#__codelineno-33-2"></a><span class="nf">library</span><span class="p">(</span><span class="n">Biostrings</span><span class="p">);</span><span class="w"> </span><span class="nf">packageVersion</span><span class="p">(</span><span class="s">&quot;Biostrings&quot;</span><span class="p">)</span>
</span><span id="__span-33-3"><a id="__codelineno-33-3" name="__codelineno-33-3" href="#__codelineno-33-3"></a><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">);</span><span class="w"> </span><span class="nf">packageVersion</span><span class="p">(</span><span class="s">&quot;ggplot2&quot;</span><span class="p">)</span>
</span><span id="__span-33-4"><a id="__codelineno-33-4" name="__codelineno-33-4" href="#__codelineno-33-4"></a><span class="nf">theme_set</span><span class="p">(</span><span class="nf">theme_bw</span><span class="p">())</span>
</span></code></pre></div>
<p>Then we add metadata.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-34-1"><a id="__codelineno-34-1" name="__codelineno-34-1" href="#__codelineno-34-1"></a><span class="n">samples.out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">seqtab.nochim</span><span class="p">)</span>
</span><span id="__span-34-2"><a id="__codelineno-34-2" name="__codelineno-34-2" href="#__codelineno-34-2"></a><span class="n">subject</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">samples.out</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;D&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">`[`</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
</span><span id="__span-34-3"><a id="__codelineno-34-3" name="__codelineno-34-3" href="#__codelineno-34-3"></a><span class="n">gender</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">substr</span><span class="p">(</span><span class="n">subject</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>
</span><span id="__span-34-4"><a id="__codelineno-34-4" name="__codelineno-34-4" href="#__codelineno-34-4"></a><span class="n">subject</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">substr</span><span class="p">(</span><span class="n">subject</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">999</span><span class="p">)</span>
</span><span id="__span-34-5"><a id="__codelineno-34-5" name="__codelineno-34-5" href="#__codelineno-34-5"></a><span class="n">day</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.integer</span><span class="p">(</span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">samples.out</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;D&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">`[`</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">))</span>
</span><span id="__span-34-6"><a id="__codelineno-34-6" name="__codelineno-34-6" href="#__codelineno-34-6"></a><span class="n">samdf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">Subject</span><span class="o">=</span><span class="n">subject</span><span class="p">,</span><span class="w"> </span><span class="n">Gender</span><span class="o">=</span><span class="n">gender</span><span class="p">,</span><span class="w"> </span><span class="n">Day</span><span class="o">=</span><span class="n">day</span><span class="p">)</span>
</span><span id="__span-34-7"><a id="__codelineno-34-7" name="__codelineno-34-7" href="#__codelineno-34-7"></a><span class="n">samdf</span><span class="o">$</span><span class="n">When</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Early&quot;</span>
</span><span id="__span-34-8"><a id="__codelineno-34-8" name="__codelineno-34-8" href="#__codelineno-34-8"></a><span class="n">samdf</span><span class="o">$</span><span class="n">When</span><span class="p">[</span><span class="n">samdf</span><span class="o">$</span><span class="n">Day</span><span class="o">&gt;</span><span class="m">100</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Late&quot;</span>
</span><span id="__span-34-9"><a id="__codelineno-34-9" name="__codelineno-34-9" href="#__codelineno-34-9"></a><span class="nf">rownames</span><span class="p">(</span><span class="n">samdf</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">samples.out</span>
</span></code></pre></div>
<p>And finally, we create a <code>phyloseq</code> object.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-35-1"><a id="__codelineno-35-1" name="__codelineno-35-1" href="#__codelineno-35-1"></a><span class="n">ps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">phyloseq</span><span class="p">(</span><span class="nf">otu_table</span><span class="p">(</span><span class="n">seqtab.nochim</span><span class="p">,</span><span class="w"> </span><span class="n">taxa_are_rows</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">),</span><span class="w"> </span>
</span><span id="__span-35-2"><a id="__codelineno-35-2" name="__codelineno-35-2" href="#__codelineno-35-2"></a><span class="w">               </span><span class="nf">sample_data</span><span class="p">(</span><span class="n">samdf</span><span class="p">),</span><span class="w"> </span>
</span><span id="__span-35-3"><a id="__codelineno-35-3" name="__codelineno-35-3" href="#__codelineno-35-3"></a><span class="w">               </span><span class="nf">tax_table</span><span class="p">(</span><span class="n">taxa</span><span class="p">))</span>
</span><span id="__span-35-4"><a id="__codelineno-35-4" name="__codelineno-35-4" href="#__codelineno-35-4"></a><span class="n">ps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">prune_samples</span><span class="p">(</span><span class="nf">sample_names</span><span class="p">(</span><span class="n">ps</span><span class="p">)</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Mock&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">ps</span><span class="p">)</span><span class="w"> </span><span class="c1"># Remove mock sample</span>
</span></code></pre></div>
<p>With it, you can already do any subsequent analysis. For example, calculate alpha diversity.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-36-1"><a id="__codelineno-36-1" name="__codelineno-36-1" href="#__codelineno-36-1"></a><span class="nf">plot_richness</span><span class="p">(</span><span class="n">ps</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="s">&quot;Day&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">measures</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Shannon&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Simpson&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;When&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<p><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/alpha_div.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/alpha_div.png" width="100%"></a></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-37-1"><a id="__codelineno-37-1" name="__codelineno-37-1" href="#__codelineno-37-1"></a><span class="n">top20</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="nf">sort</span><span class="p">(</span><span class="nf">taxa_sums</span><span class="p">(</span><span class="n">ps</span><span class="p">),</span><span class="w"> </span><span class="n">decreasing</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">))[</span><span class="m">1</span><span class="o">:</span><span class="m">20</span><span class="p">]</span>
</span><span id="__span-37-2"><a id="__codelineno-37-2" name="__codelineno-37-2" href="#__codelineno-37-2"></a><span class="n">ps.top20</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">transform_sample_counts</span><span class="p">(</span><span class="n">ps</span><span class="p">,</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">OTU</span><span class="p">)</span><span class="w"> </span><span class="n">OTU</span><span class="o">/</span><span class="nf">sum</span><span class="p">(</span><span class="n">OTU</span><span class="p">))</span>
</span><span id="__span-37-3"><a id="__codelineno-37-3" name="__codelineno-37-3" href="#__codelineno-37-3"></a><span class="n">ps.top20</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">prune_taxa</span><span class="p">(</span><span class="n">top20</span><span class="p">,</span><span class="w"> </span><span class="n">ps.top20</span><span class="p">)</span>
</span><span id="__span-37-4"><a id="__codelineno-37-4" name="__codelineno-37-4" href="#__codelineno-37-4"></a><span class="nf">plot_bar</span><span class="p">(</span><span class="n">ps.top20</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="o">=</span><span class="s">&quot;Day&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="s">&quot;Family&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">When</span><span class="p">,</span><span class="w"> </span><span class="n">scales</span><span class="o">=</span><span class="s">&quot;free_x&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<p><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/abundance.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/05_16S_amplicon_analysis/05_01_Introduction/imgs/abundance.png" width="100%"></a></p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 Ilia Popov
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy", "search", "search.suggest", "search.highlight"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>