# Repository for NGS Data Analysis Guides

> Materials of the repository were prepared on the basis of materials of the educational intensive from the Bioinformatics Institute<br>
> It can be used as a helpful repository with cheat-sheets for NGS studies. Please welcome!

To recreate any of the steps of this manual please install:

```bash
conda env create -f bioinf-intensive-2023.yml
```

## Much more to be disclosured soon:
- Reference-free approach in the comparative analysis of metagenomic data
- Whole Genome and Pangenome Analyses
- 16S Amplicon Analysis

## Genomic Variation Analysis

In the [Genomic Variation Analysis folder](1%20-%20Quality%20Control) there is a detailed guide how to conduct studies on Variant Calling using `fastqc`, `trimmomatic`, `bwa`, `samtools`, `abra2`, `bcftools`, `snpEff` & `SnpSift`

## Quality Control of raw data

In the [Quality Control folder](1%20-%20Quality%20Control) there is a detailed guide how to conduct quality control of raw data using `fastqc` and `trimmomatic`.
