
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/02_Variant_Calling/">
      
      
        <link rel="prev" href="../01_Quality_Control/">
      
      
        <link rel="next" href="../03_Whole_%28pan%29genome_analyses/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.38">
    
    
      
        <title>II. Genomic Variation Analysis - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#ii-genomic-variation-analysis" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              II. Genomic Variation Analysis
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    NGS Analysis Handbook
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../01_Quality_Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Instruction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-downloading-the-data" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Downloading the data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-check-the-reads-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Check the reads quality
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Check the reads quality">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-quality-scores" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence quality scores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-content" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-trim-low-quality-bases" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Trim low-quality bases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-index-reference" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Index reference
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-5-align-to-reference" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Align to reference
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-6-sort-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Sort alignment
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-7-make-bam-index" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7: Make bam index
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-8-realign-indels" class="md-nav__link">
    <span class="md-ellipsis">
      Step 8: Realign indels
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-9-index-final-bam" class="md-nav__link">
    <span class="md-ellipsis">
      Step 9: Index final bam
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-10-call-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Step 10: Call variants
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-11-make-snpeff-database" class="md-nav__link">
    <span class="md-ellipsis">
      Step 11: Make snpEff database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-12-annotate-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Step 12: Annotate variants
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-13-filter-high-and-moderate-effect-mutations" class="md-nav__link">
    <span class="md-ellipsis">
      Step 13: Filter high and moderate effect mutations
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_Whole_%28pan%29genome_analyses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_03_MSA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_04_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_05_Root_Date/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_07_Visualization_Pro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_01_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Instruction">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-downloading-the-data" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Downloading the data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-check-the-reads-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Check the reads quality
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Check the reads quality">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-quality-scores" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence quality scores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-content" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-trim-low-quality-bases" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Trim low-quality bases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-index-reference" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Index reference
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-5-align-to-reference" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Align to reference
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-6-sort-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Sort alignment
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-7-make-bam-index" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7: Make bam index
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-8-realign-indels" class="md-nav__link">
    <span class="md-ellipsis">
      Step 8: Realign indels
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-9-index-final-bam" class="md-nav__link">
    <span class="md-ellipsis">
      Step 9: Index final bam
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-10-call-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Step 10: Call variants
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-11-make-snpeff-database" class="md-nav__link">
    <span class="md-ellipsis">
      Step 11: Make snpEff database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-12-annotate-variants" class="md-nav__link">
    <span class="md-ellipsis">
      Step 12: Annotate variants
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-13-filter-high-and-moderate-effect-mutations" class="md-nav__link">
    <span class="md-ellipsis">
      Step 13: Filter high and moderate effect mutations
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="ii-genomic-variation-analysis"><strong>II. Genomic Variation Analysis</strong></h1>
<blockquote>
<p>This chapter contains a manual on variant calling pipeline. Details are described here only on how to conduct such a study from the time the data are downloaded to the time the genetic variant annotation files are received. The biological interpretation of the results is not given here.</p>
</blockquote>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p><a class="glightbox" href="https://img.shields.io/badge/Language-Command Line Tools-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="CLT" src="https://img.shields.io/badge/Language-Command Line Tools-steelblue" /></a><br>
<a class="glightbox" href="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" /></a></p>
<p>You can run commands below in your <code>terminal</code>.<br>
But if you want to write a convenient to read laboratory journal you can use <code>Jupyter Notebook</code>.<br>
In that case write <code>!</code> in the beggining of each cell to make it understand <code>bash</code> commands.</p>
<p>To recreate any of the steps of this manual please install:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>varcall.yaml
</span></code></pre></div>
<p>And of cource do not forget to activate the envinronment!</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>conda<span class="w"> </span>activate<span class="w"> </span>varcall
</span></code></pre></div>
<hr />
<h3 id="step-1-downloading-the-data"><strong>Step 1: Downloading the data</strong></h3>
<p>First we create all the directories to store the data</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>mkdir<span class="w"> </span>data/
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>mkdir<span class="w"> </span>data/reference/
</span></code></pre></div>
<p><strong>Download reference file from NCBI in FASTA format</strong></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>efetch<span class="w"> </span>-db<span class="w"> </span>nuccore<span class="w"> </span>-id<span class="w"> </span>U00096.3<span class="w"> </span>-format<span class="w"> </span>fasta<span class="w"> </span>&gt;<span class="w"> </span>data/reference/EcoliK12MG1655.fa
</span></code></pre></div>
<p><strong>Download reference file from NCBI in GB format</strong></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>efetch<span class="w"> </span>-db<span class="w"> </span>nuccore<span class="w"> </span>-id<span class="w"> </span>U00096.3<span class="w"> </span>-format<span class="w"> </span>gb<span class="w"> </span>&gt;<span class="w"> </span>data/reference/EcoliK12MG1655.gb
</span></code></pre></div>
<p><strong>Download sample FASTQ files from SRA via <code>sra-tools</code></strong></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>fastq-dump<span class="w"> </span>-v<span class="w"> </span>--split-3<span class="w"> </span>--gzip<span class="w"> </span>SRR17909485<span class="w"> </span>-O<span class="w"> </span>data
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>Preference setting is: Prefer SRA Normalized Format files with full base quality scores if available.
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>SRR17909485 is an SRA Normalized Format file with full base quality scores.
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>Read 1251776 spots for SRR17909485
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>Written 1251776 spots for SRR17909485
</span></code></pre></div>
<hr />
<h3 id="step-2-check-the-reads-quality"><strong>Step 2: Check the reads quality</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>mkdir<span class="w"> </span>data/fastqc_results/
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>fastqc<span class="w"> </span>data/SRR17909485_*.fastq.gz<span class="w"> </span>-o<span class="w"> </span>data/fastqc_results
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<p>Results are saved to:<br>
- <code>SRR17909485_1_fastqc.html</code><br>
- <code>SRR17909485_2_fastqc.html</code><br>
Please open them in any web browser you use:<br></p>
<h4 id="per-base-sequence-quality"><strong>Per base sequence quality</strong></h4>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20quality%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20quality%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20quality%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20quality%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<h4 id="per-sequence-quality-scores"><strong>Per sequence quality scores</strong></h4>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20quality%20scores%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20quality%20scores%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20quality%20scores%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20quality%20scores%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<h4 id="per-base-sequence-content"><strong>Per base sequence content</strong></h4>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20content%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20content%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20content%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20base%20sequence%20content%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<h4 id="per-sequence-gc-content"><strong>Per sequence GC content</strong></h4>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20GC%20content%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20GC%20content%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20GC%20content%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Per%20sequence%20GC%20content%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<h4 id="adapter-content"><strong>Adapter Content</strong></h4>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Adapter%20Content%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Adapter%20Content%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Adapter%20Content%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/02_Genomic_Variation_Analysis/imgs/Adapter%20Content%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>The main problem is <code>Per base sequence quality</code>. So we will solve it using <code>trimmomatic</code>.<br>
Details on <code>trimmomatic</code> and QC in the <a href="1%20-%20Quality%20Control">Quality Control manual</a>.</p>
<hr />
<h3 id="step-3-trim-low-quality-bases"><strong>Step 3: Trim low-quality bases</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>mkdir<span class="w"> </span>data/trimmed/
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>trimmomatic<span class="w"> </span>PE<span class="w"> </span>-threads<span class="w"> </span><span class="m">2</span><span class="w"> </span>data/SRR17909485_1.fastq.gz<span class="w"> </span>data/SRR17909485_2.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="w">    </span>data/trimmed/SRR17909485_R1.trim.paired.fastq.gz<span class="w"> </span>data/trimmed/SRR17909485_R1.trim.unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a><span class="w">    </span>data/trimmed/SRR17909485_R2.trim.paired.fastq.gz<span class="w"> </span>data/trimmed/SRR17909485_R2.trim.unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="w">    </span>LEADING:22<span class="w"> </span>TRAILING:22<span class="w"> </span>SLIDINGWINDOW:6:22<span class="w"> </span>MINLEN:32
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>TrimmomaticPE: Started with arguments:
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a> -threads 2 data/SRR17909485_1.fastq.gz data/SRR17909485_2.fastq.gz data/trimmed/SRR17909485_R1.trim.paired.fastq.gz data/trimmed/SRR17909485_R1.trim.unpaired.fastq.gz data/trimmed/SRR17909485_R2.trim.paired.fastq.gz data/trimmed/SRR17909485_R2.trim.unpaired.fastq.gz LEADING:22 TRAILING:22 SLIDINGWINDOW:6:22 MINLEN:32
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>Quality encoding detected as phred33
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>Input Read Pairs: 1251776 Both Surviving: 1205173 (96.28%) Forward Only Surviving: 34920 (2.79%) Reverse Only Surviving: 2954 (0.24%) Dropped: 8729 (0.70%)
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>TrimmomaticPE: Completed successfully
</span></code></pre></div>
<ul>
<li><code>LEADING:22</code>: Removes low-quality bases from the start of the reads if they have a quality score lower than 22.</li>
<li><code>TRAILING:22</code>: Removes low-quality bases from the end of the reads if they have a quality score lower than 22.</li>
<li><code>SLIDINGWINDOW:6:22</code>: Performs a sliding window trimming. It scans the read with a 6-base wide window, cutting when the average quality per base drops below 22.</li>
<li><code>MINLEN:32</code>: Discards reads that are shorter than 32 bases after trimming.</li>
</ul>
<p>Input Read Pairs: 1251776<br>
- Both Surviving: 1205173 (96.28%)<br>
- Forward Only Surviving: 34920 (2.79%)<br>
- Reverse Only Surviving: 2954 (0.24%)<br>
- Dropped: 8729 (0.70%)<br></p>
<hr />
<h3 id="step-4-index-reference"><strong>Step 4: Index reference</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>bwa<span class="w"> </span>index<span class="w"> </span>data/reference/EcoliK12MG1655.fa
</span></code></pre></div>
<hr />
<h3 id="step-5-align-to-reference"><strong>Step 5: Align to reference</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>mkdir<span class="w"> </span>data/bam/
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">2</span><span class="w"> </span>-R<span class="w"> </span><span class="s1">&#39;@RG\tID:1&#39;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="w">    </span>data/reference/EcoliK12MG1655.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-14-3"><a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="w">    </span>data/trimmed/SRR17909485_R1.trim.paired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-14-4"><a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="w">    </span>data/trimmed/SRR17909485_R2.trim.paired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-14-5"><a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>view<span class="w"> </span>-b<span class="w"> </span>&gt;<span class="w"> </span>data/bam/EcoliK12MG1655.SRR17909485.unsorted.bam
</span></code></pre></div>
<hr />
<h3 id="step-6-sort-alignment"><strong>Step 6: Sort alignment</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a>samtools<span class="w"> </span>sort<span class="w"> </span>--threads<span class="w"> </span><span class="m">2</span><span class="w"> </span>data/bam/EcoliK12MG1655.SRR17909485.unsorted.bam<span class="w"> </span>&gt;<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="w">    </span>data/bam/EcoliK12MG1655.SRR17909485.sorted.bam
</span></code></pre></div>
<hr />
<h3 id="step-7-make-bam-index"><strong>Step 7: Make bam index</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>samtools<span class="w"> </span>index<span class="w"> </span>data/bam/EcoliK12MG1655.SRR17909485.sorted.bam
</span></code></pre></div>
<hr />
<h3 id="step-8-realign-indels"><strong>Step 8: Realign indels</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a>abra2<span class="w"> </span>--threads<span class="w"> </span><span class="m">2</span><span class="w"> </span>--mad<span class="w"> </span><span class="m">100</span><span class="w"> </span>--mbq<span class="w"> </span><span class="m">24</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a><span class="w">    </span>--ref<span class="w"> </span>data/reference/EcoliK12MG1655.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a><span class="w">    </span>--in<span class="w"> </span>data/bam/EcoliK12MG1655.SRR17909485.sorted.bam<span class="w"> </span><span class="se">\</span>
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="w">    </span>--out<span class="w"> </span>data/bam/EcoliK12MG1655.SRR17909485.final.bam
</span></code></pre></div>
<hr />
<h3 id="step-9-index-final-bam"><strong>Step 9: Index final bam</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a>samtools<span class="w"> </span>index<span class="w"> </span>data/bam/EcoliK12MG1655.SRR17909485.final.bam
</span></code></pre></div>
<hr />
<h3 id="step-10-call-variants"><strong>Step 10: Call variants</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a>mkdir<span class="w"> </span>data/vcf/
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a>bcftools<span class="w"> </span>mpileup<span class="w"> </span>-Ou<span class="w"> </span>--max-depth<span class="w"> </span><span class="m">5000</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-20-2"><a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="w">    </span>-f<span class="w"> </span>data/reference/EcoliK12MG1655.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-20-3"><a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a><span class="w">    </span>data/bam/EcoliK12MG1655.SRR17909485.final.bam<span class="w"> </span><span class="se">\</span>
</span><span id="__span-20-4"><a id="__codelineno-20-4" name="__codelineno-20-4" href="#__codelineno-20-4"></a><span class="w">    </span><span class="p">|</span><span class="w"> </span>bcftools<span class="w"> </span>call<span class="w"> </span>-mv<span class="w"> </span>--ploidy<span class="w"> </span><span class="m">1</span><span class="w"> </span>-Ov<span class="w"> </span><span class="se">\</span>
</span><span id="__span-20-5"><a id="__codelineno-20-5" name="__codelineno-20-5" href="#__codelineno-20-5"></a><span class="w">    </span>-o<span class="w"> </span>data/vcf/EcoliK12MG1655.SRR17909485.called.bcftools.vcf
</span></code></pre></div>
<p><code>Mpileup</code> is actually a slightly different representation of alignment. <code>Mpileup</code> takes very simple things, it takes the number of reads aligned to each point and for each position in the reference genome it counts statistics - how many letters match the reference, how many mismatch, how many insertions, deletions.<br></p>
<p>Unfortunately <code>bcftools</code> is not multithreaded. If we do re-alignment with <code>abra2</code>, <code>bcftools</code> will work a bit faster.<br></p>
<p><code>bcftools --ploidy 1</code> - we tell the program that we have a haploid genome.<br></p>
<p><code>vcf</code> = variant call format.
This is actually a list of variations in a clear text format: chromosome, position, which letter was replaced by which letter, the quality of how much <code>bcftools</code> considers it important, some set of certain statistics (frequency of letter occurrence, etc.).<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a>tail<span class="w"> </span>data/vcf/EcoliK12MG1655.SRR17909485.called.bcftools.vcf
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a>U00096.3    3110421 .   C   A   225.417 .   DP=41;VDB=0.454064;SGB=-0.692914;MQSBZ=1.45774;MQ0F=0;AC=1;AN=1;DP4=0,0,8,17;MQ=59  GT:PL   1:255,0
</span><span id="__span-22-2"><a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a>U00096.3    3560455 .   C   CG  225.417 .   INDEL;IDV=82;IMF=1;DP=82;VDB=0.0327919;SGB=-0.693147;MQSBZ=0;BQBZ=0.885506;MQ0F=0;AC=1;AN=1;DP4=0,0,37,45;MQ=60 GT:PL   1:255,0
</span><span id="__span-22-3"><a id="__codelineno-22-3" name="__codelineno-22-3" href="#__codelineno-22-3"></a>U00096.3    4093770 .   C   T   225.417 .   DP=141;VDB=0.961761;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,39,52;MQ=60  GT:PL   1:255,0
</span><span id="__span-22-4"><a id="__codelineno-22-4" name="__codelineno-22-4" href="#__codelineno-22-4"></a>U00096.3    4161248 .   G   T   225.417 .   DP=117;VDB=0.400301;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,36,33;MQ=60  GT:PL   1:255,0
</span><span id="__span-22-5"><a id="__codelineno-22-5" name="__codelineno-22-5" href="#__codelineno-22-5"></a>U00096.3    4164123 .   C   G   225.417 .   DP=130;VDB=0.322528;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,32,44;MQ=60  GT:PL   1:255,0
</span><span id="__span-22-6"><a id="__codelineno-22-6" name="__codelineno-22-6" href="#__codelineno-22-6"></a>U00096.3    4296380 .   AC  ACGC    228.422 .   INDEL;IDV=98;IMF=0.989899;DP=99;VDB=8.47905e-05;SGB=-0.693147;RPBZ=-1.71529;MQBZ=1.83497;MQSBZ=-2.15139;BQBZ=-1.59515;SCBZ=-9.8995;MQ0F=0;AC=1;AN=1;DP4=0,1,47,51;MQ=54 GT:PL   1:255,0
</span><span id="__span-22-7"><a id="__codelineno-22-7" name="__codelineno-22-7" href="#__codelineno-22-7"></a>U00096.3    4474834 .   A   G   225.417 .   DP=81;VDB=0.345282;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,21,26;MQ=60   GT:PL   1:255,0
</span><span id="__span-22-8"><a id="__codelineno-22-8" name="__codelineno-22-8" href="#__codelineno-22-8"></a>U00096.3    4585480 .   G   A   225.417 .   DP=70;VDB=0.996826;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,20,26;MQ=60   GT:PL   1:255,0
</span><span id="__span-22-9"><a id="__codelineno-22-9" name="__codelineno-22-9" href="#__codelineno-22-9"></a>U00096.3    4602509 .   C   T   225.417 .   DP=136;VDB=0.843259;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,42,41;MQ=60  GT:PL   1:255,0
</span><span id="__span-22-10"><a id="__codelineno-22-10" name="__codelineno-22-10" href="#__codelineno-22-10"></a>U00096.3    4616669 .   G   T   225.417 .   DP=88;VDB=0.58582;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,25,31;MQ=60    GT:PL   1:255,0
</span></code></pre></div>
<hr />
<h3 id="step-11-make-snpeff-database"><strong>Step 11: Make snpEff database</strong></h3>
<p><code>snpEff</code> has a database of already well-sequenced, annotated genomes of many known organisms. <code>snpEff</code> loads the annotations and while annotating the <code>vcf</code> file, it checks which gene each particular variation falls into which position.<br>
By <em>E. coli</em> snpEff has over 3500 databases, which is a lot.<br>
The best way to handle this is to make your own database.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-23-1"><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>data/EcoliK12MG1655/
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-24-1"><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a>cp<span class="w"> </span>data/reference/EcoliK12MG1655.gb<span class="w"> </span>data/EcoliK12MG1655/genes.gbk
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-25-1"><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;EcoliK12MG1655.genome : EcoliK12MG1655\nEcoliK12MG1655.chromosomes : EcoliK12MG1655.gb\nEcoliK12MG1655.codonTable : Standard&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-25-2"><a id="__codelineno-25-2" name="__codelineno-25-2" href="#__codelineno-25-2"></a><span class="w">    </span>&gt;<span class="w"> </span>snpEff.config
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-26-1"><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a>cat<span class="w"> </span>snpEff.config
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-27-1"><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a>EcoliK12MG1655.genome : EcoliK12MG1655
</span><span id="__span-27-2"><a id="__codelineno-27-2" name="__codelineno-27-2" href="#__codelineno-27-2"></a>EcoliK12MG1655.chromosomes : EcoliK12MG1655.gb
</span><span id="__span-27-3"><a id="__codelineno-27-3" name="__codelineno-27-3" href="#__codelineno-27-3"></a>EcoliK12MG1655.codonTable : Standard
</span></code></pre></div>
<ol>
<li>Line 1 - under what name snpEff will know this reference genome (<code>EcoliK12MG1655</code>)</li>
<li>Line 2 - the chromosomes from this genome (all contigs with all annotations) will be in the file <code>EcoliK12MG1655.gb</code></li>
<li>Line 3 - standard genetic code table</li>
</ol>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-28-1"><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a>snpEff<span class="w"> </span>build<span class="w"> </span>-c<span class="w"> </span>snpEff.config<span class="w"> </span>-genbank<span class="w"> </span>EcoliK12MG1655
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-29-1"><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a>00:00:00 Codon table &#39;Standard&#39; for genome &#39;EcoliK12MG1655&#39;
</span><span id="__span-29-2"><a id="__codelineno-29-2" name="__codelineno-29-2" href="#__codelineno-29-2"></a>    Protein check:  EcoliK12MG1655  OK: 4298    Not found: 17   Errors: 0   Error percentage: 0.0%
</span></code></pre></div>
<hr />
<h3 id="step-12-annotate-variants"><strong>Step 12: Annotate variants</strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-30-1"><a id="__codelineno-30-1" name="__codelineno-30-1" href="#__codelineno-30-1"></a>snpEff<span class="w"> </span>ann<span class="w"> </span>-v<span class="w"> </span>EcoliK12MG1655<span class="w">  </span><span class="se">\</span>
</span><span id="__span-30-2"><a id="__codelineno-30-2" name="__codelineno-30-2" href="#__codelineno-30-2"></a><span class="w">    </span>data/vcf/EcoliK12MG1655.SRR17909485.called.bcftools.vcf<span class="w"> </span>&gt;<span class="w"> </span><span class="se">\</span>
</span><span id="__span-30-3"><a id="__codelineno-30-3" name="__codelineno-30-3" href="#__codelineno-30-3"></a><span class="w">    </span>data/vcf/EcoliK12MG1655.SRR17909485.annotated.vcf
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-31-1"><a id="__codelineno-31-1" name="__codelineno-31-1" href="#__codelineno-31-1"></a>mv<span class="w"> </span>snpEff_genes.txt<span class="w"> </span>EcoliK12MG1655.SRR17909485.snpEff_genes.txt
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-32-1"><a id="__codelineno-32-1" name="__codelineno-32-1" href="#__codelineno-32-1"></a>mv<span class="w"> </span>snpEff_summary.html<span class="w"> </span>EcoliK12MG1655.SRR17909485.snpEff_summary.html
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-33-1"><a id="__codelineno-33-1" name="__codelineno-33-1" href="#__codelineno-33-1"></a>tail<span class="w"> </span>data/vcf/EcoliK12MG1655.SRR17909485.annotated.vcf
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-34-1"><a id="__codelineno-34-1" name="__codelineno-34-1" href="#__codelineno-34-1"></a>U00096.3    3110421 .   C   A   225.417 .   DP=41;VDB=0.454064;SGB=-0.692914;MQSBZ=1.45774;MQ0F=0;AC=1;AN=1;DP4=0,0,8,17;MQ=59;ANN=A|upstream_gene_variant|MODIFIER|speC|b2965|transcript|b2965|protein_coding||c.-1266G&gt;T|||||1266|,A|upstream_gene_variant|MODIFIER|yqgH|b4785|transcript|b4785|protein_coding||c.-1021G&gt;T|||||1021|,A|upstream_gene_variant|MODIFIER|yqhJ|b4786|transcript|b4786|protein_coding||c.-845C&gt;A|||||845|WARNING_TRANSCRIPT_NO_START_CODON,A|downstream_gene_variant|MODIFIER|mltC|b2963|transcript|b2963|protein_coding||c.*4909C&gt;A|||||4909|,A|downstream_gene_variant|MODIFIER|nupG|b2964|transcript|b2964|protein_coding||c.*3451C&gt;A|||||3451|,A|downstream_gene_variant|MODIFIER|yqgA|b2966|transcript|b2966|protein_coding||c.*161C&gt;A|||||161|WARNING_TRANSCRIPT_NO_START_CODON,A|downstream_gene_variant|MODIFIER|yghD|b2968|transcript|b2968|protein_coding||c.*169G&gt;T|||||169|,A|downstream_gene_variant|MODIFIER|yghE|b2969|transcript|b2969|protein_coding||c.*707G&gt;T|||||707|WARNING_TRANSCRIPT_NO_START_CODON,A|downstream_gene_variant|MODIFIER|yghF|b2970|transcript|b2970|protein_coding||c.*1633G&gt;T|||||1633|WARNING_TRANSCRIPT_NO_START_CODON,A|downstream_gene_variant|MODIFIER|yghG|b2971|transcript|b2971|protein_coding||c.*2646G&gt;T|||||2646|,A|downstream_gene_variant|MODIFIER|pppA|b2972|transcript|b2972|protein_coding||c.*3122G&gt;T|||||3122|,A|downstream_gene_variant|MODIFIER|yghJ|b4466|transcript|b4466|protein_coding||c.*4129G&gt;T|||||4129|,A|intragenic_variant|MODIFIER|pheV|b2967|gene_variant|b2967|||n.3110421C&gt;A||||||   GT:PL   1:255,0
</span><span id="__span-34-2"><a id="__codelineno-34-2" name="__codelineno-34-2" href="#__codelineno-34-2"></a>U00096.3    3560455 .   C   CG  225.417 .   INDEL;IDV=82;IMF=1;DP=82;VDB=0.0327919;SGB=-0.693147;MQSBZ=0;BQBZ=0.885506;MQ0F=0;AC=1;AN=1;DP4=0,0,37,45;MQ=60;ANN=CG|upstream_gene_variant|MODIFIER|rtcA|b4475|transcript|b4475|protein_coding||c.-3607_-3606insC|||||3607|,CG|upstream_gene_variant|MODIFIER|rtcB|b3421|transcript|b3421|protein_coding||c.-2377_-2376insC|||||2377|,CG|upstream_gene_variant|MODIFIER|glpD|b3426|transcript|b3426|protein_coding||c.-1558_-1557insG|||||1557|,CG|downstream_gene_variant|MODIFIER|malT|b3418|transcript|b3418|protein_coding||c.*4665_*4666insG|||||4666|,CG|downstream_gene_variant|MODIFIER|rtcR|b3422|transcript|b3422|protein_coding||c.*589_*590insG|||||590|,CG|downstream_gene_variant|MODIFIER|glpG|b3424|transcript|b3424|protein_coding||c.*166_*167insC|||||166|,CG|downstream_gene_variant|MODIFIER|glpE|b3425|transcript|b3425|protein_coding||c.*1041_*1042insC|||||1041|,CG|downstream_gene_variant|MODIFIER|yzgL|b3427|transcript|b3427|protein_coding||c.*3268_*3269insC|||||3268|,CG|downstream_gene_variant|MODIFIER|glgP|b3428|transcript|b3428|protein_coding||c.*3678_*3679insC|||||3678|,CG|intragenic_variant|MODIFIER|glpR|b3423|gene_variant|b3423|||n.3560455_3560456insG||||||  GT:PL   1:255,0
</span><span id="__span-34-3"><a id="__codelineno-34-3" name="__codelineno-34-3" href="#__codelineno-34-3"></a>U00096.3    4093770 .   C   T   225.417 .   DP=141;VDB=0.961761;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,39,52;MQ=60;ANN=T|missense_variant|MODERATE|rhaD|b3902|transcript|b3902|protein_coding|1/1|c.503G&gt;A|p.Gly168Asp|503/825|503/825|168/274||,T|upstream_gene_variant|MODIFIER|frvR|b3897|transcript|b3897|protein_coding||c.-3915G&gt;A|||||3915|,T|upstream_gene_variant|MODIFIER|frvX|b3898|transcript|b3898|protein_coding||c.-2845G&gt;A|||||2845|,T|upstream_gene_variant|MODIFIER|frvB|b3899|transcript|b3899|protein_coding||c.-1404G&gt;A|||||1404|,T|upstream_gene_variant|MODIFIER|frvA|b3900|transcript|b3900|protein_coding||c.-947G&gt;A|||||947|,T|upstream_gene_variant|MODIFIER|rhaM|b3901|transcript|b3901|protein_coding||c.-332G&gt;A|||||332|,T|upstream_gene_variant|MODIFIER|rhaS|b3905|transcript|b3905|protein_coding||c.-3966C&gt;T|||||3966|,T|upstream_gene_variant|MODIFIER|rhaR|b3906|transcript|b3906|protein_coding||c.-4876C&gt;T|||||4876|WARNING_TRANSCRIPT_NO_START_CODON,T|downstream_gene_variant|MODIFIER|rhaA|b3903|transcript|b3903|protein_coding||c.*953G&gt;A|||||953|,T|downstream_gene_variant|MODIFIER|rhaB|b3904|transcript|b3904|protein_coding||c.*2209G&gt;A|||||2209|   GT:PL   1:255,0
</span><span id="__span-34-4"><a id="__codelineno-34-4" name="__codelineno-34-4" href="#__codelineno-34-4"></a>U00096.3    4161248 .   G   T   225.417 .   DP=117;VDB=0.400301;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,36,33;MQ=60;ANN=T|missense_variant|MODERATE|fabR|b3963|transcript|b3963|protein_coding|1/1|c.182G&gt;T|p.Gly61Val|182/705|182/705|61/234||,T|upstream_gene_variant|MODIFIER|sthA|b3962|transcript|b3962|protein_coding||c.-458C&gt;A|||||458|,T|upstream_gene_variant|MODIFIER|yijD|b3964|transcript|b3964|protein_coding||c.-523G&gt;T|||||523|,T|upstream_gene_variant|MODIFIER|btuB|b3966|transcript|b3966|protein_coding||c.-2391G&gt;T|||||2391|,T|upstream_gene_variant|MODIFIER|murI|b3967|transcript|b3967|protein_coding||c.-4180G&gt;T|||||4180|,T|downstream_gene_variant|MODIFIER|argB|b3959|transcript|b3959|protein_coding||c.*4459G&gt;T|||||4459|,T|downstream_gene_variant|MODIFIER|argH|b3960|transcript|b3960|protein_coding||c.*3025G&gt;T|||||3025|,T|downstream_gene_variant|MODIFIER|oxyR|b3961|transcript|b3961|protein_coding||c.*1841G&gt;T|||||1841|,T|downstream_gene_variant|MODIFIER|trmA|b3965|transcript|b3965|protein_coding||c.*922C&gt;A|||||922|    GT:PL   1:255,0
</span><span id="__span-34-5"><a id="__codelineno-34-5" name="__codelineno-34-5" href="#__codelineno-34-5"></a>U00096.3    4164123 .   C   G   225.417 .   DP=130;VDB=0.322528;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,32,44;MQ=60;ANN=G|missense_variant|MODERATE|btuB|b3966|transcript|b3966|protein_coding|1/1|c.485C&gt;G|p.Ala162Gly|485/1845|485/1845|162/614||,G|upstream_gene_variant|MODIFIER|sthA|b3962|transcript|b3962|protein_coding||c.-3333G&gt;C|||||3333|,G|upstream_gene_variant|MODIFIER|trmA|b3965|transcript|b3965|protein_coding||c.-853G&gt;C|||||853|,G|upstream_gene_variant|MODIFIER|murI|b3967|transcript|b3967|protein_coding||c.-1305C&gt;G|||||1305|,G|downstream_gene_variant|MODIFIER|oxyR|b3961|transcript|b3961|protein_coding||c.*4716C&gt;G|||||4716|,G|downstream_gene_variant|MODIFIER|fabR|b3963|transcript|b3963|protein_coding||c.*2352C&gt;G|||||2352|,G|downstream_gene_variant|MODIFIER|yijD|b3964|transcript|b3964|protein_coding||c.*1993C&gt;G|||||1993|  GT:PL   1:255,0
</span><span id="__span-34-6"><a id="__codelineno-34-6" name="__codelineno-34-6" href="#__codelineno-34-6"></a>U00096.3    4296380 .   AC  ACGC    228.422 .   INDEL;IDV=98;IMF=0.989899;DP=99;VDB=8.47905e-05;SGB=-0.693147;RPBZ=-1.71529;MQBZ=1.83497;MQSBZ=-2.15139;BQBZ=-1.59515;SCBZ=-9.8995;MQ0F=0;AC=1;AN=1;DP4=0,1,47,51;MQ=54;ANN=ACGC|downstream_gene_variant|MODIFIER|nrfD|b4073|transcript|b4073|protein_coding||c.*4949_*4950insGC|||||4950|,ACGC|downstream_gene_variant|MODIFIER|nrfE|b4074|transcript|b4074|protein_coding||c.*3211_*3212insGC|||||3212|WARNING_TRANSCRIPT_NO_START_CODON,ACGC|downstream_gene_variant|MODIFIER|nrfF|b4075|transcript|b4075|protein_coding||c.*2835_*2836insGC|||||2836|,ACGC|downstream_gene_variant|MODIFIER|nrfG|b4076|transcript|b4076|protein_coding||c.*2242_*2243insGC|||||2243|,ACGC|downstream_gene_variant|MODIFIER|gltP|b4077|transcript|b4077|protein_coding||c.*587_*588insGC|||||588|,ACGC|downstream_gene_variant|MODIFIER|yjcO|b4078|transcript|b4078|protein_coding||c.*54_*55insGC|||||54|,ACGC|downstream_gene_variant|MODIFIER|fdhF|b4079|transcript|b4079|protein_coding||c.*837_*838insGC|||||837|,ACGC|downstream_gene_variant|MODIFIER|mdtP|b4080|transcript|b4080|protein_coding||c.*3182_*3183insGC|||||3182|,ACGC|downstream_gene_variant|MODIFIER|mdtO|b4081|transcript|b4081|protein_coding||c.*4645_*4646insGC|||||4645|,ACGC|intergenic_region|MODIFIER|gltP-yjcO|b4077-b4078|intergenic_region|b4077-b4078|||n.4296381_4296382insGC||||||  GT:PL   1:255,0
</span><span id="__span-34-7"><a id="__codelineno-34-7" name="__codelineno-34-7" href="#__codelineno-34-7"></a>U00096.3    4474834 .   A   G   225.417 .   DP=81;VDB=0.345282;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,21,26;MQ=60;ANN=G|upstream_gene_variant|MODIFIER|ridA|b4243|transcript|b4243|protein_coding||c.-3921T&gt;C|||||3921|,G|upstream_gene_variant|MODIFIER|pyrI|b4244|transcript|b4244|protein_coding||c.-3387T&gt;C|||||3387|,G|upstream_gene_variant|MODIFIER|pyrB|b4245|transcript|b4245|protein_coding||c.-2439T&gt;C|||||2439|,G|upstream_gene_variant|MODIFIER|pyrL|b4246|transcript|b4246|protein_coding||c.-2301T&gt;C|||||2301|,G|upstream_gene_variant|MODIFIER|yjgH|b4248|transcript|b4248|protein_coding||c.-1625T&gt;C|||||1625|,G|upstream_gene_variant|MODIFIER|bdcA|b4249|transcript|b4249|protein_coding||c.-781T&gt;C|||||781|,G|upstream_gene_variant|MODIFIER|tabA|b4252|transcript|b4252|protein_coding||c.-28A&gt;G|||||28|,G|upstream_gene_variant|MODIFIER|yjgL|b4253|transcript|b4253|protein_coding||c.-603A&gt;G|||||603|,G|upstream_gene_variant|MODIFIER|rraB|b4255|transcript|b4255|protein_coding||c.-3639A&gt;G|||||3639|,G|upstream_gene_variant|MODIFIER|yjgN|b4257|transcript|b4257|protein_coding||c.-4896A&gt;G|||||4896|,G|downstream_gene_variant|MODIFIER|mgtA|b4242|transcript|b4242|protein_coding||c.*4513A&gt;G|||||4513|,G|downstream_gene_variant|MODIFIER|bdcR|b4251|transcript|b4251|protein_coding||c.*117A&gt;G|||||117|,G|downstream_gene_variant|MODIFIER|argI|b4254|transcript|b4254|protein_coding||c.*2473T&gt;C|||||2473|,G|downstream_gene_variant|MODIFIER|yjgM|b4256|transcript|b4256|protein_coding||c.*4200T&gt;C|||||4200|,G|intergenic_region|MODIFIER|bdcR-tabA|b4251-b4252|intergenic_region|b4251-b4252|||n.4474834A&gt;G||||||   GT:PL   1:255,0
</span><span id="__span-34-8"><a id="__codelineno-34-8" name="__codelineno-34-8" href="#__codelineno-34-8"></a>U00096.3    4585480 .   G   A   225.417 .   DP=70;VDB=0.996826;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,20,26;MQ=60;ANN=A|stop_gained|HIGH|hsdR|b4350|transcript|b4350|protein_coding|1/1|c.1282C&gt;T|p.Gln428*|1282/3513|1282/3513|428/1170||,A|upstream_gene_variant|MODIFIER|hsdS|b4348|transcript|b4348|protein_coding||c.-4018C&gt;T|||||4018|,A|upstream_gene_variant|MODIFIER|hsdM|b4349|transcript|b4349|protein_coding||c.-2432C&gt;T|||||2432|,A|upstream_gene_variant|MODIFIER|mrr|b4351|transcript|b4351|protein_coding||c.-1469G&gt;A|||||1469|,A|downstream_gene_variant|MODIFIER|yjiA|b4352|transcript|b4352|protein_coding||c.*2429C&gt;T|||||2429|,A|downstream_gene_variant|MODIFIER|yjiX|b4353|transcript|b4353|protein_coding||c.*3396C&gt;T|||||3396|,A|downstream_gene_variant|MODIFIER|btsT|b4354|transcript|b4354|protein_coding||c.*3649C&gt;T|||||3649| GT:PL   1:255,0
</span><span id="__span-34-9"><a id="__codelineno-34-9" name="__codelineno-34-9" href="#__codelineno-34-9"></a>U00096.3    4602509 .   C   T   225.417 .   DP=136;VDB=0.843259;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,42,41;MQ=60;ANN=T|stop_gained|HIGH|yjjP|b4364|transcript|b4364|protein_coding|1/1|c.350G&gt;A|p.Trp117*|350/771|350/771|117/256||,T|upstream_gene_variant|MODIFIER|opgB|b4359|transcript|b4359|protein_coding||c.-3068G&gt;A|||||3068|WARNING_TRANSCRIPT_NO_START_CODON,T|upstream_gene_variant|MODIFIER|yjjA|b4360|transcript|b4360|protein_coding||c.-2320G&gt;A|||||2320|,T|upstream_gene_variant|MODIFIER|dnaC|b4361|transcript|b4361|protein_coding||c.-1534G&gt;A|||||1534|,T|upstream_gene_variant|MODIFIER|dnaT|b4362|transcript|b4362|protein_coding||c.-992G&gt;A|||||992|,T|upstream_gene_variant|MODIFIER|yjjB|b4363|transcript|b4363|protein_coding||c.-412G&gt;A|||||412|,T|upstream_gene_variant|MODIFIER|yjjQ|b4365|transcript|b4365|protein_coding||c.-968C&gt;T|||||968|,T|upstream_gene_variant|MODIFIER|bglJ|b4366|transcript|b4366|protein_coding||c.-1651C&gt;T|||||1651|,T|upstream_gene_variant|MODIFIER|yjjZ|b4567|transcript|b4567|protein_coding||c.-3295C&gt;T|||||3295|,T|downstream_gene_variant|MODIFIER|fhuF|b4367|transcript|b4367|protein_coding||c.*2366G&gt;A|||||2366|,T|downstream_gene_variant|MODIFIER|rsmC|b4371|transcript|b4371|protein_coding||c.*4160G&gt;A|||||4160|    GT:PL   1:255,0
</span><span id="__span-34-10"><a id="__codelineno-34-10" name="__codelineno-34-10" href="#__codelineno-34-10"></a>U00096.3    4616669 .   G   T   225.417 .   DP=88;VDB=0.58582;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,25,31;MQ=60;ANN=T|missense_variant|MODERATE|yjjI|b4380|transcript|b4380|protein_coding|1/1|c.397C&gt;A|p.Leu133Ile|397/1551|397/1551|133/516||,T|upstream_gene_variant|MODIFIER|yjjW|b4379|transcript|b4379|protein_coding||c.-1126C&gt;A|||||1126|,T|upstream_gene_variant|MODIFIER|deoC|b4381|transcript|b4381|protein_coding||c.-654G&gt;T|||||654|,T|upstream_gene_variant|MODIFIER|deoA|b4382|transcript|b4382|protein_coding||c.-1560G&gt;T|||||1560|WARNING_TRANSCRIPT_NO_START_CODON,T|upstream_gene_variant|MODIFIER|deoB|b4383|transcript|b4383|protein_coding||c.-2934G&gt;T|||||2934|,T|upstream_gene_variant|MODIFIER|deoD|b4384|transcript|b4384|protein_coding||c.-4214G&gt;T|||||4214|,T|downstream_gene_variant|MODIFIER|osmY|b4376|transcript|b4376|protein_coding||c.*4668G&gt;T|||||4668|,T|downstream_gene_variant|MODIFIER|ytjA|b4568|transcript|b4568|protein_coding||c.*4380G&gt;T|||||4380|,T|downstream_gene_variant|MODIFIER|yjjU|b4377|transcript|b4377|protein_coding||c.*3185G&gt;T|||||3185|WARNING_TRANSCRIPT_NO_START_CODON,T|downstream_gene_variant|MODIFIER|yjjV|b4378|transcript|b4378|protein_coding||c.*2409G&gt;T|||||2409|WARNING_TRANSCRIPT_NO_START_CODON GT:PL   1:255,0
</span></code></pre></div>
<p>The <code>vcf</code> file became significantly larger because the <code>ANN</code> (annotation) line appeared after all the technical information<br>
Example:<br></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-35-1"><a id="__codelineno-35-1" name="__codelineno-35-1" href="#__codelineno-35-1"></a>ANN=T|missense_variant|MODERATE|yjjI|b4380|transcript|b4380|protein_coding|1/1|c.397C&gt;A|p.Leu133Ile.
</span></code></pre></div>
<ul>
<li>There's been a substitution of a letter for the <code>T</code>.</li>
<li>This is a <code>missence variant</code> mutation.</li>
<li>Its effect is <code>MODERATE</code>.</li>
<li><code>yjjI</code> - gene name</li>
<li><code>b4380</code> - number of the transcript in which the substitution occurred.</li>
<li><code>protein_coding</code> - this is the protein coding site</li>
<li><code>c.397C&gt;A</code> - replacement in nucleotide sequence</li>
<li><code>p.Leu133Ile</code> - subsequent substitution in protein sequence (133 leucine is replaced by isoleucine)</li>
</ul>
<hr />
<h3 id="step-13-filter-high-and-moderate-effect-mutations"><strong>Step 13: Filter high and moderate effect mutations</strong></h3>
<p>Let's filter out high and moderate effect mutations:<br>
- nonsense<br>
- missense<br>
- frameshifts<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-36-1"><a id="__codelineno-36-1" name="__codelineno-36-1" href="#__codelineno-36-1"></a>SnpSift<span class="w"> </span>filter<span class="w"> </span><span class="s2">&quot;(ANN[*].IMPACT has &#39;HIGH&#39;) | (ANN[*].IMPACT has &#39;MODERATE&#39;)&quot;</span><span class="w"> </span>data/vcf/EcoliK12MG1655.SRR17909485.annotated.vcf<span class="w"> </span>&gt;<span class="w"> </span>data/vcf/EcoliK12MG1655.SRR17909485.higheffect.vcf
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-37-1"><a id="__codelineno-37-1" name="__codelineno-37-1" href="#__codelineno-37-1"></a>tail<span class="w"> </span>data/vcf/EcoliK12MG1655.SRR17909485.higheffect.vcf
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-38-1"><a id="__codelineno-38-1" name="__codelineno-38-1" href="#__codelineno-38-1"></a>U00096.3    705013  .   T   C   225.417 .   DP=103;VDB=0.0387749;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,30,36;MQ=60;ANN=C|missense_variant|MODERATE|nagE|b0679|transcript|b0679|protein_coding|1/1|c.1070T&gt;C|p.Leu357Ser|1070/1947|1070/1947|357/648||,C|upstream_gene_variant|MODIFIER|umpH|b0675|transcript|b0675|protein_coding||c.-4687A&gt;G|||||4687|,C|upstream_gene_variant|MODIFIER|nagC|b0676|transcript|b0676|protein_coding||c.-3419A&gt;G|||||3419|,C|upstream_gene_variant|MODIFIER|nagA|b0677|transcript|b0677|protein_coding||c.-2262A&gt;G|||||2262|,C|upstream_gene_variant|MODIFIER|nagB|b0678|transcript|b0678|protein_coding||c.-1402A&gt;G|||||1402|,C|upstream_gene_variant|MODIFIER|glnS|b0680|transcript|b0680|protein_coding||c.-1080T&gt;C|||||1080|,C|upstream_gene_variant|MODIFIER|chiP|b0681|transcript|b0681|protein_coding||c.-3321T&gt;C|||||3321|,C|upstream_gene_variant|MODIFIER|chiQ|b0682|transcript|b0682|protein_coding||c.-4777T&gt;C|||||4777|    GT:PL   1:255,0
</span><span id="__span-38-2"><a id="__codelineno-38-2" name="__codelineno-38-2" href="#__codelineno-38-2"></a>U00096.3    1337394 .   A   G   225.417 .   DP=65;VDB=0.915279;SGB=-0.693146;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,24,19;MQ=60;ANN=G|missense_variant|MODERATE|acnA|b1276|transcript|b1276|protein_coding|1/1|c.1564A&gt;G|p.Ser522Gly|1564/2676|1564/2676|522/891||,G|upstream_gene_variant|MODIFIER|pgpB|b1278|transcript|b1278|protein_coding||c.-1936A&gt;G|||||1936|,G|upstream_gene_variant|MODIFIER|lapA|b1279|transcript|b1279|protein_coding||c.-2849A&gt;G|||||2849|WARNING_TRANSCRIPT_NO_START_CODON,G|upstream_gene_variant|MODIFIER|lapB|b1280|transcript|b1280|protein_coding||c.-3164A&gt;G|||||3164|,G|upstream_gene_variant|MODIFIER|pyrF|b1281|transcript|b1281|protein_coding||c.-4527A&gt;G|||||4527|,G|downstream_gene_variant|MODIFIER|topA|b1274|transcript|b1274|protein_coding||c.*3749A&gt;G|||||3749|,G|downstream_gene_variant|MODIFIER|cysB|b1275|transcript|b1275|protein_coding||c.*2565A&gt;G|||||2565|,G|downstream_gene_variant|MODIFIER|ymiA|b4522|transcript|b4522|protein_coding||c.*2106A&gt;G|||||2106|,G|downstream_gene_variant|MODIFIER|yciX|b4523|transcript|b4523|protein_coding||c.*1936A&gt;G|||||1936|,G|downstream_gene_variant|MODIFIER|ymiC|b4741|transcript|b4741|protein_coding||c.*1727A&gt;G|||||1727|,G|downstream_gene_variant|MODIFIER|ribA|b1277|transcript|b1277|protein_coding||c.*1176T&gt;C|||||1176|   GT:PL   1:255,0
</span><span id="__span-38-3"><a id="__codelineno-38-3" name="__codelineno-38-3" href="#__codelineno-38-3"></a>U00096.3    1514951 .   G   T   225.417 .   DP=68;VDB=0.981287;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,18,30;MQ=60;ANN=T|missense_variant|MODERATE|ydcV|b1443|transcript|b1443|protein_coding|1/1|c.190G&gt;T|p.Ala64Ser|190/795|190/795|64/264||,T|upstream_gene_variant|MODIFIER|patD|b1444|transcript|b1444|protein_coding||c.-627G&gt;T|||||627|,T|upstream_gene_variant|MODIFIER|ortT|b1445|transcript|b1445|protein_coding||c.-2438G&gt;T|||||2438|,T|upstream_gene_variant|MODIFIER|ydcY|b1446|transcript|b1446|protein_coding||c.-2697G&gt;T|||||2697|,T|upstream_gene_variant|MODIFIER|curA|b1449|transcript|b1449|protein_coding||c.-4076G&gt;T|||||4076|,T|downstream_gene_variant|MODIFIER|ydcR|b1439|transcript|b1439|protein_coding||c.*3542G&gt;T|||||3542|,T|downstream_gene_variant|MODIFIER|ydcS|b1440|transcript|b1440|protein_coding||c.*2152G&gt;T|||||2152|,T|downstream_gene_variant|MODIFIER|ydcT|b1441|transcript|b1441|protein_coding||c.*1121G&gt;T|||||1121|,T|downstream_gene_variant|MODIFIER|ydcU|b1442|transcript|b1442|protein_coding||c.*179G&gt;T|||||179|,T|downstream_gene_variant|MODIFIER|yncL|b4598|transcript|b4598|protein_coding||c.*2148C&gt;A|||||2148|,T|downstream_gene_variant|MODIFIER|ydcZ|b1447|transcript|b1447|protein_coding||c.*2931C&gt;A|||||2931|,T|downstream_gene_variant|MODIFIER|mnaT|b1448|transcript|b1448|protein_coding||c.*3377C&gt;A|||||3377|   GT:PL   1:255,0
</span><span id="__span-38-4"><a id="__codelineno-38-4" name="__codelineno-38-4" href="#__codelineno-38-4"></a>U00096.3    2405959 .   G   T   225.417 .   DP=116;VDB=0.00680486;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,42,33;MQ=60;ANN=T|missense_variant|MODERATE|lrhA|b2289|transcript|b2289|protein_coding|1/1|c.683C&gt;A|p.Ala228Glu|683/939|683/939|228/312||,T|upstream_gene_variant|MODIFIER|nuoF|b2284|transcript|b2284|protein_coding||c.-4404C&gt;A|||||4404|,T|upstream_gene_variant|MODIFIER|nuoE|b2285|transcript|b2285|protein_coding||c.-3907C&gt;A|||||3907|,T|upstream_gene_variant|MODIFIER|nuoC|b2286|transcript|b2286|protein_coding||c.-2114C&gt;A|||||2114|,T|upstream_gene_variant|MODIFIER|nuoB|b2287|transcript|b2287|protein_coding||c.-1346C&gt;A|||||1346|,T|upstream_gene_variant|MODIFIER|nuoA|b2288|transcript|b2288|protein_coding||c.-887C&gt;A|||||887|,T|upstream_gene_variant|MODIFIER|alaA|b2290|transcript|b2290|protein_coding||c.-1602G&gt;T|||||1602|,T|upstream_gene_variant|MODIFIER|yfbR|b2291|transcript|b2291|protein_coding||c.-2903G&gt;T|||||2903|,T|downstream_gene_variant|MODIFIER|yfbS|b2292|transcript|b2292|protein_coding||c.*3561C&gt;A|||||3561|WARNING_TRANSCRIPT_NO_START_CODON GT:PL   1:255,0
</span><span id="__span-38-5"><a id="__codelineno-38-5" name="__codelineno-38-5" href="#__codelineno-38-5"></a>U00096.3    4093770 .   C   T   225.417 .   DP=141;VDB=0.961761;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,39,52;MQ=60;ANN=T|missense_variant|MODERATE|rhaD|b3902|transcript|b3902|protein_coding|1/1|c.503G&gt;A|p.Gly168Asp|503/825|503/825|168/274||,T|upstream_gene_variant|MODIFIER|frvR|b3897|transcript|b3897|protein_coding||c.-3915G&gt;A|||||3915|,T|upstream_gene_variant|MODIFIER|frvX|b3898|transcript|b3898|protein_coding||c.-2845G&gt;A|||||2845|,T|upstream_gene_variant|MODIFIER|frvB|b3899|transcript|b3899|protein_coding||c.-1404G&gt;A|||||1404|,T|upstream_gene_variant|MODIFIER|frvA|b3900|transcript|b3900|protein_coding||c.-947G&gt;A|||||947|,T|upstream_gene_variant|MODIFIER|rhaM|b3901|transcript|b3901|protein_coding||c.-332G&gt;A|||||332|,T|upstream_gene_variant|MODIFIER|rhaS|b3905|transcript|b3905|protein_coding||c.-3966C&gt;T|||||3966|,T|upstream_gene_variant|MODIFIER|rhaR|b3906|transcript|b3906|protein_coding||c.-4876C&gt;T|||||4876|WARNING_TRANSCRIPT_NO_START_CODON,T|downstream_gene_variant|MODIFIER|rhaA|b3903|transcript|b3903|protein_coding||c.*953G&gt;A|||||953|,T|downstream_gene_variant|MODIFIER|rhaB|b3904|transcript|b3904|protein_coding||c.*2209G&gt;A|||||2209|   GT:PL   1:255,0
</span><span id="__span-38-6"><a id="__codelineno-38-6" name="__codelineno-38-6" href="#__codelineno-38-6"></a>U00096.3    4161248 .   G   T   225.417 .   DP=117;VDB=0.400301;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,36,33;MQ=60;ANN=T|missense_variant|MODERATE|fabR|b3963|transcript|b3963|protein_coding|1/1|c.182G&gt;T|p.Gly61Val|182/705|182/705|61/234||,T|upstream_gene_variant|MODIFIER|sthA|b3962|transcript|b3962|protein_coding||c.-458C&gt;A|||||458|,T|upstream_gene_variant|MODIFIER|yijD|b3964|transcript|b3964|protein_coding||c.-523G&gt;T|||||523|,T|upstream_gene_variant|MODIFIER|btuB|b3966|transcript|b3966|protein_coding||c.-2391G&gt;T|||||2391|,T|upstream_gene_variant|MODIFIER|murI|b3967|transcript|b3967|protein_coding||c.-4180G&gt;T|||||4180|,T|downstream_gene_variant|MODIFIER|argB|b3959|transcript|b3959|protein_coding||c.*4459G&gt;T|||||4459|,T|downstream_gene_variant|MODIFIER|argH|b3960|transcript|b3960|protein_coding||c.*3025G&gt;T|||||3025|,T|downstream_gene_variant|MODIFIER|oxyR|b3961|transcript|b3961|protein_coding||c.*1841G&gt;T|||||1841|,T|downstream_gene_variant|MODIFIER|trmA|b3965|transcript|b3965|protein_coding||c.*922C&gt;A|||||922|    GT:PL   1:255,0
</span><span id="__span-38-7"><a id="__codelineno-38-7" name="__codelineno-38-7" href="#__codelineno-38-7"></a>U00096.3    4164123 .   C   G   225.417 .   DP=130;VDB=0.322528;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,32,44;MQ=60;ANN=G|missense_variant|MODERATE|btuB|b3966|transcript|b3966|protein_coding|1/1|c.485C&gt;G|p.Ala162Gly|485/1845|485/1845|162/614||,G|upstream_gene_variant|MODIFIER|sthA|b3962|transcript|b3962|protein_coding||c.-3333G&gt;C|||||3333|,G|upstream_gene_variant|MODIFIER|trmA|b3965|transcript|b3965|protein_coding||c.-853G&gt;C|||||853|,G|upstream_gene_variant|MODIFIER|murI|b3967|transcript|b3967|protein_coding||c.-1305C&gt;G|||||1305|,G|downstream_gene_variant|MODIFIER|oxyR|b3961|transcript|b3961|protein_coding||c.*4716C&gt;G|||||4716|,G|downstream_gene_variant|MODIFIER|fabR|b3963|transcript|b3963|protein_coding||c.*2352C&gt;G|||||2352|,G|downstream_gene_variant|MODIFIER|yijD|b3964|transcript|b3964|protein_coding||c.*1993C&gt;G|||||1993|  GT:PL   1:255,0
</span><span id="__span-38-8"><a id="__codelineno-38-8" name="__codelineno-38-8" href="#__codelineno-38-8"></a>U00096.3    4585480 .   G   A   225.417 .   DP=70;VDB=0.996826;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,20,26;MQ=60;ANN=A|stop_gained|HIGH|hsdR|b4350|transcript|b4350|protein_coding|1/1|c.1282C&gt;T|p.Gln428*|1282/3513|1282/3513|428/1170||,A|upstream_gene_variant|MODIFIER|hsdS|b4348|transcript|b4348|protein_coding||c.-4018C&gt;T|||||4018|,A|upstream_gene_variant|MODIFIER|hsdM|b4349|transcript|b4349|protein_coding||c.-2432C&gt;T|||||2432|,A|upstream_gene_variant|MODIFIER|mrr|b4351|transcript|b4351|protein_coding||c.-1469G&gt;A|||||1469|,A|downstream_gene_variant|MODIFIER|yjiA|b4352|transcript|b4352|protein_coding||c.*2429C&gt;T|||||2429|,A|downstream_gene_variant|MODIFIER|yjiX|b4353|transcript|b4353|protein_coding||c.*3396C&gt;T|||||3396|,A|downstream_gene_variant|MODIFIER|btsT|b4354|transcript|b4354|protein_coding||c.*3649C&gt;T|||||3649| GT:PL   1:255,0
</span><span id="__span-38-9"><a id="__codelineno-38-9" name="__codelineno-38-9" href="#__codelineno-38-9"></a>U00096.3    4602509 .   C   T   225.417 .   DP=136;VDB=0.843259;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,42,41;MQ=60;ANN=T|stop_gained|HIGH|yjjP|b4364|transcript|b4364|protein_coding|1/1|c.350G&gt;A|p.Trp117*|350/771|350/771|117/256||,T|upstream_gene_variant|MODIFIER|opgB|b4359|transcript|b4359|protein_coding||c.-3068G&gt;A|||||3068|WARNING_TRANSCRIPT_NO_START_CODON,T|upstream_gene_variant|MODIFIER|yjjA|b4360|transcript|b4360|protein_coding||c.-2320G&gt;A|||||2320|,T|upstream_gene_variant|MODIFIER|dnaC|b4361|transcript|b4361|protein_coding||c.-1534G&gt;A|||||1534|,T|upstream_gene_variant|MODIFIER|dnaT|b4362|transcript|b4362|protein_coding||c.-992G&gt;A|||||992|,T|upstream_gene_variant|MODIFIER|yjjB|b4363|transcript|b4363|protein_coding||c.-412G&gt;A|||||412|,T|upstream_gene_variant|MODIFIER|yjjQ|b4365|transcript|b4365|protein_coding||c.-968C&gt;T|||||968|,T|upstream_gene_variant|MODIFIER|bglJ|b4366|transcript|b4366|protein_coding||c.-1651C&gt;T|||||1651|,T|upstream_gene_variant|MODIFIER|yjjZ|b4567|transcript|b4567|protein_coding||c.-3295C&gt;T|||||3295|,T|downstream_gene_variant|MODIFIER|fhuF|b4367|transcript|b4367|protein_coding||c.*2366G&gt;A|||||2366|,T|downstream_gene_variant|MODIFIER|rsmC|b4371|transcript|b4371|protein_coding||c.*4160G&gt;A|||||4160|    GT:PL   1:255,0
</span><span id="__span-38-10"><a id="__codelineno-38-10" name="__codelineno-38-10" href="#__codelineno-38-10"></a>U00096.3    4616669 .   G   T   225.417 .   DP=88;VDB=0.58582;SGB=-0.693147;MQSBZ=0;MQ0F=0;AC=1;AN=1;DP4=0,0,25,31;MQ=60;ANN=T|missense_variant|MODERATE|yjjI|b4380|transcript|b4380|protein_coding|1/1|c.397C&gt;A|p.Leu133Ile|397/1551|397/1551|133/516||,T|upstream_gene_variant|MODIFIER|yjjW|b4379|transcript|b4379|protein_coding||c.-1126C&gt;A|||||1126|,T|upstream_gene_variant|MODIFIER|deoC|b4381|transcript|b4381|protein_coding||c.-654G&gt;T|||||654|,T|upstream_gene_variant|MODIFIER|deoA|b4382|transcript|b4382|protein_coding||c.-1560G&gt;T|||||1560|WARNING_TRANSCRIPT_NO_START_CODON,T|upstream_gene_variant|MODIFIER|deoB|b4383|transcript|b4383|protein_coding||c.-2934G&gt;T|||||2934|,T|upstream_gene_variant|MODIFIER|deoD|b4384|transcript|b4384|protein_coding||c.-4214G&gt;T|||||4214|,T|downstream_gene_variant|MODIFIER|osmY|b4376|transcript|b4376|protein_coding||c.*4668G&gt;T|||||4668|,T|downstream_gene_variant|MODIFIER|ytjA|b4568|transcript|b4568|protein_coding||c.*4380G&gt;T|||||4380|,T|downstream_gene_variant|MODIFIER|yjjU|b4377|transcript|b4377|protein_coding||c.*3185G&gt;T|||||3185|WARNING_TRANSCRIPT_NO_START_CODON,T|downstream_gene_variant|MODIFIER|yjjV|b4378|transcript|b4378|protein_coding||c.*2409G&gt;T|||||2409|WARNING_TRANSCRIPT_NO_START_CODON GT:PL   1:255,0
</span></code></pre></div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 Ilia Popov
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.expand", "content.code.copy", "search", "search.suggest", "search.highlight"], "search": "../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>