
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/IV.%20Phylogenetics/04_07_Visualization_Pro/">
      
      
        <link rel="prev" href="../04_06_Phylogenomics/">
      
      
        <link rel="next" href="../04_08_Tanglegram/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.38">
    
    
      
        <title>Visualization Pro - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#visualization-pro" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Visualization Pro
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    NGS Analysis Handbook
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome to MkDocs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../00_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../01_Quality_Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../02_Variant_Calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../03_Whole_%28pan%29genome_analyses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_03_MSA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_04_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_05_Root_Date/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-1-making-the-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Part 1: Making the tree
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Part 1: Making the tree">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-multiple-sequences-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Multiple Sequences Alignment
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-selecting-a-model-in-modelfinder-iq-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Selecting a model in ModelFinder (IQ-TREE)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-build-an-ml-tree-in-iq-tree-using-the-selected-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Build an ML-tree in IQ-TREE using the selected model
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-examine-metadata" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Examine metadata
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-2-visualizing-the-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Part 2: Visualizing the tree
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Part 2: Visualizing the tree">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-preparation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Preparation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-initial-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Initial visualization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-adding-host-information-to-the-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Adding host information to the tree
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-use-the-circular-layout-and-add-bootstrap-information" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Use the circular layout and add bootstrap information
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-5-add-the-last-metadata-info-as-the-heatmaps-and-make-the-final-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Add the last metadata info as the heatmaps and make the final visualization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_01_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-1-making-the-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Part 1: Making the tree
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Part 1: Making the tree">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-multiple-sequences-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Multiple Sequences Alignment
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-selecting-a-model-in-modelfinder-iq-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Selecting a model in ModelFinder (IQ-TREE)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-build-an-ml-tree-in-iq-tree-using-the-selected-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Build an ML-tree in IQ-TREE using the selected model
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-examine-metadata" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Examine metadata
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-2-visualizing-the-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Part 2: Visualizing the tree
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Part 2: Visualizing the tree">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-preparation" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Preparation
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-initial-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Initial visualization
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-adding-host-information-to-the-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Adding host information to the tree
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-use-the-circular-layout-and-add-bootstrap-information" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Use the circular layout and add bootstrap information
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-5-add-the-last-metadata-info-as-the-heatmaps-and-make-the-final-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Add the last metadata info as the heatmaps and make the final visualization
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="visualization-pro"><strong>Visualization Pro</strong></h1>
<blockquote>
<p>For this work, we will use sequences and metadata from the viral hemorrhagic septicemia virus (VHSV). This virus is a fish novirhabdovirus (negative stranded RNA virus) with an unusually broad host spectra: it has been isolated from more than 80 fish species in locations around the Northern hemisphere.<br></p>
<p>This guide was inspired by: https://github.com/acarafat/tutorials. Thanks for the data! Yet there are some new adjustments!</p>
<p>For more guides on pro visualization please visit <a href="https://bioconnector.github.io/workshops/r-ggtree.html">bioconnector's <code>ggtree</code> workshop</a></p>
</blockquote>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p>This guide is about making the 1st class publication ready quality tree visualization.<br>
It is divided into 2 parts:<br></p>
<ol>
<li><strong>Making the tree</strong><br>
  <a class="glightbox" href="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="CLT" src="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" /></a><br>
  <a class="glightbox" href="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" /></a></li>
<li><strong>Visualizing the tree</strong><br>
  <a class="glightbox" href="https://img.shields.io/badge/Language-R-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="CLT" src="https://img.shields.io/badge/Language-R-steelblue" /></a><br>
  <a class="glightbox" href="https://img.shields.io/badge/Recommended IDE-RStudio-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-RStudio-steelblue" /></a></li>
</ol>
<hr />
<h2 id="part-1-making-the-tree"><strong>Part 1: Making the tree</strong></h2>
<h3 id="step-1-multiple-sequences-alignment"><strong>Step 1: Multiple Sequences Alignment</strong></h3>
<p>Run multiple sequences alignment with <code>mafft</code>.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>mafft<span class="w"> </span>data/vhsv.fasta<span class="w"> </span>&gt;<span class="w"> </span>data/vhsv_mafft.fa
</span></code></pre></div>
<hr />
<h3 id="step-2-selecting-a-model-in-modelfinder-iq-tree"><strong>Step 2: Selecting a model in ModelFinder (IQ-TREE)</strong></h3>
<p>Create a directory for <code>ModelFinder</code> output.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>mkdir<span class="w"> </span>data/modelfinder
</span></code></pre></div>
<p>Run <code>ModelFinder</code>.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>iqtree2<span class="w"> </span>-m<span class="w"> </span>MFP<span class="w"> </span>-s<span class="w"> </span>data/vhsv_mafft.fa<span class="w"> </span>--prefix<span class="w"> </span>data/modelfinder/vhsv_MF2
</span></code></pre></div>
<p>Examine the best model of evolution that is the most suitable for our alignment.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>head<span class="w"> </span>-42<span class="w"> </span>data/modelfinder/vhsv_MF2.iqtree<span class="w"> </span><span class="p">|</span><span class="w"> </span>tail<span class="w"> </span>-6
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>Best-fit model according to BIC: TVMe+I+R2
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>List of models sorted by BIC scores: 
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>Model                  LogL         AIC      w-AIC        AICc     w-AICc         BIC      w-BIC
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>TVMe+I+R2         -6984.748   14221.497 -  0.00151   14244.406 -   0.0031   14892.963 +    0.719
</span></code></pre></div>
<p>In total we see that the model <code>TVMe+I+R2</code> is recognised as the best!<br></p>
<hr />
<h3 id="step-3-build-an-ml-tree-in-iq-tree-using-the-selected-model"><strong>Step 3: Build an ML-tree in IQ-TREE using the selected model</strong></h3>
<p>Create a directory for <code>IQ-TREE</code> output.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>mkdir<span class="w"> </span>data/iqtree
</span></code></pre></div>
<p>Run <code>IQ-TREE</code> with 1000 replicates of <code>bootstrap</code> and <code>alrt</code>.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/vhsv_mafft.fa<span class="w"> </span>-m<span class="w"> </span>TVMe+I+R2<span class="w"> </span>-pre<span class="w"> </span>data/iqtree/vhsv<span class="w"> </span>-bb<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-alrt<span class="w"> </span><span class="m">1000</span>
</span></code></pre></div>
<p>Examine the <code>IQ-TREE</code> output folder.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>ls<span class="w"> </span>data/iqtree
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>vhsv.bionj   vhsv.contree  vhsv.log vhsv.splits.nex
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>vhsv.ckp.gz  vhsv.iqtree   vhsv.mldist  vhsv.treefile
</span></code></pre></div>
<p>Examine <code>vhsv.treefile</code> file.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>cat<span class="w"> </span>data/iqtree/vhsv.treefile
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>(AU-8-95:0.0110908144,(CH-FI262BFH:0.0131626927,(DK-200098:0.0013245216,DK-9995144:0.0000021124)99.9/100:0.0095461686)9/64:0.0007239176,((((((((DK-1p40:0.0019819713,DK-1p86:0.0019797009)76.6/97:0.0006555511,((DK-1p8:0.0000009918,(((((DK-4p37:0.0000009918,SE-SVA14:0.0006553525)86.8/99:0.0006552139,DK-5e59:0.0013137944)0/59:0.0000009918,SE-SVA-1033:0.0000009918)78.3/95:0.0006575436,((DK-6p403:0.0006555940,SE-SVA31:0.0006553577)0/84:0.0000009918,UK-MLA98-6HE1:0.0000009918)85.8/98:0.0019746439)74.9/96:0.0006544639,KRRV9601:0.0006553263)0/5:0.0000009918)0/35:0.0000009918,UK-9643:0.0033018138)77.4/95:0.0006559696)99.4/100:0.0054050544,DK-M.rhabdo:0.0026557033)97.1/99:0.0043768013,((((((DK-1p53:0.0006554948,DK-1p55:0.0000009918)100/100:0.0816239127,(US-Makah:0.0285256070,US-Goby1-5:0.0140617136)100/100:0.1210196539)45.4/78:0.0178122396,(((DK-4p101:0.0098266846,((DK-4p168:0.0013124829,(UK-H17-2-95:0.0026484120,(UK-H17-5-93:0.0006595661,UK-MLA98-6PT11:0.0013182714)76.2/99:0.0006379007)76.8/99:0.0006676816)79.4/100:0.0010717553,IR-F13.02.97:0.0055710520)87.9/96:0.0020014929)75.6/98:0.0009426132,FR-L59X:0.0128796948)94.9/100:0.0075738902,UK-860-94:0.0158845820)100/100:0.0465027783)99.6/100:0.0335713781,GE-1.2:0.0179796623)90.6/90:0.0051656186,(((DK-2835:0.0040058973,(DK-5123:0.0028538659,DK-5131:0.0044317727)78/92:0.0031564427)99.9/100:0.0132939283,DK-Hededam:0.0099438696)73.4/75:0.0008365002,DK-F1:0.0114614651)0/60:0.0000024805)79.3/82:0.0007471327,((FI-ka422:0.0032972674,FI-ka66:0.0000009918)92.8/100:0.0026339174,NO-A16368G:0.0019943447)92.3/97:0.0020403429)61.6/78:0.0002733446)98.6/96:0.0054285225,(FR-1458:0.0099883404,FR-2375:0.0066740966)90.8/95:0.0027197676)93.6/99:0.0031601931,(((((((((((DK-200027-3:0.0020089775,DK-200079-1:0.0020149815)76.7/89:0.0006222640,(DK-9795568:0.0019716416,DK-9995007:0.0013117492)0/86:0.0000009918)83/84:0.0006608276,DK-9895093:0.0013141923)0/73:0.0000020835,DK-7380:0.0026333937)52.9/85:0.0012862237,DK-9595168:0.0020301147)95.9/99:0.0026894771,DK-6045:0.0000009918)97.5/100:0.0026369103,DK-7974:0.0039639852)0/85:0.0000009918,DK-5151:0.0000009918)75.6/97:0.0006726697,DK-6137:0.0006450449)99.9/100:0.0115741245,((DK-3592B:0.0019729337,(DK-3946:0.0006552354,Fil3:0.0006618033)0/65:0.0000009918)0/86:0.0000742921,DK-3971:0.0025543911)99.5/100:0.0068250583)0/86:0.0002859106,(DK-5741:0.0000009918,(DK-9695377:0.0013153923,(DK-9895024:0.0000009918,DK-9895174:0.0013114345)85.7/98:0.0013140555)90.9/99:0.0013147689)99.7/100:0.0081963781)95.9/77:0.0050968140)79.8/66:0.0033929129,FR-0284:0.0121797246)78.3/89:0.0028127442,FR-0771:0.0001556512)99.4/100:0.0075113449);
</span></code></pre></div>
<p>This is the tree in <code>Newick</code> format that we will use for visualization!<br></p>
<hr />
<h3 id="step-4-examine-metadata"><strong>Step 4: Examine metadata</strong></h3>
<p>Import <code>pandas</code>.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
</span></code></pre></div>
<p>Read the metadata file.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;data/metadata.csv&#39;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<table>
<thead>
<tr>
<th></th>
<th>Strain</th>
<th>Host</th>
<th>Water</th>
<th>Country</th>
<th>ACCNo</th>
<th>Year</th>
</tr>
</thead>
<tbody>
<tr>
<td>0</td>
<td>AU-8-95</td>
<td>Rainbow trout</td>
<td>Fresh water</td>
<td>AU</td>
<td>AY546570.1</td>
<td>1995</td>
</tr>
<tr>
<td>1</td>
<td>CH-FI262BFH</td>
<td>Rainbow trout</td>
<td>Fresh water</td>
<td>CH</td>
<td>AY546571.1</td>
<td>1999</td>
</tr>
<tr>
<td>2</td>
<td>DK-1p40</td>
<td>Rockling</td>
<td>Sea water</td>
<td>DK</td>
<td>AY546575.1</td>
<td>1996</td>
</tr>
<tr>
<td>3</td>
<td>DK-1p53</td>
<td>Atlantic Herring</td>
<td>Sea water</td>
<td>DK</td>
<td>AY546577.1</td>
<td>1996</td>
</tr>
<tr>
<td>4</td>
<td>DK-1p55</td>
<td>Sprat</td>
<td>Sea water</td>
<td>DK</td>
<td>AY546578.1</td>
<td>1996</td>
</tr>
<tr>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr>
<td>56</td>
<td>UK-H17-5-93</td>
<td>Cod</td>
<td>Sea water</td>
<td>UK</td>
<td>AY546630.1</td>
<td>1993</td>
</tr>
<tr>
<td>57</td>
<td>UK-MLA98-6HE1</td>
<td>Herring</td>
<td>Sea water</td>
<td>UK</td>
<td>AY546631.1</td>
<td>1998</td>
</tr>
<tr>
<td>58</td>
<td>UK-MLA98-6PT11</td>
<td>Norway prout</td>
<td>Sea water</td>
<td>UK</td>
<td>AY546632.1</td>
<td>1998</td>
</tr>
<tr>
<td>59</td>
<td>US-Makah</td>
<td>Coho salmon</td>
<td>Fresh water</td>
<td>USA</td>
<td>U28747.1</td>
<td>1988</td>
</tr>
<tr>
<td>60</td>
<td>US-Goby1-5</td>
<td>Round goby</td>
<td>Unknown</td>
<td>USA</td>
<td>AB672615.1</td>
<td>2006</td>
</tr>
</tbody>
</table>
<p>61 rows × 6 columns</p>
<hr />
<h2 id="part-2-visualizing-the-tree"><strong>Part 2: Visualizing the tree</strong></h2>
<h3 id="step-1-preparation"><strong>Step 1: Preparation</strong></h3>
<p>1st step - install or call libraries.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="nf">if </span><span class="p">(</span><span class="o">!</span><span class="nf">require</span><span class="p">(</span><span class="s">&quot;pacman&quot;</span><span class="p">))</span><span class="w"> </span><span class="nf">install.packages</span><span class="p">(</span><span class="s">&quot;pacman&quot;</span><span class="p">)</span>
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="n">pacman</span><span class="o">::</span><span class="nf">p_load</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">,</span><span class="w"> </span><span class="n">ggtree</span><span class="p">,</span><span class="w"> </span><span class="n">phangorn</span><span class="p">,</span><span class="w"> </span><span class="n">treeio</span><span class="p">,</span><span class="w"> </span><span class="n">ggnewscale</span><span class="p">,</span><span class="w"> </span><span class="n">viridis</span><span class="p">)</span>
</span></code></pre></div>
<p>2nd step - set the working directory.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="n">main_dir</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">dirname</span><span class="p">(</span><span class="n">rstudioapi</span><span class="o">::</span><span class="nf">getSourceEditorContext</span><span class="p">()</span><span class="o">$</span><span class="n">path</span><span class="p">)</span>
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="nf">setwd</span><span class="p">(</span><span class="n">main_dir</span><span class="p">)</span>
</span></code></pre></div>
<hr />
<h3 id="step-2-initial-visualization"><strong>Step 2: Initial visualization</strong></h3>
<p>Create a directory for images.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;imgs&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">mode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;0777&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p>As we remember from the previous laboratory journal our tree file is in <code>Newick</code> format and it was created with <code>IQ-TREE</code>. Also, there are bootstrap values in our tree files. That is why it is better to read the tree with <code>read.iqtree</code> function instead of default <code>read.tree</code> function.<br></p>
<p>So let us read the tree, midpoint root it and make the 1st visualization.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="c1">#Read the tree</span>
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a><span class="n">vshv.tree</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.iqtree</span><span class="p">(</span><span class="s">&#39;data/iqtree/vhsv.treefile&#39;</span><span class="p">)</span>
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="c1">#Midpoint root the tree</span>
</span><span id="__span-16-5"><a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a><span class="n">vshv.tree</span><span class="o">@</span><span class="n">phylo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">midpoint</span><span class="p">(</span><span class="n">vshv.tree</span><span class="o">@</span><span class="n">phylo</span><span class="p">)</span>
</span><span id="__span-16-6"><a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a>
</span><span id="__span-16-7"><a id="__codelineno-16-7" name="__codelineno-16-7" href="#__codelineno-16-7"></a><span class="c1">#Display the tree</span>
</span><span id="__span-16-8"><a id="__codelineno-16-8" name="__codelineno-16-8" href="#__codelineno-16-8"></a><span class="nf">plot</span><span class="p">(</span><span class="n">vshv.tree</span><span class="o">@</span><span class="n">phylo</span><span class="p">)</span>
</span><span id="__span-16-9"><a id="__codelineno-16-9" name="__codelineno-16-9" href="#__codelineno-16-9"></a>
</span><span id="__span-16-10"><a id="__codelineno-16-10" name="__codelineno-16-10" href="#__codelineno-16-10"></a><span class="c1">#And let&#39;s save the tree</span>
</span><span id="__span-16-11"><a id="__codelineno-16-11" name="__codelineno-16-11" href="#__codelineno-16-11"></a><span class="nf">png</span><span class="p">(</span><span class="n">filename</span><span class="o">=</span><span class="s">&quot;imgs/1st_tree.png&quot;</span><span class="p">)</span>
</span><span id="__span-16-12"><a id="__codelineno-16-12" name="__codelineno-16-12" href="#__codelineno-16-12"></a><span class="nf">plot</span><span class="p">(</span><span class="n">vshv.tree</span><span class="o">@</span><span class="n">phylo</span><span class="p">)</span>
</span><span id="__span-16-13"><a id="__codelineno-16-13" name="__codelineno-16-13" href="#__codelineno-16-13"></a><span class="nf">dev.off</span><span class="p">()</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/1st_tree.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/1st_tree.png" align='center', width="50%"></a>
</div>

<p>Um. Okay, but nothing special! There is a lot of work ahead to make this tree pretty!<br></p>
<hr />
<h3 id="step-3-adding-host-information-to-the-tree"><strong>Step 3: Adding host information to the tree</strong></h3>
<p>Now we will read the metadata and add the host information to the tree. To do so we will use the <code>%&lt;+%</code> sign, which allow us to use different columns of the metadata to decorate different parts of the tree.<br></p>
<p>Let's add host information to the tip points.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="c1">#Read the metadata</span>
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a><span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;data/metadata.csv&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&#39;,&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a>
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="c1">#Plot the tree and add host information from the metadata to the tree&#39;s tip points</span>
</span><span id="__span-17-5"><a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a><span class="n">tree_2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggtree</span><span class="p">(</span><span class="n">vshv.tree</span><span class="p">)</span><span class="w"> </span><span class="o">%&lt;+%</span><span class="w"> </span><span class="n">meta</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-17-6"><a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a><span class="w">  </span><span class="nf">geom_tippoint</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">Host</span><span class="p">))</span>
</span><span id="__span-17-7"><a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a>
</span><span id="__span-17-8"><a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a><span class="c1">#Display the tree</span>
</span><span id="__span-17-9"><a id="__codelineno-17-9" name="__codelineno-17-9" href="#__codelineno-17-9"></a><span class="n">tree_2</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/2nd_tree.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/2nd_tree.png" align='center', width="50%"></a>
</div>

<p>And let's save the tree.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;imgs/2nd_tree.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">tree_2</span><span class="p">,</span><span class="w"> </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">600</span><span class="p">)</span>
</span></code></pre></div>
<hr />
<h3 id="step-4-use-the-circular-layout-and-add-bootstrap-information"><strong>Step 4: Use the circular layout and add bootstrap information</strong></h3>
<p><code>IQ-TREE</code> Newick file contains a bootstrap support value which has both <code>SH_aLRT</code> and <code>UFboot</code> values. Let's show branches that has high support for both of parameters (i.e. ≥ 70).<br></p>
<p>When we define a <code>ggtree</code> object using <code>tree_3 &lt;- ggtree(vshv.tree) %&lt;+% meta + geom_tippoint(aes(color=Host))</code>, it attaches the metadata as well as tree-branch and tips info coming from original Newick tree-file in the data placeholder, which we can use by <code>tree_3$data</code>. So let’s update this to have a new column of a variable that satisfy both bootstrap values:<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="c1">#Use the circular layout for the tree</span>
</span><span id="__span-19-2"><a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="n">tree_3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggtree</span><span class="p">(</span><span class="n">vshv.tree</span><span class="p">,</span><span class="w"> </span><span class="n">layout</span><span class="o">=</span><span class="s">&quot;circular&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&lt;+%</span><span class="w"> </span><span class="n">meta</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-19-3"><a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a><span class="w">  </span><span class="nf">geom_tippoint</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">Host</span><span class="p">))</span>
</span><span id="__span-19-4"><a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a>
</span><span id="__span-19-5"><a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a><span class="c1">#Create a new parameter `bootstrap` with the default value of 0</span>
</span><span id="__span-19-6"><a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a><span class="n">tree_3</span><span class="o">$</span><span class="n">data</span><span class="o">$</span><span class="n">bootstrap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;0&#39;</span>
</span><span id="__span-19-7"><a id="__codelineno-19-7" name="__codelineno-19-7" href="#__codelineno-19-7"></a>
</span><span id="__span-19-8"><a id="__codelineno-19-8" name="__codelineno-19-8" href="#__codelineno-19-8"></a><span class="c1">#Assign value 1 to the tree branches that has both SH_aLRT and UFboot values higher than 70</span>
</span><span id="__span-19-9"><a id="__codelineno-19-9" name="__codelineno-19-9" href="#__codelineno-19-9"></a><span class="n">tree_3</span><span class="o">$</span><span class="n">data</span><span class="p">[</span><span class="nf">which</span><span class="p">(</span><span class="n">tree_3</span><span class="o">$</span><span class="n">data</span><span class="o">$</span><span class="n">SH_aLRT</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">70</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">tree_3</span><span class="o">$</span><span class="n">data</span><span class="o">$</span><span class="n">UFboot</span><span class="w">  </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">70</span><span class="p">),]</span><span class="o">$</span><span class="n">bootstrap</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;1&#39;</span>
</span><span id="__span-19-10"><a id="__codelineno-19-10" name="__codelineno-19-10" href="#__codelineno-19-10"></a>
</span><span id="__span-19-11"><a id="__codelineno-19-11" name="__codelineno-19-11" href="#__codelineno-19-11"></a><span class="c1">#Plot the tree and use &quot;black&quot; color for branches with bootstrap value = 1 (&gt;=70) and &quot;grey&quot; color for branches with bootstrap value = 0 (&lt;70)</span>
</span><span id="__span-19-12"><a id="__codelineno-19-12" name="__codelineno-19-12" href="#__codelineno-19-12"></a><span class="n">tree_3</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tree_3</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">new_scale_color</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-19-13"><a id="__codelineno-19-13" name="__codelineno-19-13" href="#__codelineno-19-13"></a><span class="w">  </span><span class="nf">geom_tree</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">color</span><span class="o">=</span><span class="n">bootstrap</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&#39;1&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-19-14"><a id="__codelineno-19-14" name="__codelineno-19-14" href="#__codelineno-19-14"></a><span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">name</span><span class="o">=</span><span class="s">&#39;Bootstrap&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">values</span><span class="o">=</span><span class="nf">setNames</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;grey&quot;</span><span class="p">),</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="bp">T</span><span class="p">,</span><span class="bp">F</span><span class="p">)),</span><span class="w"> </span><span class="n">guide</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;none&quot;</span><span class="p">)</span>
</span><span id="__span-19-15"><a id="__codelineno-19-15" name="__codelineno-19-15" href="#__codelineno-19-15"></a>
</span><span id="__span-19-16"><a id="__codelineno-19-16" name="__codelineno-19-16" href="#__codelineno-19-16"></a><span class="c1">#Display the tree</span>
</span><span id="__span-19-17"><a id="__codelineno-19-17" name="__codelineno-19-17" href="#__codelineno-19-17"></a><span class="n">tree_3</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/3rd_tree.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/3rd_tree.png" align='center', width="50%"></a>
</div>

<p>And let's save the tree.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;imgs/3rd_tree.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">tree_3</span><span class="p">,</span><span class="w"> </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">600</span><span class="p">)</span>
</span></code></pre></div>
<hr />
<h3 id="step-5-add-the-last-metadata-info-as-the-heatmaps-and-make-the-final-visualization"><strong>Step 5: Add the last metadata info as the heatmaps and make the final visualization</strong></h3>
<p>Create two separate dataframes for <code>Water</code> and <code>Year</code> and parse them to make further visualize much more convenient.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="c1">#Read Water column from metadata to a separate dataframe</span>
</span><span id="__span-21-2"><a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="n">meta.water</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">meta</span><span class="p">[,</span><span class="s">&#39;Water&#39;</span><span class="p">])</span>
</span><span id="__span-21-3"><a id="__codelineno-21-3" name="__codelineno-21-3" href="#__codelineno-21-3"></a><span class="c1">#Change column name from `meta[,&#39;Water&#39;]` to `Water`</span>
</span><span id="__span-21-4"><a id="__codelineno-21-4" name="__codelineno-21-4" href="#__codelineno-21-4"></a><span class="nf">colnames</span><span class="p">(</span><span class="n">meta.water</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;Water&#39;</span>
</span><span id="__span-21-5"><a id="__codelineno-21-5" name="__codelineno-21-5" href="#__codelineno-21-5"></a><span class="c1">#Change row names from `1, 2, 3, ...` to strains `AU-8-95, CH-FI262BFH, ...`</span>
</span><span id="__span-21-6"><a id="__codelineno-21-6" name="__codelineno-21-6" href="#__codelineno-21-6"></a><span class="nf">rownames</span><span class="p">(</span><span class="n">meta.water</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">meta</span><span class="o">$</span><span class="n">Strain</span>
</span><span id="__span-21-7"><a id="__codelineno-21-7" name="__codelineno-21-7" href="#__codelineno-21-7"></a>
</span><span id="__span-21-8"><a id="__codelineno-21-8" name="__codelineno-21-8" href="#__codelineno-21-8"></a><span class="c1">#Read Year column from metadata to a separate dataframe</span>
</span><span id="__span-21-9"><a id="__codelineno-21-9" name="__codelineno-21-9" href="#__codelineno-21-9"></a><span class="n">meta.year</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">meta</span><span class="p">[,</span><span class="s">&#39;Year&#39;</span><span class="p">])</span>
</span><span id="__span-21-10"><a id="__codelineno-21-10" name="__codelineno-21-10" href="#__codelineno-21-10"></a><span class="c1">##Change column name from `meta[,&#39;Year&#39;]` to `Year`</span>
</span><span id="__span-21-11"><a id="__codelineno-21-11" name="__codelineno-21-11" href="#__codelineno-21-11"></a><span class="nf">colnames</span><span class="p">(</span><span class="n">meta.year</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&#39;Year&#39;</span>
</span><span id="__span-21-12"><a id="__codelineno-21-12" name="__codelineno-21-12" href="#__codelineno-21-12"></a><span class="c1">#Change row names from `1, 2, 3, ...` to strains `AU-8-95, CH-FI262BFH, ...`</span>
</span><span id="__span-21-13"><a id="__codelineno-21-13" name="__codelineno-21-13" href="#__codelineno-21-13"></a><span class="nf">rownames</span><span class="p">(</span><span class="n">meta.year</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">meta</span><span class="o">$</span><span class="n">Strain</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a><span class="c1">#Create a new tree that uses the previous tree (circular layout + host info at the tip points)</span>
</span><span id="__span-22-2"><a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a><span class="c1">#Add `meta.water` dataframe as the heatmap and set the width to 0.2 and offset to 0.01</span>
</span><span id="__span-22-3"><a id="__codelineno-22-3" name="__codelineno-22-3" href="#__codelineno-22-3"></a><span class="c1">#Use `viridis` &quot;A&quot; colormap and name legend title &quot;Water&quot;</span>
</span><span id="__span-22-4"><a id="__codelineno-22-4" name="__codelineno-22-4" href="#__codelineno-22-4"></a><span class="n">final_tree</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gheatmap</span><span class="p">(</span><span class="n">tree_3</span><span class="p">,</span><span class="w"> </span><span class="n">meta.water</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">0.2</span><span class="p">,</span><span class="w"> </span><span class="n">offset</span><span class="o">=</span><span class="m">0.01</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
</span><span id="__span-22-5"><a id="__codelineno-22-5" name="__codelineno-22-5" href="#__codelineno-22-5"></a><span class="w">  </span><span class="nf">scale_fill_viridis_d</span><span class="p">(</span><span class="n">option</span><span class="o">=</span><span class="s">&quot;A&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">name</span><span class="o">=</span><span class="s">&quot;Water&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-22-6"><a id="__codelineno-22-6" name="__codelineno-22-6" href="#__codelineno-22-6"></a><span class="w">  </span><span class="nf">new_scale_fill</span><span class="p">()</span><span class="w"> </span><span class="c1">#define a new fill scale using after the first gheatmap, for the second gheatmap to draw upon</span>
</span><span id="__span-22-7"><a id="__codelineno-22-7" name="__codelineno-22-7" href="#__codelineno-22-7"></a>
</span><span id="__span-22-8"><a id="__codelineno-22-8" name="__codelineno-22-8" href="#__codelineno-22-8"></a><span class="c1">#Create a new tree that uses the tree created above</span>
</span><span id="__span-22-9"><a id="__codelineno-22-9" name="__codelineno-22-9" href="#__codelineno-22-9"></a><span class="c1">#Add `meta.year` dataframe as the heatmap and set the width to 0.1 and offset to 0.05</span>
</span><span id="__span-22-10"><a id="__codelineno-22-10" name="__codelineno-22-10" href="#__codelineno-22-10"></a><span class="c1">#Use `viridis` &quot;D&quot; colormap and name legend title &quot;Year&quot;</span>
</span><span id="__span-22-11"><a id="__codelineno-22-11" name="__codelineno-22-11" href="#__codelineno-22-11"></a><span class="n">final_tree</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gheatmap</span><span class="p">(</span><span class="n">final_tree</span><span class="p">,</span><span class="w"> </span><span class="n">meta.year</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">0.1</span><span class="p">,</span><span class="w"> </span><span class="n">offset</span><span class="o">=</span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
</span><span id="__span-22-12"><a id="__codelineno-22-12" name="__codelineno-22-12" href="#__codelineno-22-12"></a><span class="w">  </span><span class="nf">scale_fill_viridis</span><span class="p">(</span><span class="n">option</span><span class="o">=</span><span class="s">&quot;D&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">name</span><span class="o">=</span><span class="s">&quot;Year&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p>Add the country information from the metadata to the tip labels and plot the final tree!<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-23-1"><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="c1">#Add the country information to the tip labels, set the color to &quot;gray40&quot;, offset to 0.03 and size to 3</span>
</span><span id="__span-23-2"><a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="n">final_tree</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">final_tree</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_tiplab2</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">label</span><span class="o">=</span><span class="n">Country</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="o">=</span><span class="s">&quot;gray40&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">offset</span><span class="o">=</span><span class="m">0.003</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">3</span><span class="p">)</span>
</span><span id="__span-23-3"><a id="__codelineno-23-3" name="__codelineno-23-3" href="#__codelineno-23-3"></a>
</span><span id="__span-23-4"><a id="__codelineno-23-4" name="__codelineno-23-4" href="#__codelineno-23-4"></a><span class="c1">#Display the tree</span>
</span><span id="__span-23-5"><a id="__codelineno-23-5" name="__codelineno-23-5" href="#__codelineno-23-5"></a><span class="n">final_tree</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/final_tree.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_07_Visualization_Pro/imgs/final_tree.png" align='center', width="50%"></a>
</div>

<p>And let's save the tree.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-r highlight"><pre><span></span><code><span id="__span-24-1"><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="nf">ggsave</span><span class="p">(</span><span class="s">&quot;imgs/final_tree.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">final_tree</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">8</span><span class="p">,</span><span class="w"> </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">600</span><span class="p">)</span>
</span></code></pre></div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 Ilia Popov
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy", "search", "search.suggest", "search.highlight"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>