
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/01_Quality_Control/">
      
      
        <link rel="prev" href="../00_Introduction/">
      
      
        <link rel="next" href="../02_Variant_Calling/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.38">
    
    
      
        <title>I. Quality Control of Raw Data - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#i-quality-control-of-raw-data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              I. Quality Control of Raw Data
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    NGS Analysis Handbook
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome to MkDocs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../00_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-run-fastqc-on-a-pair-end-sequencing-data" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Run FastQC on a pair-end sequencing data
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1: Run FastQC on a pair-end sequencing data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-tile-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Per tile sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-quality-scores" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence quality scores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-content" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-cleaning-reads-from-adapters-and-low-quality-sequences-with-trimmomatic" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Cleaning reads from adapters and low quality sequences with Trimmomatic
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Cleaning reads from adapters and low quality sequences with Trimmomatic">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#adapter-trimming" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter trimming
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-evaluate-the-data-quality-after-trimming-using-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Evaluate the data quality after trimming using FastQC
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Evaluate the data quality after trimming using FastQC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-quality_1" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-content_1" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../02_Variant_Calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../03_Whole_%28pan%29genome_analyses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_03_MSA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_04_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_05_Root_Date/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_07_Visualization_Pro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../IV.%20Phylogenetics/04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_01_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../V.%2016S_amplicon_analysis/05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-run-fastqc-on-a-pair-end-sequencing-data" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Run FastQC on a pair-end sequencing data
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1: Run FastQC on a pair-end sequencing data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-tile-sequence-quality" class="md-nav__link">
    <span class="md-ellipsis">
      Per tile sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-quality-scores" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence quality scores
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#per-sequence-gc-content" class="md-nav__link">
    <span class="md-ellipsis">
      Per sequence GC content
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-content" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-cleaning-reads-from-adapters-and-low-quality-sequences-with-trimmomatic" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Cleaning reads from adapters and low quality sequences with Trimmomatic
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Cleaning reads from adapters and low quality sequences with Trimmomatic">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#adapter-trimming" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter trimming
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-evaluate-the-data-quality-after-trimming-using-fastqc" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Evaluate the data quality after trimming using FastQC
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Evaluate the data quality after trimming using FastQC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#per-base-sequence-quality_1" class="md-nav__link">
    <span class="md-ellipsis">
      Per base sequence quality
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adapter-content_1" class="md-nav__link">
    <span class="md-ellipsis">
      Adapter Content
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="i-quality-control-of-raw-data"><strong>I. Quality Control of Raw Data</strong></h1>
<blockquote>
<p>This chapter contains a manual on quality control of raw data in NGS data analysis.</p>
</blockquote>
<p>We will use the <code>FastQC</code> tool to assess the quality of the raw sequencing data.</p>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p><a class="glightbox" href="https://img.shields.io/badge/Language-Command Line Tools-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="CLT" src="https://img.shields.io/badge/Language-Command Line Tools-steelblue" /></a><br>
<a class="glightbox" href="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" /></a></p>
<p>You can run commands below in your <code>terminal</code>.<br>
But if you want to write a convenient to read laboratory journal you can use <code>Jupyter Notebook</code>.<br>
In that case write <code>!</code> in the beggining of each cell to make it understand <code>bash</code> commands.</p>
<p>To recreate any of the steps of this manual please install:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>conda<span class="w"> </span>env<span class="w"> </span>create<span class="w"> </span>-f<span class="w"> </span>qc.yaml
</span></code></pre></div>
<p>And of cource do not forget to activate the envinronment!</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>conda<span class="w"> </span>activate<span class="w"> </span>qc
</span></code></pre></div>
<hr />
<h2 id="step-1-run-fastqc-on-a-pair-end-sequencing-data"><strong>Step 1: Run <code>FastQC</code> on a pair-end sequencing data</strong></h2>
<blockquote>
<p>Note that the folder for output results must exist, i.e. it should be created before running the program.
What problems are observed? How can they be fixed?</p>
</blockquote>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>mkdir<span class="w"> </span>fastqc_results
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>fastqc<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="w">    </span>data/Sample_1.R1.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="w">    </span>data/Sample_1.R2.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="w">    </span>-o<span class="w"> </span>fastqc_results
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>application/gzip
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>Started analysis of Sample_1.R1.fastq.gz
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>application/gzip
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>Analysis complete for Sample_1.R1.fastq.gz
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>Started analysis of Sample_1.R2.fastq.gz
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>Analysis complete for Sample_1.R2.fastq.gz
</span></code></pre></div>
<p>Results are saved to:<br>
- <code>Sample_1.R1_fastqc.html</code><br>
- <code>Sample_1.R2_fastqc.html</code><br>
Please open them in any web browser you use:</p>
<h3 id="per-base-sequence-quality"><strong>Per base sequence quality</strong></h3>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20quality%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20quality%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20quality%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20quality%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>We have boxplot graphs that show the quality by read length. We can see that for file <code>R2</code> the quality is expectedly worse than for <code>R1</code>. Reverse reads are worse a little bit, this is normal because reverse reads are timed to read later than forward reads, but similarly we can see that towards the end of the read the quality drops, which is expected.</p>
<h3 id="per-tile-sequence-quality"><strong>Per tile sequence quality</strong></h3>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20tile%20sequence%20quality%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20tile%20sequence%20quality%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20tile%20sequence%20quality%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20tile%20sequence%20quality%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>These graphs show the quality per individual "tile". There is a slight inequality here, but in fact we have too few readings to notice any trends here. The main thing is that if there are any clear situations in this graph, where there is a group of "tiles" that are doing badly and they are next to each other, this is a reason to suspect that we have a problem with the flow cell (it is dirty).</p>
<h3 id="per-sequence-quality-scores"><strong>Per sequence quality scores</strong></h3>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20quality%20scores%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20quality%20scores%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20quality%20scores%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20quality%20scores%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>Quality averaged per read. Distribution: more than 100 reads with normal quality, but there is also a rather large tail with reads that average poor quality.</p>
<h3 id="per-base-sequence-content"><strong>Per base sequence content</strong></h3>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20content%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20content%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20content%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/01_Quality_Control/imgs/pre-trim/Per%20base%20sequence%20content%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>Distribution of nucleotide composition by position in the read. It can be seen that there are no sharp peaks here, although the graph looks rather broken. This is also because we have few reads. If we had more reads, the graph would probably be more or less even. The main thing is that there are no sharp big peaks.</p>
<h3 id="per-sequence-gc-content"><strong>Per sequence GC content</strong></h3>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20GC%20content%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20GC%20content%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20GC%20content%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Per%20sequence%20GC%20content%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>GC composition distribution. It is used to analyse contamination. If we had 2 peaks on the graph, we could say that we probably have contaminants because we have organisms with different GC compositions present. It may appear that there is also a second peak in our data, but we have little data and the "little extra hump" merges with the main hump, meaning it is not a full-fledged second peak.</p>
<h3 id="adapter-content"><strong>Adapter Content</strong></h3>
<table>
<thead>
<tr>
<th>Forward read</th>
<th>Reverse read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Adapter%20Content%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Adapter%20Content%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Adapter%20Content%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/pre-trim/Adapter%20Content%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>This is the biggest problem in our data. We see that we have a fairly early increase in the percentages of adapter content in the reads. In the most recent nucleotide reads, over 60% of the reads contain adapters. This tells us that the library was short, so the adapters got in. This means that we have a very short piece of DNA, and when we read it, we have read it all, and then we go on to read the adapters.</p>
<hr />
<h2 id="step-2-cleaning-reads-from-adapters-and-low-quality-sequences-with-trimmomatic"><strong>Step 2: Cleaning reads from adapters and low quality sequences with <code>Trimmomatic</code></strong></h2>
<blockquote>
<p>What do the options mean?<br>
How many readings remain after <code>Trimmomatic</code> data processing?</p>
</blockquote>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>mkdir<span class="w"> </span>trimmomatic_result
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>trimmomatic<span class="w"> </span>PE<span class="w"> </span>-phred33<span class="w"> </span>data/Sample_1.R1.fastq.gz<span class="w"> </span>data/Sample_1.R2.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a><span class="w">    </span>trimmomatic_result/Sample_1.R1.paired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="w">    </span>trimmomatic_result/Sample_1.R1.unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a><span class="w">    </span>trimmomatic_result/Sample_1.R2.paired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a><span class="w">    </span>trimmomatic_result/Sample_1.R2.unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="w">    </span>ILLUMINACLIP:<span class="s1">&#39;adapters/NexteraPE-PE.fa&#39;</span>:2:30:10<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a><span class="w">    </span>LEADING:10<span class="w"> </span>TRAILING:10<span class="w"> </span>SLIDINGWINDOW:4:10<span class="w"> </span>MINLEN:50
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>TrimmomaticPE: Started with arguments:
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a> -phred33 data/Sample_1.R1.fastq.gz data/Sample_1.R2.fastq.gz trimmomatic_result/Sample_1.R1.paired.fastq.gz trimmomatic_result/Sample_1.R1.unpaired.fastq.gz trimmomatic_result/Sample_1.R2.paired.fastq.gz trimmomatic_result/Sample_1.R2.unpaired.fastq.gz ILLUMINACLIP:adapters/NexteraPE-PE.fa:2:30:10 LEADING:10 TRAILING:10 SLIDINGWINDOW:4:10 MINLEN:50
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>Multiple cores found: Using 4 threads
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>Using PrefixPair: &#39;AGATGTGTATAAGAGACAG&#39; and &#39;AGATGTGTATAAGAGACAG&#39;
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>Using Long Clipping Sequence: &#39;GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG&#39;
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>Using Long Clipping Sequence: &#39;TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG&#39;
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a>Using Long Clipping Sequence: &#39;CTGTCTCTTATACACATCTCCGAGCCCACGAGAC&#39;
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>Using Long Clipping Sequence: &#39;CTGTCTCTTATACACATCTGACGCTGCCGACGA&#39;
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a>ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
</span><span id="__span-7-10"><a id="__codelineno-7-10" name="__codelineno-7-10" href="#__codelineno-7-10"></a>Input Read Pairs: 767 Both Surviving: 213 (27.77%) Forward Only Surviving: 428 (55.80%) Reverse Only Surviving: 10 (1.30%) Dropped: 116 (15.12%)
</span><span id="__span-7-11"><a id="__codelineno-7-11" name="__codelineno-7-11" href="#__codelineno-7-11"></a>TrimmomaticPE: Completed successfully
</span></code></pre></div>
<p>Let's break down the options:</p>
<ul>
<li><code>trimmomatic</code>: Invokes the Trimmomatic tool.</li>
<li><code>PE</code>: Indicates that the input data consists of paired-end reads.</li>
<li><code>-phred33</code>: Specifies that the quality scores are in Phred+33 encoding. Phred+33 is a common format for quality scores in Illumina sequencing data.</li>
<li><code>data/Sample_1.R1.fastq.gz</code> and <code>data/Sample_1.R2.fastq.gz</code>: Input files containing the paired-end reads for Sample_1. <code>.fastq.gz</code> indicates that the files are in FASTQ format and compressed with gzip.</li>
<li><code>trimmomatic_result/Sample_1.R1.paired.fastq.gz</code> and <code>trimmomatic_result/Sample_1.R1.unpaired.fastq.gz</code>: Output files containing the paired and unpaired reads for Sample_1 from read 1.</li>
<li><code>trimmomatic_result/Sample_1.R2.paired.fastq.gz</code> and <code>trimmomatic_result/Sample_1.R2.unpaired.fastq.gz</code>: Output files containing the paired and unpaired reads for Sample_1 from read 2.</li>
<li><code>ILLUMINACLIP:'adapters/NexteraPE-PE.fa':2:30:10</code>: Adapter clipping. This option specifies the adapter sequences file (<code>NexteraPE-PE.fa</code>), followed by seed mismatches (2), palindrome clip threshold (30), and simple clip threshold (10). It trims adapters and other illumina-specific sequences from the reads.</li>
<li><code>LEADING:10</code>: Removes low-quality bases from the start of the reads if they have a quality score lower than 10.</li>
<li><code>TRAILING:10</code>: Removes low-quality bases from the end of the reads if they have a quality score lower than 10.</li>
<li><code>SLIDINGWINDOW:4:10</code>: Performs a sliding window trimming. It scans the read with a 4-base wide window, cutting when the average quality per base drops below 10.</li>
<li><code>MINLEN:50</code>: Discards reads that are shorter than 50 bases after trimming.</li>
</ul>
<p>In summary, this command takes paired-end FASTQ files as input, trims adapters, removes low-quality bases, and performs quality-based trimming to improve the overall quality of the reads. It then separates the trimmed reads into paired and unpaired files for both reads 1 and 2.</p>
<p>Input Read Pairs: 767<br>
- Both Surviving: 213 (27.77%)<br>
- Forward Only Surviving: 428 (55.80%)<br>
- Reverse Only Surviving: 10 (1.30%)<br>
- Dropped: 116 (15.12%)<br></p>
<h3 id="adapter-trimming"><strong>Adapter trimming</strong></h3>
<ul>
<li>Cleaning data from adapters and low quality bases at the ends can improve the results of some programmes</li>
<li>It is critical to clean raw data before genome assembly</li>
<li>There are a number of tools for removing adapters:<ul>
<li><code>Trimmomatic</code></li>
<li><code>fastp</code></li>
<li><code>BBduk</code></li>
<li><code>cutadapt</code></li>
<li>...</li>
</ul>
</li>
</ul>
<p><strong>NB!</strong><br>
When searching for genetic variants using modern tools, removing adapters may not only not improve but even worsen the results of the analysis!</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/01_Quality_Control/imgs/Adapters%20trimming.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/01_Quality_Control/imgs/Adapters%20trimming.png" align='center', width="50%"></a>
</div>

<hr />
<h2 id="step-3-evaluate-the-data-quality-after-trimming-using-fastqc"><strong>Step 3: Evaluate the data quality after trimming using <code>FastQC</code></strong></h2>
<blockquote>
<p>Did the data quality get better after trimming?</p>
</blockquote>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>fastqc<span class="w"> </span>trimmomatic_result/Sample_1.*paired.fastq.gz<span class="w"> </span>-o<span class="w"> </span>fastqc_results
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>application/gzip
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>application/gzip
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>application/gzip
</span><span id="__span-9-4"><a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a>application/gzip
</span><span id="__span-9-5"><a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>Started analysis of Sample_1.R1.paired.fastq.gz
</span><span id="__span-9-6"><a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a>Analysis complete for Sample_1.R1.paired.fastq.gz
</span><span id="__span-9-7"><a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a>Started analysis of Sample_1.R1.unpaired.fastq.gz
</span><span id="__span-9-8"><a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a>Analysis complete for Sample_1.R1.unpaired.fastq.gz
</span><span id="__span-9-9"><a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a>Started analysis of Sample_1.R2.paired.fastq.gz
</span><span id="__span-9-10"><a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a>Analysis complete for Sample_1.R2.paired.fastq.gz
</span><span id="__span-9-11"><a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a>Started analysis of Sample_1.R2.unpaired.fastq.gz
</span><span id="__span-9-12"><a id="__codelineno-9-12" name="__codelineno-9-12" href="#__codelineno-9-12"></a>Analysis complete for Sample_1.R2.unpaired.fastq.gz
</span></code></pre></div>
<p>Results are saved to:<br>
- <code>Sample_1.R1.paired_fastqc.html</code><br>
- <code>Sample_1.R1.unpaired_fastqc.html</code><br>
- <code>Sample_2.R1.paired_fastqc.html</code><br>
- <code>Sample_2.R1.unpaired_fastqc.html</code><br>
Please open them in any web browser you use:<br></p>
<h3 id="per-base-sequence-quality_1"><strong>Per base sequence quality</strong></h3>
<table>
<thead>
<tr>
<th>Forward paired read</th>
<th>Forward unpaired read</th>
<th>Reverse paired read</th>
<th>Reverse unpaired read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20P%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20P%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20UnP%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20UnP%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20P%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20P%20R2.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20UnP%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Per%20base%20sequence%20quality%20UnP%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>Paired readings are those that have a pair preserved. There are far fewer of them.</p>
<h3 id="adapter-content_1"><strong>Adapter Content</strong></h3>
<table>
<thead>
<tr>
<th>Forward paired read</th>
<th>Forward unpaired read</th>
<th>Reverse paired read</th>
<th>Reverse unpaired read</th>
</tr>
</thead>
<tbody>
<tr>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20P%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20P%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20UnP%20R1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20UnP%20R1.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20P%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20P%20R2.png" width="100%"></a></td>
<td><a class="glightbox" href="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20UnP%20R2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/BI-NGS/blob/main/01_Quality_Control/imgs/post-trim/Adapter%20Content%20UnP%20R2.png" width="100%"></a></td>
</tr>
</tbody>
</table>
<p>But if we look at the adapter content chart, we see that adapters have been trimmed, there are hardly any left.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 Ilia Popov
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.code.copy", "search", "search.suggest", "search.highlight"], "search": "../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>