
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/IV.%20Phylogenetics/04_04_Trees/">
      
      
        <link rel="prev" href="../04_03_MSA/">
      
      
        <link rel="next" href="../04_05_Root_Date/">
      
      
      <link rel="icon" href="../../assets/favicon.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.39">
    
    
      
        <title>Preparing the alignment and building trees - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config","G-4RBMQR50GK"),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","G-4RBMQR50GK",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=G-4RBMQR50GK",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script>
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
    <html data-lt-installed="true">
        <head>
            <!-- HTML Meta Tags -->
            <title>NGS Data Analysis Handbook</title>
            <meta name="description" content="">

            <!-- Facebook Meta Tags -->
            <meta property="og:url" content="https://iliapopov17.github.io/NGS-Handbook/">
            <meta property="og:type" content="website">
            <meta property="og:title" content="NGS Data Analysis Handbook">
            <meta property="og:description" content="">
            <meta property="og:image" content="https://iliapopov17.github.io/images/ngs-handbook/metatag.png">

            <!-- Twitter Meta Tags -->
            <meta name="twitter:card" content="summary_large_image">
            <meta property="twitter:domain" content="iliapopov17.github.io">
            <meta property="twitter:url" content="https://iliapopov17.github.io/NGS-Handbook/">
            <meta name="twitter:title" content="NGS Data Analysis Handbook">
            <meta name="twitter:description" content="">
            <meta name="twitter:image" content="https://iliapopov17.github.io/images/ngs-handbook/metatag.png">

            <!-- Meta Tags Generated via https://www.opengraph.xyz -->
        </head>
        <body>
        </body>
    </html>

  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#preparing-the-alignment-and-building-trees" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  <img src="../../assets/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Preparing the alignment and building trees
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9zM20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12zm-9.15 3.96h2.3L12 9z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_2">
    
      <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/iliapopov17/NGS-Handbook" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    iliapopov17/NGS-Handbook
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  <img src="../../assets/logo.svg" alt="logo">

    </a>
    NGS Analysis Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/iliapopov17/NGS-Handbook" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    iliapopov17/NGS-Handbook
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../01_Quality_Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../02_Variant_Calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../03_Whole_%28pan%29genome_analyses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_00_Overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_03_MSA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-cut-bad-areas-out-of-the-alignment-using-trimal" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Cut bad areas out of the alignment using trimAl
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-fit-an-evolution-model-in-modeltest-modeltest-ng" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: fit an evolution model in ModelTest (ModelTest-NG)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-build-an-ml-tree-in-raxml-ng-using-the-selected-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Build an ML-tree in RAxML-NG using the selected model
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-draw-the-resulting-tree-the-best-ml-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Draw the resulting tree (the best ML tree)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-model-selection-in-modelfinder-iq-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Model selection in ModelFinder (IQ-TREE)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-build-an-ml-tree-in-iq-tree-using-the-selected-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Build an ML tree in IQ-TREE using the selected model
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-draw-the-resulting-tree-the-best-ml-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7: Draw the resulting tree (the best ML tree)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-8-comparison-of-likelihood-log-likelihood-of-trees-obtained-with-different-models-and-before-and-after-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Step 8: Comparison of likelihood (log likelihood) of trees obtained with different models and before and after filtering
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-9-generation-of-100-replicas-of-a-regular-bootstrap" class="md-nav__link">
    <span class="md-ellipsis">
      Step 9: Generation of 100 replicas of a regular bootstrap
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-10-generation-of-1000-ultrafast-bootstrap-replicas" class="md-nav__link">
    <span class="md-ellipsis">
      Step 10: Generation of 1000 ultrafast bootstrap replicas
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-11-running-the-previous-command-but-with-generation-1000-ultrafast-bootstrap-1000-alrt-abayes" class="md-nav__link">
    <span class="md-ellipsis">
      Step 11: Running the previous command, but with generation: 1000 ultrafast bootstrap + 1000 alrt + abayes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-12-drawing-the-resulting-tree-with-three-supports" class="md-nav__link">
    <span class="md-ellipsis">
      Step 12: Drawing the resulting tree with three supports
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_05_Root_Date/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_07_Visualization_Pro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_00_Overview/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Overview
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_01_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-cut-bad-areas-out-of-the-alignment-using-trimal" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Cut bad areas out of the alignment using trimAl
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-fit-an-evolution-model-in-modeltest-modeltest-ng" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: fit an evolution model in ModelTest (ModelTest-NG)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-build-an-ml-tree-in-raxml-ng-using-the-selected-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Build an ML-tree in RAxML-NG using the selected model
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-4-draw-the-resulting-tree-the-best-ml-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Draw the resulting tree (the best ML tree)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-5-model-selection-in-modelfinder-iq-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 5: Model selection in ModelFinder (IQ-TREE)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-6-build-an-ml-tree-in-iq-tree-using-the-selected-model" class="md-nav__link">
    <span class="md-ellipsis">
      Step 6: Build an ML tree in IQ-TREE using the selected model
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-7-draw-the-resulting-tree-the-best-ml-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 7: Draw the resulting tree (the best ML tree)
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-8-comparison-of-likelihood-log-likelihood-of-trees-obtained-with-different-models-and-before-and-after-filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Step 8: Comparison of likelihood (log likelihood) of trees obtained with different models and before and after filtering
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-9-generation-of-100-replicas-of-a-regular-bootstrap" class="md-nav__link">
    <span class="md-ellipsis">
      Step 9: Generation of 100 replicas of a regular bootstrap
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-10-generation-of-1000-ultrafast-bootstrap-replicas" class="md-nav__link">
    <span class="md-ellipsis">
      Step 10: Generation of 1000 ultrafast bootstrap replicas
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-11-running-the-previous-command-but-with-generation-1000-ultrafast-bootstrap-1000-alrt-abayes" class="md-nav__link">
    <span class="md-ellipsis">
      Step 11: Running the previous command, but with generation: 1000 ultrafast bootstrap + 1000 alrt + abayes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-12-drawing-the-resulting-tree-with-three-supports" class="md-nav__link">
    <span class="md-ellipsis">
      Step 12: Drawing the resulting tree with three supports
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="preparing-the-alignment-and-building-trees"><strong>Preparing the alignment and building trees</strong></h1>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p><img alt="CLT" src="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" /><br>
<img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" /></p>
<details class="question" open="open">
<summary>Question</summary>
<p>Where to get the data?</p>
</details>
<p>Download it from GitHub repository:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>wget<span class="w"> </span>https://github.com/iliapopov17/NGS-Handbook/raw/refs/heads/main/data/04_Phylogenetics/04_04_Trees.zip
</span></code></pre></div>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>unzip<span class="w"> </span>04_04_Trees.zip<span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span>rm<span class="w"> </span>-rf<span class="w"> </span>04_04_Trees.zip
</span></code></pre></div>
<p>These are the data we will be working with:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>Archive:  04_04_Trees.zip
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>   creating: data/
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>  inflating: data/SUP35_10seqs.fa
</span></code></pre></div>
<blockquote>
<p>For this work, we will use the alignment of <a href="https://www.yeastgenome.org/locus/S000002579">SUP35 gene</a> obtained by <code>prank</code> considering codons.<br></p>
</blockquote>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>prank<span class="w"> </span>-codon<span class="w"> </span>-d<span class="o">=</span>data/SUP35_10seqs.fa<span class="w"> </span>-o<span class="o">=</span>data/SUP35_aln_prank.best.fas<span class="w"> </span>-F
</span></code></pre></div>
<div style='justify-content: center'>
<img src="/images/ngs-handbook/04_Phylogenetics/04_04_Trees/tree_building_overview.png" align='center', width="100%">
</div>

<p><em>Tree building overview</em></p>
<hr />
<h2 id="step-1-cut-bad-areas-out-of-the-alignment-using-trimal"><strong>Step 1: Cut bad areas out of the alignment using <code>trimAl</code></strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>trimal<span class="w"> </span>-in<span class="w"> </span>data/SUP35_aln_prank.best.fas<span class="w"> </span>-out<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-automated1
</span></code></pre></div>
<hr />
<h2 id="step-2-fit-an-evolution-model-in-modeltest-modeltest-ng"><strong>Step 2: fit an evolution model in <code>ModelTest</code> (<code>ModelTest-NG</code>)</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>modeltest-ng<span class="w"> </span>-i<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-o<span class="w"> </span>data/modeltest/SUP35_trim_modeltest
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>                     Model         Score        Weight
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>   BIC             TIM3+G4    18180.5614        0.3950
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>   AIC           TIM3+I+G4    18041.1550        0.5377
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>  AICc           TIM3+I+G4    18041.1550        0.5377
</span></code></pre></div>
<p>In total we see that the <code>TIM3+G4</code> is recognised as the best model!</p>
<hr />
<h2 id="step-3-build-an-ml-tree-in-raxml-ng-using-the-selected-model"><strong>Step 3: Build an ML-tree in RAxML-NG using the selected model</strong></h2>
<p>First, let's check that our tree is being built at all</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>raxml-ng<span class="w"> </span>--check<span class="w"> </span>--msa<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w">  </span>--model<span class="w"> </span>TIM3+G4<span class="w"> </span>--prefix<span class="w"> </span>data/raxml/SUP35_raxml_test
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>Alignment can be successfully read by RAxML-NG.
</span></code></pre></div>
<p>Great! Let's go!</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>raxml-ng<span class="w"> </span>--msa<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>--model<span class="w"> </span>TIM3+G4<span class="w"> </span>--prefix<span class="w"> </span>data/raxml/SUP35_raxml<span class="w"> </span>--threads<span class="w"> </span><span class="m">2</span><span class="w"> </span>--seed<span class="w"> </span><span class="m">222</span><span class="w">  </span>--outgroup<span class="w"> </span>SUP35_Kla_AB039749
</span></code></pre></div>
<hr />
<h2 id="step-4-draw-the-resulting-tree-the-best-ml-tree"><strong>Step 4: Draw the resulting tree (the best ML tree)</strong></h2>
<p>For drawing trees we will use a simple prewritten R script. Let's take a look at it to understand how it works:</p>
<div class="language-R highlight"><span class="filename">draw_tree.R</span><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="c1">#!/usr/bin/env Rscript</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="n">args</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="nf">library</span><span class="p">(</span><span class="n">ggtree</span><span class="p">)</span>
</span><span id="__span-10-5"><a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a><span class="n">tr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.tree</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="c1">##SUP35_raxml.raxml.bestTree</span>
</span><span id="__span-10-6"><a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a><span class="nf">ggtree</span><span class="p">(</span><span class="n">tr</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_tiplab</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">xlim</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
</span><span id="__span-10-7"><a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a><span class="w">  </span><span class="nf">geom_treescale</span><span class="p">()</span>
</span><span id="__span-10-8"><a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a>
</span><span id="__span-10-9"><a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a><span class="nf">ggsave</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="m">2</span><span class="p">])</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>Rscript<span class="w"> </span>scripts/draw_tree.R<span class="w"> </span>data/raxml/SUP35_raxml.raxml.bestTree<span class="w"> </span>imgs/SUP35_raxml.png
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<img src="/images/ngs-handbook/04_Phylogenetics/04_04_Trees/SUP35_raxml.png" align='center', width="100%">
</div>

<hr />
<h2 id="step-5-model-selection-in-modelfinder-iq-tree"><strong>Step 5: Model selection in ModelFinder (IQ-TREE)</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>iqtree2<span class="w"> </span>-m<span class="w"> </span>MFP<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>--prefix<span class="w"> </span>data/modelfinder/SUP35_MF2
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>head<span class="w"> </span>-42<span class="w"> </span>data/modelfinder/SUP35_MF2.iqtree<span class="w"> </span><span class="p">|</span><span class="w"> </span>tail<span class="w"> </span>-6
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>Best-fit model according to BIC: TIM3+F+G4
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>List of models sorted by BIC scores: 
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>Model                  LogL         AIC      w-AIC        AICc     w-AICc         BIC      w-BIC
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a>TIM3+F+G4         -8993.686   18035.372 +   0.0517   18035.972 +   0.0549   18170.092 +    0.737
</span></code></pre></div>
<p>We see that the model <code>TIM3+F+G4</code> is recognised as the best!</p>
<details class="question" open="open">
<summary>Question</summary>
<p>Do the models selected by ModelTest and ModelFinder differ, and how much?</p>
</details>
<details class="success" open="open">
<summary>Success</summary>
<p>In general, we got the same thing. Only ModelFinder also threw in information about the empirical frequencies of the letters themselves in the alignment.</p>
<table>
<thead>
<tr>
<th></th>
<th>ModelTest</th>
<th>ModelFinder</th>
</tr>
</thead>
<tbody>
<tr>
<td>Model</td>
<td>TIM3+G4</td>
<td>TIM3+F+G4</td>
</tr>
<tr>
<td>BIC</td>
<td>18180.5614</td>
<td>18170.092</td>
</tr>
</tbody>
</table>
</details>
<hr />
<h2 id="step-6-build-an-ml-tree-in-iq-tree-using-the-selected-model"><strong>Step 6: Build an ML tree in IQ-TREE using the selected model</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>iqtree2<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>--prefix<span class="w"> </span>data/iqtree/SUP35_iqtree
</span></code></pre></div>
<hr />
<h2 id="step-7-draw-the-resulting-tree-the-best-ml-tree"><strong>Step 7: Draw the resulting tree (the best ML tree)</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a>Rscript<span class="w"> </span>draw_tree.R<span class="w"> </span>data/iqtree/SUP35_iqtree.treefile<span class="w"> </span>imgs/SUP35_iqtree.png
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<img src="/images/ngs-handbook/04_Phylogenetics/04_04_Trees/SUP35_iqtree.png" align='center', width="100%">
</div>

<p>The trees obtained with RAxML and IQTREE are fundamentally similar.<br>
They have different views on how well the outer groups diverge, and they have different topologies.</p>
<hr />
<h2 id="step-8-comparison-of-likelihood-log-likelihood-of-trees-obtained-with-different-models-and-before-and-after-filtering"><strong>Step 8: Comparison of likelihood (log likelihood) of trees obtained with different models and before and after filtering</strong></h2>
<details class="question" open="open">
<summary>Question</summary>
<p>What conclusion can be drawn from this?</p>
</details>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.best.fas<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_unfilt/SUP35_iqtree_unfilt
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>grep<span class="w"> </span><span class="s2">&quot;Log-likelihood&quot;</span><span class="w"> </span>data/iqtree_unfilt/SUP35_iqtree_unfilt.iqtree
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a>Log-likelihood of the tree: -9696.9044 (s.e. 160.3706)
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a>grep<span class="w"> </span><span class="s2">&quot;Log-likelihood&quot;</span><span class="w"> </span>data/iqtree/SUP35_iqtree.iqtree
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a>Log-likelihood of the tree: -8993.1633 (s.e. 149.1347)
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.best.fas<span class="w"> </span>-m<span class="w"> </span>JC<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_JC/SUP35_iqtree_JC
</span><span id="__span-20-2"><a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a>grep<span class="w"> </span><span class="s2">&quot;Log-likelihood&quot;</span><span class="w"> </span>data/iqtree_JC/SUP35_iqtree_JC.iqtree
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a>Log-likelihood of the tree: -10482.7253 (s.e. 176.1729)
</span></code></pre></div>
<table>
<thead>
<tr>
<th></th>
<th>Unfilt</th>
<th>TIM3+F+G4</th>
<th>JC</th>
</tr>
</thead>
<tbody>
<tr>
<td>Log-likelihood</td>
<td>-9696.9044</td>
<td>-8993.1633</td>
<td>-10482.7253</td>
</tr>
</tbody>
</table>
<ul>
<li>Before filtering Log-likelihood = -9696.9044</li>
<li>When using the <code>TIM3+F+G4</code> model = -8993.1633</li>
<li>When using <code>JC</code> model = -10482.7253<br></li>
</ul>
<details class="success" open="open">
<summary>Success</summary>
<p>That said, the topology is the same everywhere! Anyway, if we only need to look at the topology, we can run iqtree with either model...</p>
</details>
<hr />
<h2 id="step-9-generation-of-100-replicas-of-a-regular-bootstrap"><strong>Step 9: Generation of 100 replicas of a regular bootstrap</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a><span class="nb">time</span><span class="w"> </span>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-redo<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_bootstrap/SUP35_TIM3_b<span class="w"> </span>-b<span class="w"> </span><span class="m">100</span>
</span></code></pre></div>
<hr />
<h2 id="step-10-generation-of-1000-ultrafast-bootstrap-replicas"><strong>Step 10: Generation of 1000 ultrafast bootstrap replicas</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-23-1"><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="nb">time</span><span class="w"> </span>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-redo<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_ultrafast_bootstrap/SUP35_TIM3_ufb<span class="w"> </span>-bb<span class="w"> </span><span class="m">1000</span>
</span></code></pre></div>
<details class="question" open="open">
<summary>Question</summary>
<p>What is the difference between normal and ultrafast bootstrap runtimes and the values obtained?</p>
</details>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-24-1"><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="n">bootstrap_100</span> <span class="o">=</span> <span class="mi">3</span> <span class="o">*</span> <span class="mi">60</span> <span class="o">+</span> <span class="mi">17</span>
</span><span id="__span-24-2"><a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a><span class="n">ultrafast_bootstrap_1000</span> <span class="o">=</span> <span class="mf">3.258</span>
</span><span id="__span-24-3"><a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a>
</span><span id="__span-24-4"><a id="__codelineno-24-4" name="__codelineno-24-4" href="#__codelineno-24-4"></a><span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;Generation of ultrafast bootstrap is faster: </span><span class="si">{</span><span class="n">bootstrap_100</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="n">ultrafast_bootstrap_1000</span><span class="si">}</span><span class="s1"> times&#39;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-25-1"><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a>Generation of ultrafast bootstrap is faster: 60.46654389195826 times
</span></code></pre></div>
<details class="success" open="open">
<summary>Success</summary>
<p>Generating 100 replicas of regular bootstrap took 3:17.00, while generating 1000 replicas of ultrafast bootstrap took 3.258. That's a huge difference!</p>
</details>
<hr />
<h2 id="step-11-running-the-previous-command-but-with-generation-1000-ultrafast-bootstrap-1000-alrt-abayes"><strong>Step 11: Running the previous command, but with generation: 1000 ultrafast bootstrap + 1000 alrt + abayes</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-26-1"><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_ufb_alrt_abayes/SUP35_TIM3_B_alrt_abayes<span class="w"> </span>-bb<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-alrt<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-abayes
</span></code></pre></div>
<hr />
<h2 id="step-12-drawing-the-resulting-tree-with-three-supports"><strong>Step 12: Drawing the resulting tree with three supports</strong></h2>
<p>For this step we will use a bit different R script. Let's take a look at it:</p>
<div class="language-R highlight"><span class="filename">draw_tree_max.R</span><pre><span></span><code><span id="__span-27-1"><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a><span class="n">args</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-27-2"><a id="__codelineno-27-2" name="__codelineno-27-2" href="#__codelineno-27-2"></a>
</span><span id="__span-27-3"><a id="__codelineno-27-3" name="__codelineno-27-3" href="#__codelineno-27-3"></a><span class="nf">library</span><span class="p">(</span><span class="n">ggtree</span><span class="p">)</span>
</span><span id="__span-27-4"><a id="__codelineno-27-4" name="__codelineno-27-4" href="#__codelineno-27-4"></a><span class="n">tree_alrt_abayes_ufb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.tree</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="m">1</span><span class="p">])</span>
</span><span id="__span-27-5"><a id="__codelineno-27-5" name="__codelineno-27-5" href="#__codelineno-27-5"></a><span class="nf">ggtree</span><span class="p">(</span><span class="n">tree_alrt_abayes_ufb</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
</span><span id="__span-27-6"><a id="__codelineno-27-6" name="__codelineno-27-6" href="#__codelineno-27-6"></a><span class="w">  </span><span class="nf">geom_tiplab</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_nodelab</span><span class="p">()</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-27-7"><a id="__codelineno-27-7" name="__codelineno-27-7" href="#__codelineno-27-7"></a><span class="w">  </span><span class="nf">geom_treescale</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">xlim</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">)</span>
</span><span id="__span-27-8"><a id="__codelineno-27-8" name="__codelineno-27-8" href="#__codelineno-27-8"></a><span class="c1"># funny labels</span>
</span><span id="__span-27-9"><a id="__codelineno-27-9" name="__codelineno-27-9" href="#__codelineno-27-9"></a><span class="n">label</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tree_alrt_abayes_ufb</span><span class="o">$</span><span class="n">node.label</span>
</span><span id="__span-27-10"><a id="__codelineno-27-10" name="__codelineno-27-10" href="#__codelineno-27-10"></a><span class="n">alrt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">label</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">FUN</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;[&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="c1">#sun</span>
</span><span id="__span-27-11"><a id="__codelineno-27-11" name="__codelineno-27-11" href="#__codelineno-27-11"></a><span class="n">abayes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">label</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">FUN</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;[&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">))</span><span class="w"> </span><span class="c1">#yin yang</span>
</span><span id="__span-27-12"><a id="__codelineno-27-12" name="__codelineno-27-12" href="#__codelineno-27-12"></a><span class="n">ufb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="nf">sapply</span><span class="p">(</span><span class="nf">strsplit</span><span class="p">(</span><span class="n">label</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">FUN</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;[&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">))</span><span class="w"> </span><span class="c1">#star</span>
</span><span id="__span-27-13"><a id="__codelineno-27-13" name="__codelineno-27-13" href="#__codelineno-27-13"></a><span class="n">large_alrt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">alrt</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">70</span><span class="p">,</span><span class="w"> </span><span class="nf">intToUtf8</span><span class="p">(</span><span class="m">9728</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span>
</span><span id="__span-27-14"><a id="__codelineno-27-14" name="__codelineno-27-14" href="#__codelineno-27-14"></a><span class="n">large_abayes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">abayes</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0.7</span><span class="p">,</span><span class="w"> </span><span class="nf">intToUtf8</span><span class="p">(</span><span class="m">9775</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span>
</span><span id="__span-27-15"><a id="__codelineno-27-15" name="__codelineno-27-15" href="#__codelineno-27-15"></a><span class="n">large_ufb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">ufb</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">95</span><span class="p">,</span><span class="w"> </span><span class="nf">intToUtf8</span><span class="p">(</span><span class="m">9733</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">)</span>
</span><span id="__span-27-16"><a id="__codelineno-27-16" name="__codelineno-27-16" href="#__codelineno-27-16"></a><span class="n">newlabel</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">large_alrt</span><span class="p">,</span><span class="w"> </span><span class="n">large_abayes</span><span class="p">,</span><span class="w"> </span><span class="n">large_ufb</span><span class="p">)</span>
</span><span id="__span-27-17"><a id="__codelineno-27-17" name="__codelineno-27-17" href="#__codelineno-27-17"></a><span class="n">tree_alrt_abayes_ufb</span><span class="o">$</span><span class="n">node.label</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">newlabel</span>
</span><span id="__span-27-18"><a id="__codelineno-27-18" name="__codelineno-27-18" href="#__codelineno-27-18"></a><span class="nf">ggtree</span><span class="p">(</span><span class="n">tree_alrt_abayes_ufb</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
</span><span id="__span-27-19"><a id="__codelineno-27-19" name="__codelineno-27-19" href="#__codelineno-27-19"></a><span class="w">  </span><span class="nf">geom_tiplab</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_nodelab</span><span class="p">(</span><span class="n">nudge_x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-.01</span><span class="p">,</span><span class="w"> </span><span class="n">nudge_y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
</span><span id="__span-27-20"><a id="__codelineno-27-20" name="__codelineno-27-20" href="#__codelineno-27-20"></a><span class="w">  </span><span class="nf">geom_treescale</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">xlim</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">0.9</span><span class="p">)</span>
</span><span id="__span-27-21"><a id="__codelineno-27-21" name="__codelineno-27-21" href="#__codelineno-27-21"></a>
</span><span id="__span-27-22"><a id="__codelineno-27-22" name="__codelineno-27-22" href="#__codelineno-27-22"></a><span class="nf">ggsave</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="m">2</span><span class="p">])</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-28-1"><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a>Rscript<span class="w"> </span>scripts/draw_tree_max.R<span class="w"> </span>data/iqtree_ufb_alrt_abayes/SUP35_TIM3_B_alrt_abayes.treefile<span class="w"> </span>imgs/SUP35_TIM3_B_alrt_abayes.png
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<img src="/images/ngs-handbook/04_Phylogenetics/04_04_Trees/SUP35_TIM3_B_alrt_abayes.png" align='center', width="100%">
</div>

<p>All values - <code>alrt</code>, <code>abayes</code> and <code>ufb</code> are positively correlated - that is, if the indicators are high, then they are all high, as a rule. But this is not proportional.<br>
In the tree above, the indicator values are replaced with the symbols of the sun, yin-yang and asterisk for educational purposes. In real life they dont use this, but giving bare numbers through a slash is also not comme il faut...<br></p>









  




                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../04_03_MSA/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Multiple Sequence Alignment">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Multiple Sequence Alignment
              </div>
            </div>
          </a>
        
        
          
          <a href="../04_05_Root_Date/" class="md-footer__link md-footer__link--next" aria-label="Next: Rooting and comparing trees; Dating">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Rooting and comparing trees; Dating
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 Ilia Popov
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
    <a href="mailto:<iljapopov17@gmail.com>" target="_blank" rel="noopener" title="" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M498.1 5.6c10.1 7 15.4 19.1 13.5 31.2l-64 416c-1.5 9.7-7.4 18.2-16 23s-18.9 5.4-28 1.6L284 427.7l-68.5 74.1c-8.9 9.7-22.9 12.9-35.2 8.1S160 493.2 160 480v-83.6c0-4 1.5-7.8 4.2-10.8l167.6-182.8c5.8-6.3 5.6-16-.4-22s-15.7-6.4-22-.7L106 360.8l-88.3-44.2C7.1 311.3.3 300.7 0 288.9s5.9-22.8 16.1-28.7l448-256c10.7-6.1 23.9-5.5 34 1.4"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://github.com/iliapopov17" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8M97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.expand", "content.code.copy", "search", "search.suggest", "search.highlight", "navigation.footer"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.525ec568.min.js"></script>
      
    
  </body>
</html>