
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/IV.%20Phylogenetics/04_03_MSA/">
      
      
        <link rel="prev" href="../04_02_Mining_Data/">
      
      
        <link rel="next" href="../04_04_Trees/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.38">
    
    
      
        <title>Multiple Sequence Alignment - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#multiple-sequence-alignment" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Multiple Sequence Alignment
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    NGS Analysis Handbook
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../01_Quality_Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../02_Variant_Calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../03_Whole_%28pan%29genome_analyses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" checked>
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-run-6-possible-alignment-algorithms-for-10-dna-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Run 6 possible alignment algorithms for 10 DNA sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1: Run 6 possible alignment algorithms for 10 DNA sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comparative-table" class="md-nav__link">
    <span class="md-ellipsis">
      Comparative table
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#how-to-fix-the-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      How to fix the alignment
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-run-6-possible-alignment-algorithms-for-250-dna-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Run 6 possible alignment algorithms for 250 DNA sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Run 6 possible alignment algorithms for 250 DNA sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comparative-table_1" class="md-nav__link">
    <span class="md-ellipsis">
      Comparative table
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-25-how-to-get-amino-acid-sequences-from-nucleotide-sequences-translate" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.5: How to get amino acid sequences from nucleotide sequences (translate)?
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2.5: How to get amino acid sequences from nucleotide sequences (translate)?">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#option-1-transeq" class="md-nav__link">
    <span class="md-ellipsis">
      Option 1: transeq
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#option-2-getorf" class="md-nav__link">
    <span class="md-ellipsis">
      Option 2: getorf
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-run-7-possible-alignment-algorithms-for-10-protein-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Run 7 possible alignment algorithms for 10 protein sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Run 7 possible alignment algorithms for 10 protein sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comparative-table_2" class="md-nav__link">
    <span class="md-ellipsis">
      Comparative table
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice-1" class="md-nav__link">
    <span class="md-ellipsis">
      Practice 1
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice-2" class="md-nav__link">
    <span class="md-ellipsis">
      Practice 2
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Practice 2">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#option-1-muscle" class="md-nav__link">
    <span class="md-ellipsis">
      Option 1: muscle
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#option-2-mafft" class="md-nav__link">
    <span class="md-ellipsis">
      Option 2: mafft
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice-3" class="md-nav__link">
    <span class="md-ellipsis">
      Practice 3
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_04_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_05_Root_Date/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_07_Visualization_Pro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_01_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-1-run-6-possible-alignment-algorithms-for-10-dna-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Run 6 possible alignment algorithms for 10 DNA sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 1: Run 6 possible alignment algorithms for 10 DNA sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comparative-table" class="md-nav__link">
    <span class="md-ellipsis">
      Comparative table
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#how-to-fix-the-alignment" class="md-nav__link">
    <span class="md-ellipsis">
      How to fix the alignment
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-2-run-6-possible-alignment-algorithms-for-250-dna-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Run 6 possible alignment algorithms for 250 DNA sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Run 6 possible alignment algorithms for 250 DNA sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comparative-table_1" class="md-nav__link">
    <span class="md-ellipsis">
      Comparative table
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-25-how-to-get-amino-acid-sequences-from-nucleotide-sequences-translate" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.5: How to get amino acid sequences from nucleotide sequences (translate)?
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2.5: How to get amino acid sequences from nucleotide sequences (translate)?">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#option-1-transeq" class="md-nav__link">
    <span class="md-ellipsis">
      Option 1: transeq
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#option-2-getorf" class="md-nav__link">
    <span class="md-ellipsis">
      Option 2: getorf
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#step-3-run-7-possible-alignment-algorithms-for-10-protein-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Run 7 possible alignment algorithms for 10 protein sequences
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 3: Run 7 possible alignment algorithms for 10 protein sequences">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#comparative-table_2" class="md-nav__link">
    <span class="md-ellipsis">
      Comparative table
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice-1" class="md-nav__link">
    <span class="md-ellipsis">
      Practice 1
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice-2" class="md-nav__link">
    <span class="md-ellipsis">
      Practice 2
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Practice 2">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#option-1-muscle" class="md-nav__link">
    <span class="md-ellipsis">
      Option 1: muscle
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#option-2-mafft" class="md-nav__link">
    <span class="md-ellipsis">
      Option 2: mafft
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practice-3" class="md-nav__link">
    <span class="md-ellipsis">
      Practice 3
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="multiple-sequence-alignment"><strong>Multiple Sequence Alignment</strong></h1>
<blockquote>
<p>This chapter contains a manual on aligning multiple sequences at once.<br></p>
</blockquote>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p><a class="glightbox" href="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="CLT" src="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" /></a><br>
<a class="glightbox" href="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" /></a></p>
<blockquote>
<p>For this work, we will use the alignments of <a href="https://www.yeastgenome.org/locus/S000002579">SUP35 gene</a>.<br></p>
</blockquote>
<p>As we will be working with several tools we will compare them.<br>
We will compare them based on:<br></p>
<ul>
<li>Running time.<br></li>
<li>Alignment length.<br></li>
</ul>
<p>These are not the best metrics to compare the alignments. Yet for the purposes of this manual they do well.<br></p>
<hr />
<h2 id="step-1-run-6-possible-alignment-algorithms-for-10-dna-sequences"><strong>Step 1: Run 6 possible alignment algorithms for 10 DNA sequences</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:6"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><input id="__tabbed_1_3" name="__tabbed_1" type="radio" /><input id="__tabbed_1_4" name="__tabbed_1" type="radio" /><input id="__tabbed_1_5" name="__tabbed_1" type="radio" /><input id="__tabbed_1_6" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1"><code>clustalw</code></label><label for="__tabbed_1_2"><code>muscle</code></label><label for="__tabbed_1_3"><code>mafft</code></label><label for="__tabbed_1_4"><code>kalign</code></label><label for="__tabbed_1_5"><code>t_coffee</code></label><label for="__tabbed_1_6"><code>prank</code></label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="nb">time</span><span class="w"> </span>clustalw<span class="w"> </span>-INFILE<span class="o">=</span>data/SUP35_10seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="w">    </span>-OUTPUT<span class="o">=</span>FASTA<span class="w"> </span><span class="se">\</span>
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="w">    </span>-OUTFILE<span class="o">=</span>10_DNA_seqs/02_SUP35_10seqs.clustalw.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="nb">time</span><span class="w"> </span>muscle<span class="w"> </span>-align<span class="w"> </span>data/SUP35_10seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="w">    </span>-output<span class="w"> </span>10_DNA_seqs/02_SUP35_10seqs_muscle.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="nb">time</span><span class="w"> </span>mafft<span class="w"> </span>--auto<span class="w"> </span>data/SUP35_10seqs.fa<span class="w"> </span>&gt;<span class="se">\</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="w">    </span>10_DNA_seqs/02_SUP35_10seqs_mafft.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="nb">time</span><span class="w"> </span>kalign<span class="w"> </span>&lt;data/SUP35_10seqs.fa<span class="w"> </span>&gt;<span class="se">\</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="w">    </span>10_DNA_seqs/02_SUP35_10seqs_kalign.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="nb">time</span><span class="w"> </span>t_coffee<span class="w"> </span>-infile<span class="o">=</span>data/SUP35_10seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="w">    </span>-outfile<span class="o">=</span>10_DNA_seqs/02_SUP35_10seqs_tcoffee.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a><span class="w">    </span>-output<span class="o">=</span>fasta_aln
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="nb">time</span><span class="w"> </span>prank<span class="w"> </span>-d<span class="o">=</span>data/SUP35_10seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="w">    </span>-o<span class="o">=</span>10_DNA_seqs/02_SUP35_10seqs_prank.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a><span class="w">    </span>-codon
</span></code></pre></div>
</div>
</div>
</div>
<p>Now let's use a simple script to assess the alignment length.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">AlignIO</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a><span class="kn">import</span> <span class="nn">os</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="n">folder_path</span> <span class="o">=</span> <span class="s1">&#39;10_DNA_seqs&#39;</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="n">alignment_files</span> <span class="o">=</span> <span class="p">[</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">folder_path</span><span class="p">,</span> <span class="n">file</span><span class="p">)</span> <span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">folder_path</span><span class="p">)</span> <span class="k">if</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;.fa&#39;</span><span class="p">)]</span>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">alignment_files</span><span class="p">:</span>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>    <span class="n">alignment</span> <span class="o">=</span> <span class="n">AlignIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">file</span><span class="p">,</span> <span class="s1">&#39;fasta&#39;</span><span class="p">)</span>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>    <span class="n">alignment_length</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">get_alignment_length</span><span class="p">()</span>
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Alignment length in file </span><span class="si">{</span><span class="n">file</span><span class="si">}</span><span class="s2">: </span><span class="si">{</span><span class="n">alignment_length</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>Alignment length in file 10_DNA_seqs/02_SUP35_10seqs_tcoffee.fa: 2210
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>Alignment length in file 10_DNA_seqs/02_SUP35_10seqs_mafft.fa: 2166
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>Alignment length in file 10_DNA_seqs/02_SUP35_10seqs.clustalw.fa: 2148
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>Alignment length in file 10_DNA_seqs/02_SUP35_10seqs_kalign.fa: 2152
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>Alignment length in file 10_DNA_seqs/02_SUP35_10seqs_muscle.fa: 2333
</span></code></pre></div>
<h3 id="comparative-table"><strong>Comparative table</strong></h3>
<details class="question" open="open">
<summary>Question</summary>
<p>Which algorithm is better to use?</p>
</details>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Time</th>
<th>Alignment length</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>clustalw</code></td>
<td>5.641 s</td>
<td>2148</td>
</tr>
<tr>
<td><code>muscle</code></td>
<td>5.009 s</td>
<td>2333</td>
</tr>
<tr>
<td><code>mafft</code></td>
<td>4.382 s</td>
<td>2166</td>
</tr>
<tr>
<td><code>kalign</code></td>
<td>0.475 s</td>
<td>2152</td>
</tr>
<tr>
<td><code>t_coffee</code></td>
<td>20.869 s</td>
<td>2210</td>
</tr>
<tr>
<td><code>prank</code></td>
<td>&gt;10 min</td>
<td>NA</td>
</tr>
</tbody>
</table>
<details class="success" open="open">
<summary>Success</summary>
<p>It's best to use <code>muscle</code>!</p>
</details>
<h3 id="how-to-fix-the-alignment"><strong>How to fix the alignment</strong></h3>
<details class="question" open="open">
<summary>Question</summary>
<p>What's wrong with the alignment of <code>SUP35_10seqs_strange_aln.fa</code> and how to fix it?</p>
</details>
<p>Let's take a look at this alignment in <code>UGENE</code></p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_03_MSA/imgs/1.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_03_MSA/imgs/1.png" align='center', width="50%"></a>
</div>

<p>It can be seen that the sequence <code>SUP35_Spar_A12_Liti_</code> is strange. Most likely it is a reverse, i.e. it is reverse complementary.</p>
<p>Let's do a couple of youtz, youtz, youtz.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_03_MSA/imgs/2.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_03_MSA/imgs/2.png" align='center', width="50%"></a>
</div>

<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_03_MSA/imgs/3.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_03_MSA/imgs/3.png" align='center', width="50%"></a>
</div>

<p>It's beautiful!</p>
<hr />
<h2 id="step-2-run-6-possible-alignment-algorithms-for-250-dna-sequences"><strong>Step 2: Run 6 possible alignment algorithms for 250 DNA sequences</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="tabbed-set tabbed-alternate" data-tabs="2:6"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><input id="__tabbed_2_3" name="__tabbed_2" type="radio" /><input id="__tabbed_2_4" name="__tabbed_2" type="radio" /><input id="__tabbed_2_5" name="__tabbed_2" type="radio" /><input id="__tabbed_2_6" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="__tabbed_2_1"><code>clustalw</code></label><label for="__tabbed_2_2"><code>muscle</code></label><label for="__tabbed_2_3"><code>mafft</code></label><label for="__tabbed_2_4"><code>kalign</code></label><label for="__tabbed_2_5"><code>t_coffee</code></label><label for="__tabbed_2_6"><code>prank</code></label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="nb">time</span><span class="w"> </span>clustalw<span class="w"> </span>-INFILE<span class="o">=</span>data/SUP35_250seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="w">    </span>-OUTPUT<span class="o">=</span>FASTA<span class="w"> </span><span class="se">\</span>
</span><span id="__span-9-3"><a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="w">    </span>-OUTFILE<span class="o">=</span>250_DNA_seqs/05_SUP35_250seqs.clustalw.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="nb">time</span><span class="w"> </span>muscle<span class="w"> </span>-align<span class="w"> </span>data/SUP35_250seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="w">    </span>-output<span class="w"> </span>250_DNA_seqs/05_SUP35_250seqs_muscle.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="nb">time</span><span class="w"> </span>mafft<span class="w"> </span>--auto<span class="w"> </span>data/SUP35_250seqs.fa<span class="w"> </span>&gt;<span class="se">\</span>
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a><span class="w">    </span>250_DNA_seqs/05_SUP35_250seqs_mafft.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="nb">time</span><span class="w"> </span>kalign<span class="w"> </span>&lt;data/SUP35_250seqs.fa<span class="w"> </span>&gt;<span class="se">\</span>
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a><span class="w">    </span>250_DNA_seqs/05_SUP35_250seqs_kalign.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="nb">time</span><span class="w"> </span>t_coffee<span class="w"> </span>-infile<span class="o">=</span>data/SUP35_250seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="w">    </span>-outfile<span class="o">=</span>250_DNA_seqs/05_SUP35_250seqs_tcoffee.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="w">    </span>-output<span class="o">=</span>fasta_aln
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="nb">time</span><span class="w"> </span>prank<span class="w"> </span>-d<span class="o">=</span>data/SUP35_250seqs.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="w">    </span>-o<span class="o">=</span>250_DNA_seqs/05_SUP35_250seqs_prank.fa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-14-3"><a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="w">    </span>-codon
</span></code></pre></div>
</div>
</div>
</div>
<p>Now let's use a simple script to assess the alignment length.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="n">folder_path</span> <span class="o">=</span> <span class="s1">&#39;250_DNA_seqs&#39;</span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="n">alignment_files</span> <span class="o">=</span> <span class="p">[</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">folder_path</span><span class="p">,</span> <span class="n">file</span><span class="p">)</span> <span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">folder_path</span><span class="p">)</span> <span class="k">if</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;.fa&#39;</span><span class="p">)]</span>
</span><span id="__span-15-3"><a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a>
</span><span id="__span-15-4"><a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">alignment_files</span><span class="p">:</span>
</span><span id="__span-15-5"><a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a>    <span class="n">alignment</span> <span class="o">=</span> <span class="n">AlignIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">file</span><span class="p">,</span> <span class="s1">&#39;fasta&#39;</span><span class="p">)</span>
</span><span id="__span-15-6"><a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a>    <span class="n">alignment_length</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">get_alignment_length</span><span class="p">()</span>
</span><span id="__span-15-7"><a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Alignment length in file </span><span class="si">{</span><span class="n">file</span><span class="si">}</span><span class="s2">: </span><span class="si">{</span><span class="n">alignment_length</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>Alignment length in file 250_DNA_seqs/05_SUP35_250seqs.clustalw.fa: 2179
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>Alignment length in file 250_DNA_seqs/05_SUP35_250seqs_mafft.fa: 2322
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>Alignment length in file 250_DNA_seqs/05_SUP35_250seqs_muscle.fa: 2365
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a>Alignment length in file 250_DNA_seqs/05_SUP35_250seqs_kalign.fa: 2210
</span></code></pre></div>
<h3 id="comparative-table_1"><strong>Comparative table</strong></h3>
<details class="question" open="open">
<summary>Question</summary>
<p>Has our choice of algorithm changed?</p>
</details>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Time</th>
<th>Alignment length</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>clustalw</code></td>
<td>48:41.97 min</td>
<td>2179</td>
</tr>
<tr>
<td><code>muscle</code></td>
<td>30:44.42 min</td>
<td>2365</td>
</tr>
<tr>
<td><code>mafft</code></td>
<td>41.962 s</td>
<td>2322</td>
</tr>
<tr>
<td><code>kalign</code></td>
<td>7.996 s</td>
<td>2210</td>
</tr>
<tr>
<td><code>t_coffee</code></td>
<td>&gt;1 h</td>
<td>NA</td>
</tr>
<tr>
<td><code>prank</code></td>
<td>&gt;1 h</td>
<td>NA</td>
</tr>
</tbody>
</table>
<details class="succes" open="open">
<summary>Succes</summary>
<p>All the sympathies are on the side of <code>kalign</code> for the reason that it aligned 250 sequences in 8 seconds.<br>
But to be fair, <code>mafft</code> is not bad either. Its alignment is longer, and its working time is 42 seconds, not 30 or 48 minutes...<br></p>
</details>
<h2 id="step-25-how-to-get-amino-acid-sequences-from-nucleotide-sequences-translate"><strong>Step 2.5: How to get amino acid sequences from nucleotide sequences (translate)?</strong></h2>
<h3 id="option-1-transeq"><strong>Option 1: <code>transeq</code></strong></h3>
<p>The simplest and fastest variant. With its help, we "stupidly" do the translation starting from the first nucleotide and up to the last one.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a>transeq<span class="w"> </span>-sequence<span class="w"> </span>data/SUP35_10seqs.fa<span class="w"> </span>-outseq<span class="w"> </span>data/SUP35_10seqs.t.faa
</span></code></pre></div>
<h3 id="option-2-getorf"><strong>Option 2: <code>getorf</code></strong></h3>
<p><code>getorf</code> operates based on the assumption that it was given a sequence that has an open reading frame.<br>
<code>getorf</code> is not highly intelligent.<br>
Anything that starts with a methionine and ends with a stop codon is an open reading frame!<br>
It needs to tune our representation, otherwise we get a bunch of junk. Especially in fairly long sequences.<br>
But if we know how long this junk should be and we need to predict proteins quickly from our data of some Sanger sequencing, it is a very good option!<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a>getorf<span class="w"> </span>-sequence<span class="w"> </span>data/SUP35_10seqs.fa<span class="w"> </span>-outseq<span class="w"> </span>data/SUP35_10seqs.g.faa<span class="w"> </span>-noreverse<span class="w"> </span>-minsize<span class="w"> </span><span class="m">500</span>
</span></code></pre></div>
<hr />
<h2 id="step-3-run-7-possible-alignment-algorithms-for-10-protein-sequences"><strong>Step 3: Run 7 possible alignment algorithms for 10 protein sequences</strong></h2>
<p><strong><em>Input</em></strong></p>
<div class="tabbed-set tabbed-alternate" data-tabs="3:7"><input checked="checked" id="__tabbed_3_1" name="__tabbed_3" type="radio" /><input id="__tabbed_3_2" name="__tabbed_3" type="radio" /><input id="__tabbed_3_3" name="__tabbed_3" type="radio" /><input id="__tabbed_3_4" name="__tabbed_3" type="radio" /><input id="__tabbed_3_5" name="__tabbed_3" type="radio" /><input id="__tabbed_3_6" name="__tabbed_3" type="radio" /><input id="__tabbed_3_7" name="__tabbed_3" type="radio" /><div class="tabbed-labels"><label for="__tabbed_3_1"><code>clustalw</code></label><label for="__tabbed_3_2"><code>clustalo</code></label><label for="__tabbed_3_3"><code>muscle</code></label><label for="__tabbed_3_4"><code>mafft</code></label><label for="__tabbed_3_5"><code>kalign</code></label><label for="__tabbed_3_6"><code>t_coffee</code></label><label for="__tabbed_3_7"><code>prank</code></label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="nb">time</span><span class="w"> </span>clustalw<span class="w"> </span>-INFILE<span class="o">=</span>data/SUP35_10seqs.g.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-19-2"><a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="w">    </span>-OUTFILE<span class="o">=</span>10_protein_seqs/08_SUP35_10seqs.clustalw.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-19-3"><a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a><span class="w">    </span>-OUTPUT<span class="o">=</span>FASTA<span class="w"> </span><span class="se">\</span>
</span><span id="__span-19-4"><a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a><span class="w">    </span>-TYPE<span class="o">=</span>protein
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="nb">time</span><span class="w"> </span>clustalo<span class="w"> </span>--infile<span class="o">=</span>data/SUP35_10seqs.g.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-20-2"><a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="w">    </span>--outfile<span class="o">=</span>10_protein_seqs/08_SUP35_10seqs.clustalo.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-20-3"><a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a><span class="w">    </span>--verbose
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="nb">time</span><span class="w"> </span>muscle<span class="w"> </span>-align<span class="w"> </span>data/SUP35_10seqs.g.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-21-2"><a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="w">    </span>-output<span class="w"> </span>10_protein_seqs/08_SUP35_10seqs_muscle.faa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a><span class="nb">time</span><span class="w"> </span>mafft<span class="w"> </span>--auto<span class="w"> </span>data/SUP35_10seqs.g.faa<span class="w"> </span>&gt;<span class="se">\</span>
</span><span id="__span-22-2"><a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a><span class="w">    </span>10_protein_seqs/08_SUP35_250seqs_mafft.fa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-23-1"><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="nb">time</span><span class="w"> </span>kalign<span class="w"> </span>&lt;data/SUP35_10seqs.g.faa<span class="w"> </span>&gt;<span class="se">\</span>
</span><span id="__span-23-2"><a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="w">    </span>10_protein_seqs/08_SUP35_10seqs_kalign.faa
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-24-1"><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="nb">time</span><span class="w"> </span>t_coffee<span class="w"> </span>-infile<span class="o">=</span>data/SUP35_10seqs.g.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-24-2"><a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a><span class="w">    </span>-outfile<span class="o">=</span>10_protein_seqs/08_SUP35_10seqs_tcoffee.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-24-3"><a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a><span class="w">    </span>-output<span class="o">=</span>fasta_aln
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-25-1"><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="nb">time</span><span class="w"> </span>prank<span class="w"> </span>-d<span class="o">=</span>data/SUP35_10seqs.g.faa<span class="w"> </span><span class="se">\</span>
</span><span id="__span-25-2"><a id="__codelineno-25-2" name="__codelineno-25-2" href="#__codelineno-25-2"></a><span class="w">    </span>-o<span class="o">=</span>10_protein_seqs/08_SUP35_10seqs_prank.faa
</span></code></pre></div>
</div>
</div>
</div>
<p>Now let's use a simple script to assess the alignment length.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-26-1"><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a><span class="n">folder_path</span> <span class="o">=</span> <span class="s1">&#39;10_protein_seqs&#39;</span>
</span><span id="__span-26-2"><a id="__codelineno-26-2" name="__codelineno-26-2" href="#__codelineno-26-2"></a><span class="n">alignment_files</span> <span class="o">=</span> <span class="p">[</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">folder_path</span><span class="p">,</span> <span class="n">file</span><span class="p">)</span> <span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">folder_path</span><span class="p">)</span> <span class="k">if</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;.fa&#39;</span><span class="p">)</span> <span class="o">|</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;.faa&#39;</span><span class="p">)</span> <span class="o">|</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;.fas&#39;</span><span class="p">)]</span>
</span><span id="__span-26-3"><a id="__codelineno-26-3" name="__codelineno-26-3" href="#__codelineno-26-3"></a>
</span><span id="__span-26-4"><a id="__codelineno-26-4" name="__codelineno-26-4" href="#__codelineno-26-4"></a><span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">alignment_files</span><span class="p">:</span>
</span><span id="__span-26-5"><a id="__codelineno-26-5" name="__codelineno-26-5" href="#__codelineno-26-5"></a>    <span class="n">alignment</span> <span class="o">=</span> <span class="n">AlignIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">file</span><span class="p">,</span> <span class="s1">&#39;fasta&#39;</span><span class="p">)</span>
</span><span id="__span-26-6"><a id="__codelineno-26-6" name="__codelineno-26-6" href="#__codelineno-26-6"></a>    <span class="n">alignment_length</span> <span class="o">=</span> <span class="n">alignment</span><span class="o">.</span><span class="n">get_alignment_length</span><span class="p">()</span>
</span><span id="__span-26-7"><a id="__codelineno-26-7" name="__codelineno-26-7" href="#__codelineno-26-7"></a>    <span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Alignment length in file </span><span class="si">{</span><span class="n">file</span><span class="si">}</span><span class="s2">: </span><span class="si">{</span><span class="n">alignment_length</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-27-1"><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a>Alignment length in file 10_protein_seqs/08_SUP35_250seqs_mafft.fa: 759
</span><span id="__span-27-2"><a id="__codelineno-27-2" name="__codelineno-27-2" href="#__codelineno-27-2"></a>Alignment length in file 10_protein_seqs/08_SUP35_10seqs_muscle.faa: 765
</span><span id="__span-27-3"><a id="__codelineno-27-3" name="__codelineno-27-3" href="#__codelineno-27-3"></a>Alignment length in file 10_protein_seqs/08_SUP35_10seqs_kalign.faa: 721
</span><span id="__span-27-4"><a id="__codelineno-27-4" name="__codelineno-27-4" href="#__codelineno-27-4"></a>Alignment length in file 10_protein_seqs/08_SUP35_10seqs_tcoffee.faa: 752
</span><span id="__span-27-5"><a id="__codelineno-27-5" name="__codelineno-27-5" href="#__codelineno-27-5"></a>Alignment length in file 10_protein_seqs/08_SUP35_10seqs.clustalw.faa: 719
</span><span id="__span-27-6"><a id="__codelineno-27-6" name="__codelineno-27-6" href="#__codelineno-27-6"></a>Alignment length in file 10_protein_seqs/08_SUP35_10seqs.clustalo.faa: 757
</span><span id="__span-27-7"><a id="__codelineno-27-7" name="__codelineno-27-7" href="#__codelineno-27-7"></a>Alignment length in file 10_protein_seqs/08_SUP35_10seqs_prank.faa.best.fas: 776
</span></code></pre></div>
<h3 id="comparative-table_2"><strong>Comparative table</strong></h3>
<details class="question" open="open">
<summary>Question</summary>
<p>What is the best algorithm to use?</p>
</details>
<table>
<thead>
<tr>
<th>Tool</th>
<th>Time</th>
<th>Alignment length</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>clustalw</code></td>
<td>0.684 s</td>
<td>719</td>
</tr>
<tr>
<td><code>clustalo</code></td>
<td>0.742 s</td>
<td>757</td>
</tr>
<tr>
<td><code>muscle</code></td>
<td>0.582 s</td>
<td>765</td>
</tr>
<tr>
<td><code>mafft</code></td>
<td>0.754 s</td>
<td>759</td>
</tr>
<tr>
<td><code>kalign</code></td>
<td>0.062 s</td>
<td>721</td>
</tr>
<tr>
<td><code>t_coffee</code></td>
<td>2.697 s</td>
<td>752</td>
</tr>
<tr>
<td><code>prank</code></td>
<td>4:50.84 min</td>
<td>776</td>
</tr>
</tbody>
</table>
<details class="succes" open="open">
<summary>Succes</summary>
<p>This is where <code>muscle</code> is the best. It worked for less than 1 second and its length is quite respectable.</p>
</details>
<hr />
<h2 id="practice-1"><strong>Practice 1</strong></h2>
<details class="question" open="open">
<summary>Question</summary>
<p>How to add 2 more nucleotide sequences to an alignment of 250 nucleotide sequences, previously aligning them, with <code>mafft</code>?</p>
</details>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-28-1"><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a>mafft<span class="w"> </span>--auto<span class="w"> </span>data/SUP35_2addseqs.fa<span class="w"> </span>&gt;<span class="w"> </span>252_DNA_seqs/10_SUP35_2addseqs_mafft.fa
</span><span id="__span-28-2"><a id="__codelineno-28-2" name="__codelineno-28-2" href="#__codelineno-28-2"></a>mafft<span class="w"> </span>--add<span class="w"> </span>252_DNA_seqs/10_SUP35_2addseqs_mafft.fa<span class="w"> </span>250_DNA_seqs/05_SUP35_250seqs_mafft.fa<span class="w"> </span>&gt;<span class="w"> </span>252_DNA_seqs/10_SUP35_252seqs_mafft.fa
</span></code></pre></div>
<hr />
<h2 id="practice-2"><strong>Practice 2</strong></h2>
<details class="question" open="open">
<summary>Question</summary>
<p>Extract from NCBI all sequences for the query "Parapallasea 18S" (Parapallasea is a taxon and 18S is a gene) and save to the file fasta</p>
</details>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-29-1"><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a>esearch<span class="w"> </span>-db<span class="w"> </span>nucleotide<span class="w"> </span>-query<span class="w"> </span><span class="s2">&quot;Parapallasea 18S&quot;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>efetch<span class="w"> </span>-format<span class="w"> </span>fasta<span class="w"> </span>&gt;data/Parapallasea_18.fa
</span></code></pre></div>
<h3 id="option-1-muscle"><strong>Option 1: <code>muscle</code></strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-30-1"><a id="__codelineno-30-1" name="__codelineno-30-1" href="#__codelineno-30-1"></a>muscle<span class="w"> </span>-align<span class="w"> </span>data/Parapallasea_18.fa<span class="w"> </span>-output<span class="w"> </span>data/Parapallasea_18.fa.muscle.aln
</span></code></pre></div>
<h3 id="option-2-mafft"><strong>Option 2: <code>mafft</code></strong></h3>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-31-1"><a id="__codelineno-31-1" name="__codelineno-31-1" href="#__codelineno-31-1"></a>mafft<span class="w"> </span>--auto<span class="w"> </span>data/Parapallasea_18.fa<span class="w"> </span>&gt;<span class="w"> </span>data/Parapallasea_18.fa.mafft.aln
</span></code></pre></div>
<hr />
<h2 id="practice-3"><strong>Practice 3</strong></h2>
<details class="question" open="open">
<summary>Question</summary>
<p>Create a blast database from a set of <code>Ommatogammarus_flavus_transcriptome_assembly.fa</code> sequences, and search this database for the protein sequence <code>Acanthogammarus_victorii_COI.faa</code> and record the results in a table (tab-separated text)</p>
</details>
<details class="warning" open="open">
<summary>Warning</summary>
<p>Attention: the origin of the sequence is mitochondrial. What is important to consider when searching?</p>
</details>
<p>Extract the sequence with the best match into a separate file.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-32-1"><a id="__codelineno-32-1" name="__codelineno-32-1" href="#__codelineno-32-1"></a>makeblastdb<span class="w"> </span>-in<span class="w"> </span>data/Ommatogammarus_flavus_transcriptome_assembly.fa<span class="w"> </span>-dbtype<span class="w"> </span>nucl<span class="w"> </span>-parse_seqids
</span></code></pre></div>
<p>The gene is mitochondrial.<br>
Accordingly, the genetic code is different, and since here we are dealing with the communication between the protein query and the nucleotide base, it may matter. Not catastrophic here. But it is better to use the <code>-db_gencode 5</code> option, because this way the identity will be higher.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-33-1"><a id="__codelineno-33-1" name="__codelineno-33-1" href="#__codelineno-33-1"></a>tblastn<span class="w"> </span>-query<span class="w"> </span>data/Acanthogammarus_victorii_COI.faa<span class="w"> </span>-db<span class="w"> </span>data/Ommatogammarus_flavus_transcriptome_assembly.fa<span class="w"> </span>-outfmt<span class="w"> </span><span class="m">6</span><span class="w"> </span>-db_gencode<span class="w"> </span><span class="m">5</span>
</span><span id="__span-33-2"><a id="__codelineno-33-2" name="__codelineno-33-2" href="#__codelineno-33-2"></a><span class="c1">## fields: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-34-1"><a id="__codelineno-34-1" name="__codelineno-34-1" href="#__codelineno-34-1"></a>Acanthogammarus_victorii_COI    TRINITY_DN8878_c0_g1_i2 89.621  501 52  0   9   509 3   1505    0.0 781
</span><span id="__span-34-2"><a id="__codelineno-34-2" name="__codelineno-34-2" href="#__codelineno-34-2"></a>Acanthogammarus_victorii_COI    TRINITY_DN58613_c0_g1_i1    50.000  20  9   1   206 225 32  88  6.3 20.8
</span></code></pre></div>
<p>Percentage of identity is <strong>89.621</strong>! Yay!</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-35-1"><a id="__codelineno-35-1" name="__codelineno-35-1" href="#__codelineno-35-1"></a>blastdbcmd<span class="w"> </span>-db<span class="w"> </span>data/Ommatogammarus_flavus_transcriptome_assembly.fa<span class="w"> </span>-entry<span class="w"> </span>TRINITY_DN8878_c0_g1_i2<span class="w"> </span>-out<span class="w"> </span>data/Ommatogammarus_flavus_COI.fa
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-36-1"><a id="__codelineno-36-1" name="__codelineno-36-1" href="#__codelineno-36-1"></a>cat<span class="w"> </span>data/Ommatogammarus_flavus_COI.fa
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-37-1"><a id="__codelineno-37-1" name="__codelineno-37-1" href="#__codelineno-37-1"></a>&gt;TRINITY_DN8878_c0_g1_i2 len=1505
</span><span id="__span-37-2"><a id="__codelineno-37-2" name="__codelineno-37-2" href="#__codelineno-37-2"></a>CGACCAACCACAAAGATATTGGCACTCTTTATTTTATGCTAGGGCTCTGGTCTGGGTTAGTCGGAACCTCCATAAGACTT
</span><span id="__span-37-3"><a id="__codelineno-37-3" name="__codelineno-37-3" href="#__codelineno-37-3"></a>ATCCTCCGCTCAGAACTTAGTGCGCCGGGTAGCCTGATTGGTGATGATCAACTGTATAACGTAATGGTAACCTCCCATGC
</span><span id="__span-37-4"><a id="__codelineno-37-4" name="__codelineno-37-4" href="#__codelineno-37-4"></a>TTTTATTATAATTTTTTTTATAGTTATGCCTATCATAATTGGCGGGTTTGGTAACTGGCTGCTTCCTTTAATACTAGGTA
</span><span id="__span-37-5"><a id="__codelineno-37-5" name="__codelineno-37-5" href="#__codelineno-37-5"></a>GACCTGATATAGCCTTCCCTCGAATAAACAACATGAGCTTTTGACTACTACCTCCTTCCCTTACACTTCTTATATCTAGA
</span><span id="__span-37-6"><a id="__codelineno-37-6" name="__codelineno-37-6" href="#__codelineno-37-6"></a>AGCTTAGTAGAAAGAGGAGTCGGCACAGGTTGAACTGTCTACCCTCCTTTATCTGGGTCTACAGCCCATAGAGGTAGCGC
</span><span id="__span-37-7"><a id="__codelineno-37-7" name="__codelineno-37-7" href="#__codelineno-37-7"></a>TGTAGATTTGGCTATTTTCTCACTTCATTTAGCCGGAGCTTCCTCTATCTTAGGGGCTGTAAATTTTATTTCTACCGCCA
</span><span id="__span-37-8"><a id="__codelineno-37-8" name="__codelineno-37-8" href="#__codelineno-37-8"></a>TTAATATGCGAGCGCCTGGGATAAAATTAGACCAAATGCCTTTATTCGTCTGAGCTATTATTATTACTACCGTCCTCCTA
</span><span id="__span-37-9"><a id="__codelineno-37-9" name="__codelineno-37-9" href="#__codelineno-37-9"></a>GTCTTATCCCTACCAGTCCTAGCTGGGGCCATTACGATACTACTTACAGACCGTAACATAAATACCTCTTTTTTTGACCC
</span><span id="__span-37-10"><a id="__codelineno-37-10" name="__codelineno-37-10" href="#__codelineno-37-10"></a>TAGTGGGGGGGGTGACCCTATCCTATACCAACACTTATTTTGATTTTTTGGGCACCCAGAGGTGTATATTTTAATCCTGC
</span><span id="__span-37-11"><a id="__codelineno-37-11" name="__codelineno-37-11" href="#__codelineno-37-11"></a>CTGCATTTGGCATAATCTCTCATATTGTTAGACAGGAGTCCGGTAAAAAAGAAACATTTGGCCCCCTAGGGATAATTTAT
</span><span id="__span-37-12"><a id="__codelineno-37-12" name="__codelineno-37-12" href="#__codelineno-37-12"></a>GCTATATTAGCTATTGGGTTCCTCGGATTTATTGTGTGAGCCCATCATATGTTTACAGTCGGTATGGATGTAGATACCCG
</span><span id="__span-37-13"><a id="__codelineno-37-13" name="__codelineno-37-13" href="#__codelineno-37-13"></a>AGCCTATTTTACATCAGCTACAATAATTATTGCAGTCCCCACCGGCATCAAAGTATTTAGGTGACTAGGTACTCTACAAG
</span><span id="__span-37-14"><a id="__codelineno-37-14" name="__codelineno-37-14" href="#__codelineno-37-14"></a>GCGGAAAAATTAACTTTTCTCCAGCTCTAATTTGAAGACTAGGTTTTATTTTCCTTTTCTCTATTGGAGGTTTAACTGGA
</span><span id="__span-37-15"><a id="__codelineno-37-15" name="__codelineno-37-15" href="#__codelineno-37-15"></a>GTTATATTAGCTAACTCATCAATTGACATCGTACTACACGACACTTACTATGTAGTTGCCCACTTTCATTATGTTTTATC
</span><span id="__span-37-16"><a id="__codelineno-37-16" name="__codelineno-37-16" href="#__codelineno-37-16"></a>TATGGGGGCTGTTTTCGGTATTTTTGCGGGGTTTGCTTACTGGTTTCCACTATTTACAGGTATAACTATCAATCCTATCC
</span><span id="__span-37-17"><a id="__codelineno-37-17" name="__codelineno-37-17" href="#__codelineno-37-17"></a>TAGCTAAAATTCATTTTTACGTCATATTCATGGGAGTAAACTTAACTTTTTTCCCCCAACATTTCCTTGGTTTAACGGGC
</span><span id="__span-37-18"><a id="__codelineno-37-18" name="__codelineno-37-18" href="#__codelineno-37-18"></a>ATACCTCGGCGATACTCAGACTATCCTGACTTCTTCACAGCCTGAAATATTGTTTCCTCCTTAGGCTCTTATATCTCTGT
</span><span id="__span-37-19"><a id="__codelineno-37-19" name="__codelineno-37-19" href="#__codelineno-37-19"></a>TTTAGCTATAGTGATCTTTATTGCTATAATCATAGAAGCTTTTATCTCTAAGCGGTCCGCTTTATTTTCCTTAACCTTGT
</span><span id="__span-37-20"><a id="__codelineno-37-20" name="__codelineno-37-20" href="#__codelineno-37-20"></a>CGTCTGCTTTAGAGTGGTACCACTCATACCCGCCAGCCGACCATAGCTACAACGATACCCCTATT
</span></code></pre></div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 Ilia Popov
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.expand", "content.code.copy", "search", "search.suggest", "search.highlight"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>