
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://iliapopov17.github.io/NGS-Handbook/IV.%20Phylogenetics/04_05_Root_Date/">
      
      
        <link rel="prev" href="../04_04_Trees/">
      
      
        <link rel="next" href="../04_06_Phylogenomics/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.38">
    
    
      
        <title>Rooting and comparing trees; Dating - NGS Analysis Handbook</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#rooting-and-comparing-trees-dating" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="NGS Analysis Handbook" class="md-header__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NGS Analysis Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Rooting and comparing trees; Dating
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NGS Analysis Handbook" class="md-nav__button md-logo" aria-label="NGS Analysis Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    NGS Analysis Handbook
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome to MkDocs
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../00_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../01_Quality_Control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    I. Quality Control of Raw Data
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../02_Variant_Calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    II. Genomic Variation Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../03_Whole_%28pan%29genome_analyses/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    III. Whole Genome and Pangenome Analyses
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IV. Phylogenetics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            IV. Phylogenetics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_01_Intro_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Drawing Trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_02_Mining_Data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Working with NCBI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_03_MSA/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Multiple Sequence Alignment
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_04_Trees/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Preparing the alignment and building trees
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Rooting and comparing trees; Dating
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-1-rooting-trees" class="md-nav__link">
    <span class="md-ellipsis">
      Part 1: Rooting trees
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Part 1: Rooting trees">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-rooting-a-tree-by-a-known-external-clade-in-iq-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Rooting a tree by a known external clade in IQ-TREE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-rooting-a-tree-using-midpoint-rooting" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Rooting a tree using midpoint rooting
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Rooting a tree using midpoint rooting">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-21-visualisation-of-the-rooted-trees" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.1: Visualisation of the rooted trees
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-rooting-a-tree-using-an-irreversible-non-reversible-model-iq-tree2" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Rooting a tree using an irreversible (non-reversible) model (iq-tree2)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-root-supported-tree-visualisation-rootstrap" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Root-supported tree visualisation (rootstrap)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-2-dating" class="md-nav__link">
    <span class="md-ellipsis">
      Part 2: Dating
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_06_Phylogenomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Phylogenomics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_07_Visualization_Pro/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Visualization Pro
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04_08_Tanglegram/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tanglegram
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    V. 16S amplicon analysis
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            V. 16S amplicon analysis
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_01_Introduction/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    16S analysis with DADA2
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../V.%2016S_amplicon_analysis/05_02_Crohns_disease/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Crohn's disease
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#instruction" class="md-nav__link">
    <span class="md-ellipsis">
      Instruction
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-1-rooting-trees" class="md-nav__link">
    <span class="md-ellipsis">
      Part 1: Rooting trees
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Part 1: Rooting trees">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-rooting-a-tree-by-a-known-external-clade-in-iq-tree" class="md-nav__link">
    <span class="md-ellipsis">
      Step 1: Rooting a tree by a known external clade in IQ-TREE
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-rooting-a-tree-using-midpoint-rooting" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2: Rooting a tree using midpoint rooting
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Step 2: Rooting a tree using midpoint rooting">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-21-visualisation-of-the-rooted-trees" class="md-nav__link">
    <span class="md-ellipsis">
      Step 2.1: Visualisation of the rooted trees
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-rooting-a-tree-using-an-irreversible-non-reversible-model-iq-tree2" class="md-nav__link">
    <span class="md-ellipsis">
      Step 3: Rooting a tree using an irreversible (non-reversible) model (iq-tree2)
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-4-root-supported-tree-visualisation-rootstrap" class="md-nav__link">
    <span class="md-ellipsis">
      Step 4: Root-supported tree visualisation (rootstrap)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-2-dating" class="md-nav__link">
    <span class="md-ellipsis">
      Part 2: Dating
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="rooting-and-comparing-trees-dating"><strong>Rooting and comparing trees; Dating</strong></h1>
<h2 id="instruction"><strong>Instruction</strong></h2>
<p><a class="glightbox" href="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="CLT" src="https://img.shields.io/badge/Language-Command Line Tools &amp; Python-steelblue" /></a><br>
<a class="glightbox" href="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img alt="IDE" src="https://img.shields.io/badge/Recommended IDE-Jupyter Notebook-steelblue" /></a></p>
<blockquote>
<p>For this work, we will use a filtered alignment (this is the same one we got in the <a href="https://github.com/iliapopov17/NGS-Handbook/tree/main/04_Phylogenetics/04_04_Trees">Trees step</a>)</p>
</blockquote>
<hr />
<h2 id="part-1-rooting-trees"><strong>Part 1: Rooting trees</strong></h2>
<h3 id="step-1-rooting-a-tree-by-a-known-external-clade-in-iq-tree"><strong>Step 1: Rooting a tree by a known external clade in <code>IQ-TREE</code></strong></h3>
<p>Our goal in this part of the manual is to compare different rooting methods.<br>
That's why we shall start by making different trees.<br>
First, let's make a 1000 ufb tree.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_ufb/SUP35_TIM3_ufb<span class="w"> </span>-bb<span class="w"> </span><span class="m">1000</span>
</span></code></pre></div>
<p>Then let's make a <code>-bb 1000 -alrt 1000 -abayes</code> tree.<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_ufb_alrt_abayes/SUP35_TIM3_ufb_alrt_abayes<span class="w"> </span>-bb<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-alrt<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-abayes
</span></code></pre></div>
<p>Finally, let's make a tree rooted by the outgroup!<br></p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_root_outgroup/SUP35_TIM3_root_outgroup<span class="w"> </span>-bb<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-alrt<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-abayes<span class="w">  </span>-o<span class="w"> </span>SUP35_Kla_AB039749,SUP35_Agos_ATCC_10895_NM_211584
</span></code></pre></div>
<hr />
<h3 id="step-2-rooting-a-tree-using-midpoint-rooting"><strong>Step 2: Rooting a tree using <code>midpoint rooting</code></strong></h3>
<p>There are two ways to root the tree at the midpoint: by using Python or R.<br>
Below we will use prewritten scripts.<br>
Let's take a look at them:</p>
<div class="tabbed-set tabbed-alternate" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><div class="tabbed-labels"><label for="__tabbed_1_1">Python</label><label for="__tabbed_1_2">R</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-py highlight"><span class="filename">midpoint_root.py</span><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="kn">import</span> <span class="nn">sys</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="kn">from</span> <span class="nn">ete3</span> <span class="kn">import</span> <span class="n">Tree</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="n">intre</span> <span class="o">=</span> <span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="n">tre</span> <span class="o">=</span> <span class="n">Tree</span><span class="p">(</span><span class="n">intre</span><span class="p">,</span> <span class="n">quoted_node_names</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="n">midpoint</span> <span class="o">=</span> <span class="n">tre</span><span class="o">.</span><span class="n">get_midpoint_outgroup</span><span class="p">()</span>
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a><span class="n">tre</span><span class="o">.</span><span class="n">set_outgroup</span><span class="p">(</span><span class="n">midpoint</span><span class="p">)</span>
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="nb">print</span><span class="p">(</span><span class="n">tre</span><span class="o">.</span><span class="n">write</span><span class="p">())</span>
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-R highlight"><span class="filename">midpoint_root.R</span><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1">#install.packages(&quot;phytools&quot;)</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="nf">library</span><span class="p">(</span><span class="n">phytools</span><span class="p">)</span>
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a><span class="n">tree_alrt_abayes_ufb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.tree</span><span class="p">(</span><span class="s">&quot;iqtree_ufb_alrt_abayes/SUP35_TIM3_ufb_alrt_abayes.treefile&quot;</span><span class="p">)</span>
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="nf">midpoint.root</span><span class="p">(</span><span class="n">tree_alrt_abayes_ufb</span><span class="p">)</span>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a><span class="nf">write.tree</span><span class="p">(</span><span class="n">tree_alrt_abayes_ufb</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;iqtree_ufb_alrt_abayes/SUP35_TIM3_ufb_alrt_abayes_rooted.treefile&quot;</span><span class="p">)</span>
</span></code></pre></div>
</div>
</div>
</div>
<p><strong><em>Input</em></strong></p>
<div class="tabbed-set tabbed-alternate" data-tabs="2:2"><input checked="checked" id="__tabbed_2_1" name="__tabbed_2" type="radio" /><input id="__tabbed_2_2" name="__tabbed_2" type="radio" /><div class="tabbed-labels"><label for="__tabbed_2_1">Python script</label><label for="__tabbed_2_2">R script</label></div>
<div class="tabbed-content">
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>python3<span class="w"> </span>scripts/midpoint_root.py<span class="w"> </span>data/iqtree_ufb/SUP35_TIM3_ufb.treefile<span class="w"> </span>&gt;data/iqtree_ufb/SUP35_TIM3_ufb_midpoint.treefile
</span></code></pre></div>
</div>
<div class="tabbed-block">
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>Rscript<span class="w"> </span>scripts/midpoint_root.R
</span></code></pre></div>
</div>
</div>
</div>
<h4 id="step-21-visualisation-of-the-rooted-trees"><strong>Step 2.1: Visualisation of the rooted trees</strong></h4>
<p>For drawing trees we will use a simple prewritten R script. Let's take a look at it to understand how it works:</p>
<div class="language-R highlight"><span class="filename">draw_tree.R</span><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1">#!/usr/bin/env Rscript</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="n">args</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">commandArgs</span><span class="p">(</span><span class="n">trailingOnly</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="nf">library</span><span class="p">(</span><span class="n">ggtree</span><span class="p">)</span>
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a><span class="n">tr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.tree</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="c1">##SUP35_raxml.raxml.bestTree</span>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="nf">ggtree</span><span class="p">(</span><span class="n">tr</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">geom_tiplab</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="nf">xlim</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> </span>
</span><span id="__span-7-7"><a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="w">  </span><span class="nf">geom_treescale</span><span class="p">()</span>
</span><span id="__span-7-8"><a id="__codelineno-7-8" name="__codelineno-7-8" href="#__codelineno-7-8"></a>
</span><span id="__span-7-9"><a id="__codelineno-7-9" name="__codelineno-7-9" href="#__codelineno-7-9"></a><span class="nf">ggsave</span><span class="p">(</span><span class="n">args</span><span class="p">[</span><span class="m">2</span><span class="p">])</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>Rscript<span class="w"> </span>scripts/draw_tree.R<span class="w"> </span>data/iqtree_ufb/SUP35_TIM3_ufb.treefile<span class="w"> </span>imgs/SUP35_TIM3_ufb.png
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>Rscript<span class="w"> </span>scripts/draw_tree.R<span class="w"> </span>data/iqtree_ufb/SUP35_TIM3_ufb_midpoint.treefile<span class="w"> </span>imgs/SUP35_TIM3_ufb_midpoint.png
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>Rscript<span class="w"> </span>scripts/draw_tree.R<span class="w"> </span>data/iqtree_root_outgroup/SUP35_TIM3_root_outgroup.treefile<span class="w"> </span>imgs/SUP35_TIM3_root_outgroup.png
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>Rscript<span class="w"> </span>scripts/draw_tree.R<span class="w"> </span>data/iqtree_ufb_alrt_abayes/SUP35_TIM3_ufb_alrt_abayes_rooted.treefile<span class="w"> </span>imgs/SUP35_TIM3_ufb_alrt_abayes_rooted.png
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<table>
<thead>
<tr>
<th>SUP35_TIM3_ufb.png</th>
<th>SUP35_TIM3_ufb_midpoint.png</th>
<th>SUP35_TIM3_root_outgroup.png</th>
<th>SUP35_TIM3_ufb_alrt_abayes_rooted.png</th>
</tr>
</thead>
<tbody>
<tr>
<td>test</td>
<td>test</td>
<td>test</td>
<td>test</td>
</tr>
</tbody>
</table>
<ul>
<li>Unrooted and rooted by external group are completely identical. 0 differences.<br></li>
<li>Rooted by <code>midpoint</code> looks neater. Topology looks better.<br></li>
</ul>
<hr />
<h3 id="step-3-rooting-a-tree-using-an-irreversible-non-reversible-model-iq-tree2"><strong>Step 3: Rooting a tree using an irreversible (<code>non-reversible</code>) model (<code>iq-tree2</code>)</strong></h3>
<p>If we have a rather complex tree structure (it is huge, there are long branches, imbalance in sampling by different taxa), rooting the tree by external group will not give us the result we expect.<br>
There are more intelligent models for this.<br>
One of them is the easy-to-apply <code>non-reversible</code> model <code>iq-tree2</code>.<br>
The idea is that they allow you to predict where the root was! This, by analogy with <code>bootstrap</code> is called <code>rootstrap</code>.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/SUP35_aln_prank.trim.fas<span class="w"> </span>-m<span class="w"> </span>TIM3+F+G4<span class="w"> </span>-pre<span class="w"> </span>data/iqtree_root_auto/SUP35_TIM3_root_auto<span class="w"> </span>--model-joint<span class="w"> </span><span class="m">12</span>.12<span class="w"> </span>-B<span class="w"> </span><span class="m">1000</span>
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="c1"># -B 1000 -  it&#39;s not `bootstrap`, it&#39;s how many times to run `rootstrap`</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>cat<span class="w"> </span>data/iqtree_root_auto/SUP35_TIM3_root_auto.rootstrap.nex
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="c1"># - contains information about the algorithm&#39;s confidence in where the root is located</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>#NEXUS
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>[ This file is best viewed in FigTree. ]
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>begin trees;
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>  tree tree_1 = ((SUP35_Kla_AB039749:0.2581582648[&amp;id=&quot;2&quot;,rootstrap=&quot;26.8&quot;],SUP35_Agos_ATCC_10895_NM_211584:0.3420323394[&amp;id=&quot;3&quot;,rootstrap=&quot;5.4&quot;]):0.1209998432[&amp;id=&quot;1&quot;,rootstrap=&quot;42.4&quot;],(((((((SUP35_Scer_74-D694_GCA_001578265.1:0.0004800339[&amp;id=&quot;11&quot;,rootstrap=&quot;0&quot;],SUP35_Scer_beer078_CM005938:0.0000010000[&amp;id=&quot;12&quot;,rootstrap=&quot;0&quot;]):0.0000010000[&amp;id=&quot;10&quot;,rootstrap=&quot;0&quot;],SUP35_Sbou_unique28_CM003560:0.0004800702[&amp;id=&quot;13&quot;,rootstrap=&quot;0&quot;]):0.0463459057[&amp;id=&quot;9&quot;,rootstrap=&quot;0&quot;],SUP35_Spar_A12_Liti:0.0325384431[&amp;id=&quot;14&quot;,rootstrap=&quot;0.1&quot;]):0.0354767121[&amp;id=&quot;8&quot;,rootstrap=&quot;0.2&quot;],SUP35_Smik_IFO1815T_30:0.0736998639[&amp;id=&quot;15&quot;,rootstrap=&quot;0.6&quot;]):0.0322607827[&amp;id=&quot;7&quot;,rootstrap=&quot;0.5&quot;],SUP35_Skud_IFO1802T_36:0.0970836557[&amp;id=&quot;16&quot;,rootstrap=&quot;0.7&quot;]):0.0154599513[&amp;id=&quot;6&quot;,rootstrap=&quot;1.8&quot;],SUP35_Sarb_H-6_chrXIII_CM001575:0.0787155739[&amp;id=&quot;17&quot;,rootstrap=&quot;4.8&quot;]):0.0099429593[&amp;id=&quot;5&quot;,rootstrap=&quot;8.6&quot;],SUP35_Seub_CBS12357_chr_II_IV_DF968535:0.0912344001[&amp;id=&quot;18&quot;,rootstrap=&quot;5.1&quot;]):0.1942253516[&amp;id=&quot;4&quot;,rootstrap=&quot;42.4&quot;]):0.0000010000[&amp;id=&quot;0&quot;,rootstrap=&quot;42.4&quot;];
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>end;
</span></code></pre></div>
<p>This is basically a <code>Newick</code> file, but strange <code>Newick</code>  because it has square brackets in it.<br>
Programs that read <code>Newick</code> format will not be able to read this tree. According to the developers of <code>iqtree</code> it is better to read this tree in <code>FigTree</code>.<br></p>
<hr />
<h3 id="step-4-root-supported-tree-visualisation-rootstrap"><strong>Step 4: Root-supported tree visualisation (<code>rootstrap</code>)</strong></h3>
<details class="question" open="open">
<summary>Question</summary>
<p>What can we say about the algorithm's confidence in root selection?</p>
</details>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>figtree<span class="w"> </span>data/iqtree_root_auto/SUP35_TIM3_root_auto.rootstrap.nex
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-16%20at%2000.24.10.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-16%20at%2000.24.10.png" align='center', width="50%"></a>
</div>

<details class="success" open="open">
<summary>Success</summary>
<p>It can't say anything specific about where the tree splits. There's a 42.4% chance the root is either in one place or the other.<br></p>
</details>
<hr />
<h2 id="part-2-dating"><strong>Part 2: Dating</strong></h2>
<details class="abstract" open="open">
<summary>Abstract</summary>
<p>Analyze the age of the common ancestor of the two species of smoky leopards from the article https://doi.org/10.1016/j.cub.2006.08.066 based on sequencing data of the <code>atp8</code> gene region, relying on known data on the frequency of substitutions in mtDNA (approximately 2% per million years) in <code>beauti</code> and <code>beast</code><br></p>
<ul>
<li>Check the quality in <code>Tracer</code><br></li>
<li>Combine trees in <code>treeannotator</code>.<br></li>
<li>Draw the final tree.<br></li>
<li>Be sure to show estimates of the age of the common ancestor at the nodes!<br></li>
</ul>
</details>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>efetch<span class="w"> </span>-db<span class="w"> </span>popset<span class="w"> </span>-id<span class="w"> </span><span class="m">126256179</span><span class="w"> </span>-format<span class="w"> </span>fasta<span class="w"> </span>&gt;data/atp8/felidae_atp8.fa
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>cut<span class="w"> </span>-d<span class="w"> </span><span class="s1">&#39; &#39;</span><span class="w"> </span>-f<span class="w"> </span><span class="m">1</span>,2,3<span class="w"> </span>data/atp8/felidae_atp8.fa<span class="w"> </span><span class="p">|</span><span class="w"> </span>sed<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;s/ /_/g&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>data/atp8/felidae_atp8.renamed.fa
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a>mafft<span class="w"> </span>--auto<span class="w"> </span>data/atp8/felidae_atp8.renamed.fa<span class="w"> </span>&gt;data/atp8/felidae_atp8.aln
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>trimal<span class="w"> </span>-in<span class="w"> </span>data/atp8/felidae_atp8.aln<span class="w"> </span>-out<span class="w"> </span>data/atp8/felidae_atp8.trim.fas<span class="w"> </span>-nogaps
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a>iqtree2<span class="w"> </span>-s<span class="w"> </span>data/atp8/felidae_atp8.trim.fas<span class="w"> </span>-o<span class="w"> </span>EF437591.1_Felis_catus<span class="w"> </span>-alrt<span class="w"> </span><span class="m">1000</span><span class="w"> </span>-abayes
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">Phylo</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="n">tree</span> <span class="o">=</span> <span class="n">Phylo</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;data/atp8/felidae_atp8.trim.fas.treefile&quot;</span><span class="p">,</span> <span class="s2">&quot;newick&quot;</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Input</em></strong></p>
<div class="language-python highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="n">Phylo</span><span class="o">.</span><span class="n">draw_ascii</span><span class="p">(</span><span class="n">tree</span><span class="p">)</span>
</span></code></pre></div>
<p><strong><em>Output</em></strong></p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a>                                          , EF437567.1_Neofelis_nebulosa
</span><span id="__span-21-2"><a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a>                                          |
</span><span id="__span-21-3"><a id="__codelineno-21-3" name="__codelineno-21-3" href="#__codelineno-21-3"></a>                                          | EF437569.1_Neofelis_nebulosa
</span><span id="__span-21-4"><a id="__codelineno-21-4" name="__codelineno-21-4" href="#__codelineno-21-4"></a>                                          |
</span><span id="__span-21-5"><a id="__codelineno-21-5" name="__codelineno-21-5" href="#__codelineno-21-5"></a>                                          | EF437570.1_Neofelis_nebulosa
</span><span id="__span-21-6"><a id="__codelineno-21-6" name="__codelineno-21-6" href="#__codelineno-21-6"></a>                                          |
</span><span id="__span-21-7"><a id="__codelineno-21-7" name="__codelineno-21-7" href="#__codelineno-21-7"></a>                                   _______| EF437568.1_Neofelis_nebulosa
</span><span id="__span-21-8"><a id="__codelineno-21-8" name="__codelineno-21-8" href="#__codelineno-21-8"></a>                                  |       |
</span><span id="__span-21-9"><a id="__codelineno-21-9" name="__codelineno-21-9" href="#__codelineno-21-9"></a>  ________________________________|       |_ EF437571.1_Neofelis_nebulosa
</span><span id="__span-21-10"><a id="__codelineno-21-10" name="__codelineno-21-10" href="#__codelineno-21-10"></a> |                                |
</span><span id="__span-21-11"><a id="__codelineno-21-11" name="__codelineno-21-11" href="#__codelineno-21-11"></a> |                                |            , EF437572.1_Neofelis_diardi
</span><span id="__span-21-12"><a id="__codelineno-21-12" name="__codelineno-21-12" href="#__codelineno-21-12"></a> |                                |____________|
</span><span id="__span-21-13"><a id="__codelineno-21-13" name="__codelineno-21-13" href="#__codelineno-21-13"></a> |                                             | EF437573.1_Neofelis_diardi
</span><span id="__span-21-14"><a id="__codelineno-21-14" name="__codelineno-21-14" href="#__codelineno-21-14"></a> |
</span><span id="__span-21-15"><a id="__codelineno-21-15" name="__codelineno-21-15" href="#__codelineno-21-15"></a> |                             __ EF437581.1_Panthera_onca
</span><span id="__span-21-16"><a id="__codelineno-21-16" name="__codelineno-21-16" href="#__codelineno-21-16"></a> |                           ,|
</span><span id="__span-21-17"><a id="__codelineno-21-17" name="__codelineno-21-17" href="#__codelineno-21-17"></a>_|                      _____||____ EF437587.1_Panthera_tigris
</span><span id="__span-21-18"><a id="__codelineno-21-18" name="__codelineno-21-18" href="#__codelineno-21-18"></a> |                     |     |
</span><span id="__span-21-19"><a id="__codelineno-21-19" name="__codelineno-21-19" href="#__codelineno-21-19"></a> |            _________|     |_______ EF437583.1_Uncia_uncia
</span><span id="__span-21-20"><a id="__codelineno-21-20" name="__codelineno-21-20" href="#__codelineno-21-20"></a> |           |         |
</span><span id="__span-21-21"><a id="__codelineno-21-21" name="__codelineno-21-21" href="#__codelineno-21-21"></a> |___________|         |________ EF437585.1_Panthera_leo
</span><span id="__span-21-22"><a id="__codelineno-21-22" name="__codelineno-21-22" href="#__codelineno-21-22"></a> |           |
</span><span id="__span-21-23"><a id="__codelineno-21-23" name="__codelineno-21-23" href="#__codelineno-21-23"></a> |           |______________ EF437589.1_Panthera_pardus
</span><span id="__span-21-24"><a id="__codelineno-21-24" name="__codelineno-21-24" href="#__codelineno-21-24"></a> |
</span><span id="__span-21-25"><a id="__codelineno-21-25" name="__codelineno-21-25" href="#__codelineno-21-25"></a> |______________ EF437591.1_Felis_catus
</span></code></pre></div>
<p>The outside group is the house cat. Because everyone else is a big cat.<br>
Fundamentally our tree is similar to that published in articles.<br>
In foreign colleagues the tree was based on several genes, we take only 1 piece of data.<br></p>
<p><strong>Beauti</strong></p>
<p><code>Beauti</code> is the GUI application. So I will just provide as many screenshots as possible.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2021.59.15.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2021.59.15.png" align='center', width="50%"></a>
</div>

<p>When loading the file, we select that we have nucleotide sequences</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2021.59.21.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2021.59.21.png" align='center', width="50%"></a>
</div>

<p>Everything is okay.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.01.40.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.01.40.png" align='center', width="50%"></a>
</div>

<p>In <code>Site model</code> select TN93 and empirical frequencies</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.02.46.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.02.46.png" align='center', width="50%"></a>
</div>

<p>In <code>Clock model</code> we choose 0.02. Why? Because we rely on the known data on the frequency of substitutions in mtDNA (approximately 2% per million years)</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.06.15.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.06.15.png" align='center', width="50%"></a>
</div>

<p>Everything is okay.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.07.12.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.07.12.png" align='center', width="50%"></a>
</div>

<p>Save everything to <code>felidae_2percent.xml</code>.</p>
<p><strong><em>Input</em></strong></p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a>beast<span class="w"> </span>data/atp8/felidae_2percent.xml
</span></code></pre></div>
<p><strong>Tracer</strong></p>
<p><code>Tracer</code> is the GUI application. So I will just provide as many screenshots as possible.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.37.25.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.37.25.png" align='center', width="50%"></a>
</div>

<p>All <code>ESS</code> scores are in perfect order.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.38.23.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.38.23.png" align='center', width="50%"></a>
</div>

<p>The so-called "hairy caterpillar".</p>
<p><strong>TreeAnnotator</strong></p>
<p><code>TreeAnnotator</code> is the GUI application. So I will just provide as many screenshots as possible.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.51.48.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.51.48.png" align='center', width="50%"></a>
</div>

<p>Set parameters, and set <code>input</code> and <code>output</code>. Useful hint - output can be named the same way, but not .trees, just .tree!</p>
<p><strong>FigTree</strong></p>
<p><code>FigTree</code> is the GUI application. So I will just provide as many screenshots as possible.</p>
<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.59.15.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/Screenshot%202024-03-07%20at%2022.59.15.png" align='center', width="50%"></a>
</div>

<div style='justify-content: center'>
<a class="glightbox" href="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/felidae_atp8.tree.png" data-type="image" data-width="auto" data-height="auto" data-desc-position="bottom"><img src="https://github.com/iliapopov17/NGS-Handbook/blob/main/04_Phylogenetics/04_05_Root_Date/imgs/felidae_atp8.tree.png" align='center', width="50%"></a>
</div>

<p>Fiddled with the parameters and got these results.</p>
<details class="success" open="open">
<summary>Success</summary>
<p>Well. The common ancestor of our smoky leopards is about 2.5 million years old.</p>
</details>
<hr />
<details class="question" open="open">
<summary>Question</summary>
<p>Compare the results of this analysis (age of the last common ancestor of <em>Neofelis</em>) with published articles (https://www.science.org/doi/10.1126/sciadv.adh9143, https://www.sciencedirect.com/science/article/pii/S2589004222019198).<br>
What conclusions can be drawn?</p>
</details>
<details class="success" open="open">
<summary>Success</summary>
<p>In the first article - https://www.science.org/doi/10.1126/sciadv.adh9143 there was a full genome analysis. Their estimate of the age of the common ancestor of smoky leopards is 2.2 million years.<br>
And we hit 100 nucleotides pretty good!<br></p>
<p>But in the second article - https://www.sciencedirect.com/science/article/pii/S2589004222019198 - the age of the ancestor is 5.1 million years old.... Well. Interesting. Can't explain it yet.<br></p>
</details>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 Ilia Popov
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy", "search", "search.suggest", "search.highlight"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body>
</html>